<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <title>DRAFT_insectIndicators.knit</title>

  
   
   <meta name="generator" content="placeholder" />
  <meta property="og:title" content="DRAFT_insectIndicators.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="DRAFT_insectIndicators.knit" />
  
  
  
  <!-- JS -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script>
  <script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script>
    <script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet" />
    <script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script>
    <script src="libs/bs3compat-0.4.2/transition.js"></script>
    <script src="libs/bs3compat-0.4.2/tabs.js"></script>
    <script src="libs/bs3compat-0.4.2/bs3compat.js"></script>
    <link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet" />
    <script src="libs/bs4_book-1.0.0/bs4_book.js"></script>
    <script src="libs/kePrint-0.0.1/kePrint.js"></script>
    <link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
    <script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script>

  <!-- CSS -->
  <style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
      <link rel="stylesheet" href="style.css" />
  
</head>

<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<!--bookdown:title:start-->
<!--bookdown:title:end-->

<!--bookdown:toc:start-->
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book">
    <a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title=""></a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
      </form>

      <nav aria-label="Table of contents">
        <h2>Table of contents</h2>
        <div id="book-toc"></div>

        <div class="book-extra">
          <p><a id="book-repo" href="#">View book source <i class="fab fa-github"></i></a></li></p>
        </div>
      </nav>
    </div>
  </header>

  <main class="col-sm-12 col-md-9 col-lg-7" id="content">
<!--bookdown:toc:end-->
<!--bookdown:body:start-->
<!-- This is a template for how to document the indicator analyses. 

Make a copy of this file and store it in the root folder (same as where this template is stored) folder. 
Name te file DRAFT_myIndicatorName.Rmd. 

Because this is a stand-alone Rmd file inside a bookdown project, in order to knit the file for preview purposes,
you need to type rmarkdown::render("template.Rmd") in the console, 
as the the usual Ctrl + Alt + k will not work. 

Make sure also to not change the order, or modify the headers, unless you really need to. This is because it
easier to read the web page as a whole of the indicators are presented using the same layout. 

When the analyses is complete, Anders Kolstad, or someone else, will perform a rapid code review, making sure 
the analysis is repeatable by someone else. Be therefore very careful to annotate 
and comment on every step in the analysis. When completed, Anders will make a copy
of the working file and put it in the same project root folder, but removing the 'DRAFT_' part of the name. 
After modifying it slightly (mainly removing the top YAML header and updating the rmd_files argument in '_bookdown.yml') 
the output will become available on the website (https://ninanor.github.io/ecosystemCondition/). -->
<!-- Examples: 

For an example, see 'indicators/breareal-workingFile.Rmd' 
and the subsequent 'breareal.Rmd' 
which again, when rendered, produces this:
https://ninanor.github.io/ecosystemCondition/areal-av-isbreer.html

Notice the degree of annotation.

-->
<!-- Replace 'indicator name' with yor own indicator name  -->
<div id="insect-indicators" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Insect indicators</h1>
<p><br /></p>
<p><em>Author and date:</em>
Jens Åström 2023-10-23</p>
<p><br /></p>
<!--# Notes on this draft
Most things are done but some things needs updates or touching up.

Incomplete list of things that will change:

* Upgrade genetic diversity calculation. The current calculation is just a placeholder, awaiting some new data from Marie. Waiting for software installation by IT. Unsure of timeline.
* Calculate NorIns indicators on 5 year interval. (Moving window?)
* Fix remaining comments in this draft inside hard brackets.-->
<!-- Load all you dependencies here -->
<!-- Fill in which ecosystem the indicator belongs to, as well as the ecosystem characteristic it should be linked to. It's OK to use some Norwegian here -->
<table>
<thead>
<tr>
<th style="text-align:left;">
Ecosystem
</th>
<th style="text-align:left;">
Økologisk egenskap
</th>
<th style="text-align:left;">
ECT class
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Semi-naturlig mark
</td>
<td style="text-align:left;">
Biologisk mangfold, Funksjonelle grupper innen trofiske nivåer, Funksjonelt viktige arter og biofysiske strukturer
</td>
<td style="text-align:left;">
Functional state characteristics
</td>
</tr>
<tr>
<td style="text-align:left;">
Semi-naturlig mark
</td>
<td style="text-align:left;">
Biologisk mangfold, Funksjonelle grupper innen trofiske nivåer, Funksjonelt viktige arter og biofysiske strukturer
</td>
<td style="text-align:left;">
Functional state characteristics
</td>
</tr>
<tr>
<td style="text-align:left;">
Semi-naturlig mark
</td>
<td style="text-align:left;">
Biologisk mangfold, Funksjonelle grupper innen trofiske nivåer, Funksjonelt viktige arter og biofysiske strukturer
</td>
<td style="text-align:left;">
Functional state characteristics
</td>
</tr>
</tbody>
</table>
<!-- Don't remove these three html lines -->
<br />
<br />
<hr />
<!-- Document you work below. Try not to change  the headers too much. Data can be stored on NINA server. Since the book is rendered on the R Server this works fine, but note that directory paths are different on the server compared to you local machine. If it is not too big you may store under /data/ on this repository -->
<div id="introduction" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Introduction</h2>
<p>We here show how to calculate a set of indicators for terrestrial insects in Norway. Two of them are adapted from earlier work on the Nature Index for Norway and use data from the Norwegian monitoring program on bumblebees and butterflies (<strong>NMBB</strong>). The others are developed during 2023, and use data from the Norwegian insect monitoring program (<strong>NorIns</strong>).</p>
<p>This workflow is shortened by putting most of the underlying code in separate R-packages, which are freely available through Github. The relevant packages are <code>bombLepiSurv</code> <a href="https://github.com/jenast/bombLepiSurv" class="uri">https://github.com/jenast/bombLepiSurv</a> for the bumblebee and butterfly indicators (<strong>Not public yet</strong>), and <code>Norimon</code> <a href="https://github.com/jenast/Norimon" class="uri">https://github.com/jenast/Norimon</a> for the rest of the indicators.</p>
<p>For the indicators from the NMBB program, community reference values has been elicited from experts, and is explained further below. For the indicators from the NorIns program, this is significantly harder to do, because of the size of the communities, lack of historical timeseries and changes in sampling and identification techniques. We therefore currently lack reference values for many of these indicators.</p>
<p>Below is a list of the indicators calculated in this document, and their state of development. Some details is further expanded on below.</p>
<table>
<thead>
<tr>
<th style="text-align:left;">
Dataset
</th>
<th style="text-align:left;">
Indicator
</th>
<th style="text-align:left;">
Norwegian type
</th>
<th style="text-align:left;">
ECT
</th>
<th style="text-align:left;">
Current geographical scope
</th>
<th style="text-align:left;">
Indicator data
</th>
<th style="text-align:left;">
Reference values
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
NMBB
</td>
<td style="text-align:left;">
Bumblebees in semi-natural grasslands
</td>
<td style="text-align:left;">
Funksjonelt viktige arter og biofysiske stukturer
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østfold-Vestfold, Vestlandet, Trøndelag
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Present
</td>
</tr>
<tr>
<td style="text-align:left;">
NMBB
</td>
<td style="text-align:left;">
Butterflies in semi-natural grasslands
</td>
<td style="text-align:left;">
Funksjonelt viktige arter og biofysiske stukturer
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østfold-Vestfold, Vestlandet, Trøndelag
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Present
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Biomass of flying insects
</td>
<td style="text-align:left;">
Funksjonelt viktige arter og biofysiske stukturer
</td>
<td style="text-align:left;">
B2
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Species richness of flying insects
</td>
<td style="text-align:left;">
Biologisk mangfold
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Species richness of pollinating insects
</td>
<td style="text-align:left;">
Biologisk mangfold
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Species richness of dung associated insects
</td>
<td style="text-align:left;">
Biologisk mangfold
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Quotent of Apocrita vs Symphyta species richness
</td>
<td style="text-align:left;">
Funksjonelt viktige arter og biofysiske strukturer
</td>
<td style="text-align:left;">
B2
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Number of alien species
</td>
<td style="text-align:left;">
Biologisk mangfold
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
<tr>
<td style="text-align:left;">
NorIns
</td>
<td style="text-align:left;">
Intraspesific genetic variation
</td>
<td style="text-align:left;">
Biologisk mangfold
</td>
<td style="text-align:left;">
B1
</td>
<td style="text-align:left;">
Sørlandet, Østlandet, Trøndelag, Nord-Norge
</td>
<td style="text-align:left;">
Present
</td>
<td style="text-align:left;">
Lacking
</td>
</tr>
</tbody>
</table>
<p>Many of these indicators are self-explanatory and don’t require much justification. <strong>Bumblebees in semi-natural grasslands</strong> and <strong>Butterflies in semi-natural grasslands</strong> are functionally important pollinators, and are as a community dependent on a varied herbaceous flora. They therefore both represent key functionality, as well as indicators of habitat quality. <strong>Biomass of flying insects</strong> represent a key functional characteristic in ecosystem functioning as food for insectivores, and as a response to habitat quality and management. <strong>Species richness of flying insects</strong> represents the total breadth of the insect community, and <strong>Species richness of pollinating insects</strong> complements the butterfly and bumblebee indicators in taxonomic breadth and spatial resolution.
<strong>Species richness of dung associated insects</strong> represent a key functional group in decomposition, and function as an indicator of grazing intensity and diversity. <strong>Quotent of Apocrita vs Symphyta species richness</strong> tries to go beyond simple amounts, and capture compositional changes in the insect community. Apocrita covers predatory and parasitic wasps, while Symphyta is plant-feeding wasps. A relative decrease in predatory/parasitic wasps could indicate a degradation in insect prey communities, and a decrease in the potential for natural pest control. <strong>Number of alien species</strong> takes into account the number of alien species found in a region and optionally, the frequency of sampling events they are observed in. This is based on the alien species list from the Norwegian Biodiversity Information Center, which lists known alien species and an increasing list of door-knocker species. Note that due to large historical knowledge gaps of insect species distributions, there could potentially be many more alien species in Norway, that are not picked up on this alien species list. Lastly, <strong>Intraspecific genetic variation</strong> captures biodiversity that until now has been hidden, due to lack of practical technologies to measure it. The need to conserve genetic diversity is identified in Norwegian law, and decreased intraspecific diversity could signal population decreases, and decreased potential to adapt to environmental changes, before whole species are lost to an area.</p>
</div>
<div id="about-the-underlying-data" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> About the underlying data</h2>
<p>The data used here comes from the ongoing monitoring programs “Norwegian insect monitoring” (<a href="https://www.nina.no/Naturmangfold/Insekter/Overv%C3%A5king-av-insekter">NorIns</a>) and “Norwegian monitoring of bumblebees and butterflies” (<a href="https://www.nina.no/V%C3%A5re-fagomr%C3%A5der/Milj%C3%B8overv%C3%A5king-p%C3%A5-land/Humler-og-dagsommerfugler">NMBB</a>). Both programs are initiated and financed by the Environmental agency, with the aim to produce continuous, areal representative, time-series of insect community data. The <strong>NorIns</strong> program use malaise and window traps to sample a broad community of insect species, which are identified through metabarcoding of DNA. The <strong>NMBB</strong> program use sweep netting along transect walks where the species are identified manually in the field. More information is available in the links above.</p>
<div id="representativity-in-time-and-space" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Representativity in time and space</h3>
<p>The <strong>NorIns</strong> program started in 2020 with monitoring in semi-natural land and forests in the south-eastern region of Norway (Østlandet), with the long-term ambition to cover the entire country. In 2021, the program was expanded to semi-natural land in central Norway (Trøndelag), in 2022 to semi-natural land in south-western Norway (Sørlandet), and in 2023 to semi-natural land in northern Norway (Nord-Norge), with a possible expansion to western Norway (Vestlandet) in 2024. So far, forest habitats are only monitored in south-eastern Norway.</p>
<p>These 5 regions represents the planned spatial precision of the survey. Finer resolution could be evaluated in the future, although it is unlikely to be feasible without more intense sampling. This program uses a staggered sampling scheme where we visit different locations each year, and return to each location every 5 years. This sampling scheme, coupled with the strong yearly variation in insect abundances makes it reasonable to use a temporal resolution of 5 years. [How to deal with this now? average over 5 years? Add 5 year aggregation in bootstrap_value()]</p>
<p>The <strong>NMBB</strong> program started in 2009 in a subset of south-eastern Norway (counties of Vestfold and Østfold), and was expanded to central Norway (Trøndelag) in 2010, south-western Norway in 2013, and to western Norway (Vestland and Møre og Romsdal fylke) in 2022. The spatial resolution aligns with these (as of now) 4 regions, and it is not recommended to downscale the predictions due to the limited number of sampling locations in each region. Since the same locations are visited each year, yearly estimates of the condition could be calculated, but should be interpreted by caution due to natural yearly variability.</p>
<p>Both programs are designed to be area representative, within their respective habitat types and regions. Both use semi-randomized selection schemes of localities, the <strong>NorIns</strong> selects by random draws, and the <strong>NMBB</strong> by a predefined continuous network of survey squares. In both programs however, some subjective decisions are made to define habitat criteria and to discard unpractical survey locations.</p>
</div>
<div id="original-units" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Original units</h3>
<p>The original units from <strong>NorIns</strong> include number of species, total biomass as wet weight, and inter-specific genetic diversity (unit to be determined). For the <strong>NMBB</strong> program, original units are number of individuals of each species within a transect, although only presences in transects (as presence frequencies) are used for the indicator calculations.</p>
</div>
<div id="temporal-coverage" class="section level3" number="1.2.3">
<h3><span class="header-section-number">1.2.3</span> Temporal coverage</h3>
<p>2009-present for the <strong>NMBB</strong> data, and 2020-present for the <strong>NorIns</strong> data.</p>
</div>
</div>
<div id="ecosystem-characteristic" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Ecosystem characteristic</h2>
<div id="norwegian-standard" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Norwegian standard</h3>
<p>Four of the indicators are tied to the <em>Økologisk egenskap</em> (Ecological characteristic) of <strong>Funksjonelt viktige arter og biofysiske stukturer</strong> (Funktionally important species and structures), and the other four to <strong>Biologisk mangfold</strong> (Biodiversity).</p>
</div>
<div id="un-standard" class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> UN standard</h3>
<p>Six of the indicators are tied to the UN standard (ECT-classes) <strong>B1 Compositional state characteristics</strong>, and the two remaining to <strong>B2 Structural state characteristics</strong>.</p>
</div>
</div>
<div id="collinearities-with-other-indicators" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Collinearities with other indicators</h2>
<p>None measured so far, but it is reasonable to assume that some of these indicators will share some causal relationships with each other, and therefore correlate to some degree. They might also correlate to other indicators, not described here, as well.</p>
</div>
<div id="reference-condition-and-values" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Reference condition and values</h2>
<p>The reference condition is meant to be one with minimal negative human impact, but this has a special interpretation for semi-natural land. These habitats are formed and maintained through human activities. Thus, “minimal negative human impact” is replaced by a state of “good maintenance”, here understood as resembling the traditional agricultural maintenance regime, existing for several hundred years up until the late 1800s. This is characterized by extensive grazing, meadows and lays, crop rotation with legumes to bind nitrogen in addition to fertilization from manure, relatively small field sizes and abundant field margins, and a lack of artificial fertilization and mechanized tilling.</p>
<div id="reference-condition" class="section level3" number="1.5.1">
<h3><span class="header-section-number">1.5.1</span> Reference condition</h3>
<p>Assessing the current state of insect communities may be a complicated task, due to their taxonomic and functional breadth and high temporal and spatial variability. Still, even more challenging is to establishing reference values, when the reference states are extinct or prohibitively difficult to measure empirically. We currently lack reference values for all indicators from the <strong>NorIns</strong> program. It is unlikely that this can be solved “simply” by surveying a state in a reference area, since intact reference areas of sufficient size and numbers likely no longer exist. The question of how to handle these reference values is currently unresolved, and we present these indicator value calculations with made up reference values as placeholders in the code.</p>
<p>For the <strong>NMBB</strong> indicators, the communities are small enough and well known enough to identify reference communities by expert opinion. These communities specify the expected rate of observing each species. The current rates of observation are then compared to the expected rates. The calculations are further explained below.</p>
</div>
<div id="reference-values-thresholds-for-defining-good-ecological-condition-minimum-andor-maximum-values" class="section level3" number="1.5.2">
<h3><span class="header-section-number">1.5.2</span> Reference values, thresholds for defining <em>good ecological condition</em>, minimum and/or maximum values</h3>
<!-- Text here -->
<p>The specific reference communities for the <strong>NMBB</strong> indicators won’t be spelled out here, see instead the scripts below. We have not specified a value for <em>good ecological condition</em> as of yet, since there lack a straightforward empirical basis to do so.</p>
</div>
</div>
<div id="uncertainties" class="section level2" number="1.6">
<h2><span class="header-section-number">1.6</span> Uncertainties</h2>
<p>The uncertainties for the indicator values is calculated by bootstrapping the average values for each locality within a year. This takes the variability between the localities into account. For the <strong>NorIns</strong> indicators, this uncertainties are given as standard deviation, as well as confidence intervals.The uncertainties for the <strong>NMBB</strong> indicators are constituted by a distribution of discrete values, and therefore we only provide confidence intervals and not a standard deviation.</p>
</div>
<div id="references" class="section level2" number="1.7">
<h2><span class="header-section-number">1.7</span> References</h2>
<ul>
<li><p>French, C. M., Bertola, L. D., Carnaval, A. C., Economo, E. P., Kass, J. M., Lohman, D. J., … Hickerson, M. J. 2023, juli 17. Global determinants of insect mitochondrial genetic diversity. bioRxiv. bioRxiv. <a href="https://doi.org/10.1101/2022.02.09.479762" class="uri">https://doi.org/10.1101/2022.02.09.479762</a></p></li>
<li><p>Theodoridis, S., Fordham, D. A., Brown, S. C., Li, S., Rahbek, C., &amp; Nogues-Bravo, D. 2020. Evolutionary history and past climate change shape the distribution of genetic diversity in terrestrial mammals. Nature Communications 11(1): 2557. <a href="https://doi.org/10.1038/s41467-020-16449-5" class="uri">https://doi.org/10.1038/s41467-020-16449-5</a></p></li>
<li><p>Åström, Jens; Birkemoe, Tone; Brandsegg, Hege; Dahle, Sondre; Davey, Marie Louise; Ekrem, Torbjørn; Fossøy, Frode; Hanssen, Oddvar; Laugsand, Arne Endre; Majaneva, Markus; Staverløkk, Arnstein; Sverdrup-Thygeson, Anne; Ødegaard, Frode.
Insektovervåking på Østlandet, Sørlandet og i Trøndelag. Rapport fra feltsesong 2022. Trondheim: Norsk institutt for naturforskning (NINA) 2023 (ISBN 978‐82‐426‐5037‐5) 99 s. NINA rapport(2241)</p></li>
<li><p>Åström, Sandra Charlotte Helene; Åström, Jens; Bøhn, Kristoffer; Gjershaug, Jan Ove; Staverløkk, Arnstein; Dahle, Sondre; Ødegaard, Frode. Nasjonal overvåking av dagsommerfugler og humler i Norge. Oppsummering av aktiviteten i 2022. Trondheim: Norsk institutt for naturforskning (NINA) 2023 (ISBN 978-82-426-5009-2) 54 s. NINA rapport(2214)</p></li>
<li><p>Öberg, S., Gjershaug, J. O., Diserud, O. &amp; Ødegaard, F. 2011. Videreutvikling av metodikk for
arealrepresentativ overvåking av dagsommerfugler og humler. Naturindeks for Norge. – NINA
Rapport 663.</p></li>
</ul>
</div>
<div id="analyses" class="section level2" number="1.8">
<h2><span class="header-section-number">1.8</span> Analyses</h2>
<div id="calculation-principles-for-norins-indicators" class="section level3" number="1.8.1">
<h3><span class="header-section-number">1.8.1</span> Calculation principles for NorIns indicators</h3>
<p>The indicators from the Norwegian insect monitoring are calculated with functions in the <code>Norimon</code> R-package <a href="https://github.com/NINAnor/Norimon">link</a>. This methodology is meant to facilitate the calculation of a broad variety of insect indicators using data from <strong>NorIns</strong>.</p>
<p><strong>Main steps of workflow:</strong></p>
<ol style="list-style-type: decimal">
<li>Fetch data of biomass or community diversity from a centralized database. Diversity data can be filtered on several taxonomic levels, but biomass is only available for whole samples.</li>
<li>Bootstrap observations and calculate mean and uncertainty of point estimates.</li>
<li>Compare observations to reference points.</li>
<li>Display and plot the results</li>
</ol>
<p><img src="images/NorIns_flow_chart.png" width="400pt" /></p>
<p>Most of the ecology comes into step 1, in choosing what data to use to describe a quality. This can be a custom selection of species, or a set of higher taxa such as specific genuses, families or even order. The goal is to choose a set of taxa that represent specific qualities of the community that can indicate the ecological status of the ecosystem. In addition to the selection of taxa, we have to decide on the level of spatial and temporal aggregation, e.g. if we should aggregate the data spatially on the region scale, habitat scale or locality scale, and if we should aggregate it temporally on yearly or even the individual sample occasions within years. For the ecological indicators, we will aggregate the raw observation values to the year locality scale, where the data summarize all catches in a locality in a year.</p>
<p>The second step is to get an estimate of the uncertainty of the data, were we use the bootstrap method. This method is also flexible, as we could bootstrap the samples on different sampling levels (e.g. samples within localities, localities within a region, or regions within country). When working with raw data on the year locality level, the most granular bootstrap would be to summarize the variables to the yearly, habitat, and regional level, for example expressing the mean values of insect biomass in semi-natural land and forests in Østlandet in 2021, with uncertainty.</p>
<p>Ecological knowledge also comes into step 3, comparing the values to a reference state. Here we have several options. We could for example use a single defined value as the reference state. But we can also use a point in time as a reference point (e.g. the start of a time series).</p>
<p>Taken together, this framework is meant to facilitate the calculation of an arbitrary set of insect indicators, based on the combination of choices in data to fetch, aggregation level, and reference comparison. We can work through an example with pollinators, to showcase the framework.</p>
<p>In the example, we start with connecting to the database in order to fetch the data.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">connect_to_insect_db</span>()</span></code></pre></div>
<p>For convenience, the pollinator families can be retrieved by the <code>get_pollinators()</code> function.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>pollinators <span class="ot">&lt;-</span> <span class="fu">get_pollinators</span>()</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>pollinators_fam <span class="ot">&lt;-</span> pollinators <span class="sc">%&gt;%</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(family_latin) <span class="sc">%&gt;%</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:left;">
family_norwegian
</th>
<th style="text-align:left;">
family_latin
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Gravebier
</td>
<td style="text-align:left;">
Andrenidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Langtungbebier
</td>
<td style="text-align:left;">
Apidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Korttungebier
</td>
<td style="text-align:left;">
Colletidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Markbier
</td>
<td style="text-align:left;">
Halictidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Buksamlerbier
</td>
<td style="text-align:left;">
Megachilidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Blomsterbier
</td>
<td style="text-align:left;">
Melittidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Blomfluer
</td>
<td style="text-align:left;">
Syrphidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Smygere
</td>
<td style="text-align:left;">
Hesperiidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Glansvinger
</td>
<td style="text-align:left;">
Lycaenidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Nymfevinger
</td>
<td style="text-align:left;">
Nymphalidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Svalestjerter
</td>
<td style="text-align:left;">
Papilionidae
</td>
</tr>
<tr>
<td style="text-align:left;">
Hvitvinger
</td>
<td style="text-align:left;">
Pieridae
</td>
</tr>
<tr>
<td style="text-align:left;">
Metallmerker (uoffisiell)
</td>
<td style="text-align:left;">
Riodinidae
</td>
</tr>
</tbody>
</table>
<p>We then fetch the community data for these families through the <code>get_observations</code> function. We additionally subset the habitat type that we are interested in, here semi-natural land. The result is a tibble of the aggregated number of species, shannon diversity, and the mean number of genetic variants per species. The default aggregation level is “year_locality”, meaning the total observations for a locality within a year.</p>
<p>Note that the output currently contains some experimental values, awaiting a more robust methodology. “Shannon diversity” doesn’t really make sense without counts or amounts, etc. This will be replaced by measurements of genetic diversity that are based on published peer-reviewed methodology (Connor et al. 2023), in the (hopefully near) future.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>poll_loc_year <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_families =</span> pollinators_fam,</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>poll_loc_year <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>kableExtra<span class="sc">::</span><span class="fu">kbl</span>()</span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:right;">
year
</th>
<th style="text-align:left;">
locality
</th>
<th style="text-align:left;">
habitat_type
</th>
<th style="text-align:left;">
region_name
</th>
<th style="text-align:right;">
no_species
</th>
<th style="text-align:right;">
shannon_div
</th>
<th style="text-align:right;">
mean_no_asv_per_species
</th>
<th style="text-align:right;">
GDE_by_asv
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_01
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
49.47
</td>
<td style="text-align:right;">
2.68
</td>
<td style="text-align:right;">
0.50
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_02
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
28.64
</td>
<td style="text-align:right;">
1.44
</td>
<td style="text-align:right;">
0.84
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_03
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
34.07
</td>
<td style="text-align:right;">
1.57
</td>
<td style="text-align:right;">
0.85
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_04
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:right;">
42.90
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0.77
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_05
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
78
</td>
<td style="text-align:right;">
52.19
</td>
<td style="text-align:right;">
1.90
</td>
<td style="text-align:right;">
0.67
</td>
</tr>
</tbody>
</table>
<div id="bootstrap-observations" class="section level4" number="1.8.1.1">
<h4><span class="header-section-number">1.8.1.1</span> Bootstrap observations</h4>
<p>The localities in the NorIns program are (semi-)randomly selected and can be viewed as samples of a larger population. Individually, they represent a single measurement of an insect community in a habitat type in a region. To get the sampling uncertainty for this representation, we can bootstrap the values by this simple process: we choose a random set of localities within a year and region (with replacement) and calculate the average values. We then repeat this process a large number of times to get a bootstrap sample of values, which can be used to express the uncertainty in the dataset.</p>
<p>We typically have 10 localities for a given habitat type and region each year, which is not very much to base our uncertainty estimates on. But bootstrapping works fairly well on small samples and has the advantage that is doesn’t make assumptions of the statistical distribution of the errors. We will use this level to showcase the functionality and to visualize some variation between years. However, since the sampling scheme semi-randomly selects 10 new localities for each of the 5 years the rotating scheme, each year will include some randomness that is caused by variability between localities. This means that the values between individual years might appear more random than the underlying overall trend. To get robust estimates that averages over this variability, we will summarize the actual indicator values to 5 year periods, further described below.</p>
<p>When <strong>NorIns</strong> is fully scaled up, it will have 50 localities for each covered habitat type in every country region, with 10 localities every year, in a rotating survey scheme over 5 years. Since the program is rolled out sequentially starting in 2020, for semi-natural land it currently covers 4/5 of the country, with varying amounts of localities within each region. The map below displays the localities surveyed so far.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>semi_loc <span class="ot">&lt;-</span> <span class="fu">get_localities</span>(<span class="at">dataset =</span> <span class="st">&quot;NasIns&quot;</span>,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">habitat_type =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>norway_regions <span class="ot">&lt;-</span> <span class="fu">get_map</span>()</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tm_shape</span>(norway_regions) <span class="sc">+</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_polygons</span>(<span class="st">&quot;region&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_shape</span>(semi_loc) <span class="sc">+</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_symbols</span>(<span class="at">col =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">shape =</span> <span class="dv">22</span>,</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/location_map-1.png" width="672" /></p>
<p>The bootstrap routine is implemented in the function <code>bootstrap_value()</code>, which takes a community (from <code>get_observation</code>) or weight dataset (from <code>get_biomass</code>) as its first input. It also needs to know what measurement in the dataset to bootstrap, and what, if any, grouping structure to aggregate the results on. In this example, we bootstrap the number of pollinator species, and aggregate the results on the year and regional scale.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>poll_richness_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(poll_loc_year,</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">value =</span> no_species,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">rolling_year_window =</span> <span class="cn">FALSE</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                      )</span></code></pre></div>
<p>This creates an object of type <code>boot_stat</code>. Calling it prints a simple summary of the bootstrap values.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>poll_richness_boot</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Østlandet         60.5    6.25          48.7           72.6
## 2  2021 Trøndelag         41.9    2.98          35.9           47.2
## 3  2021 Østlandet         53.7    6.07          41.2           64.9
## 4  2022 Sørlandet         29.6    5.59          18.3           40.6
## 5  2022 Trøndelag         34.8    2.51          29.9           39.7
## 6  2022 Østlandet         35.9    2.82          30.3           41.3</code></pre>
<p>But the <code>boot_stat</code> object also stores the individual bootstrap values for later computation. By default, we use 999 bootstrap samples, which here results in 999 samples * 6 groups = 5994 rows of bootstrap values.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>poll_richness_boot[<span class="dv">2</span>]</span></code></pre></div>
<pre><code>## $bootstrap_values
## # A tibble: 5,994 × 3
## # Groups:   year, region_name [6]
##     year region_name boot_values
##    &lt;int&gt; &lt;chr&gt;             &lt;dbl&gt;
##  1  2020 Østlandet          49.8
##  2  2020 Østlandet          67  
##  3  2020 Østlandet          65  
##  4  2020 Østlandet          61  
##  5  2020 Østlandet          57.1
##  6  2020 Østlandet          65.6
##  7  2020 Østlandet          59.8
##  8  2020 Østlandet          64.3
##  9  2020 Østlandet          45.3
## 10  2020 Østlandet          63.3
## # ℹ 5,984 more rows</code></pre>
</div>
<div id="comparing-bootstrap-values-to-a-reference-point" class="section level4" number="1.8.1.2">
<h4><span class="header-section-number">1.8.1.2</span> Comparing bootstrap values to a reference point</h4>
<p>Setting aside the practical difficulties in establishing robust reference values, the next step in the methodology is to compare the observed values (with bootstrapped uncertainty) to a chosen reference value. This can be made in several ways. Most simply, if we have a single numeric value as a reference value, we can simply subtract that from the observed values. For example, if we observe 13 species of pollinators in 2022 at a location, and the reference point is 10, the 2022 value has increased by 13 - 10 = 3 species. Such subtractions should be made on the set of bootstrapped values, followed by new summary statistics being calculated, preserving the uncertainty from the bootstrap. The <code>boot_stat</code> has its own subtraction method <code>-</code> do to just that. In this example, we set the reference value arbitrarily to 30 species.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>poll_richness_boot</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Østlandet         60.5    6.25          48.7           72.6
## 2  2021 Trøndelag         41.9    2.98          35.9           47.2
## 3  2021 Østlandet         53.7    6.07          41.2           64.9
## 4  2022 Sørlandet         29.6    5.59          18.3           40.6
## 5  2022 Trøndelag         34.8    2.51          29.9           39.7
## 6  2022 Østlandet         35.9    2.82          30.3           41.3</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot <span class="ot">&lt;-</span> poll_richness_boot <span class="sc">-</span> <span class="dv">30</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Østlandet       30.5      6.25        18.7             42.6
## 2  2021 Trøndelag       11.9      2.98         5.9             17.2
## 3  2021 Østlandet       23.7      6.07        11.2             34.9
## 4  2022 Sørlandet       -0.423    5.59       -11.7             10.6
## 5  2022 Trøndelag        4.77     2.51        -0.100            9.7
## 6  2022 Østlandet        5.95     2.82         0.300           11.3</code></pre>
<p>Alternatively, we could use a reference point in the time series itself. Say for example that we want to use the values for species richness of pollinators in semi-natural land i Østlandet 2020 as a reference point. We can then calculate the difference (the contrast) between this level and all the other levels. We do this by the function <code>boot_contrast()</code></p>
<p><strong>NB! This functionality is in development. It currently works for single rows as reference points, but needs updating to allow for referencing several values simultaneously, e.g using the start values for all regions and habitat types as their own reference points.</strong></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot2 <span class="ot">&lt;-</span> poll_richness_boot <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boot_contrast</span>(year <span class="sc">==</span> <span class="dv">2020</span> <span class="sc">&amp;</span> region_name <span class="sc">==</span> <span class="st">&#39;Østlandet&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot2</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Østlandet         0       0              0              0  
## 2  2021 Trøndelag       -18.6     6.92         -32.6           -5.1
## 3  2021 Østlandet        -6.82    8.74         -23.3            9.9
## 4  2022 Sørlandet       -31.0     8.46         -48.4          -14.8
## 5  2022 Trøndelag       -25.8     6.79         -38.8          -12.3
## 6  2022 Østlandet       -24.6     6.85         -38.3          -12</code></pre>
</div>
<div id="normalizing-the-values" class="section level4" number="1.8.1.3">
<h4><span class="header-section-number">1.8.1.3</span> Normalizing the values</h4>
<p>After the comparison to a reference value, we need to normalize the indicator values so that they lie between 0 and 1. This can be done in several ways (se e.g. <a href="https://github.com/NINAnor/eaTools">eaTools</a>). The simplest case is to use a linear scaling, with a natural zero, which e.g. can be done by dividing the indicator values by the highest value state. A <code>boot_stat</code> class has a <code>/</code> function that divides each bootstrap value by a given value, truncates the highest values to 1, and recalculates the summary values. We can also simply divide by the reference state, to compare the values to the reference state, and normalize them in one go. Here, we exemplify the method by dividing the value by 30.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot3 <span class="ot">&lt;-</span> poll_richness_boot <span class="sc">/</span> <span class="dv">30</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot3</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##    year region_name no_species  boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;    &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Østlandet        1     0                1                  1
## 2  2021 Trøndelag        1.00  0.000527         1                  1
## 3  2021 Østlandet        1     0                1                  1
## 4  2022 Sørlandet        0.919 0.112            0.61               1
## 5  2022 Trøndelag        1.00  0.00407          0.997              1
## 6  2022 Østlandet        1.00  0.00464          1                  1</code></pre>
</div>
<div id="display-and-plot-bootstrap-values" class="section level4" number="1.8.1.4">
<h4><span class="header-section-number">1.8.1.4</span> Display and plot bootstrap values</h4>
<p>The boot_stat class also has its own plot function. It tries to plot a comparison of the bootstrap distributions over years, for each group. For example, if we plot the object <code>diff_poll_richness_boot</code>, we can look at the yearly differences in beetles species richness in the two geographic regions: (Note that we have only 1 year of data from Sørlandet so far)</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(diff_poll_richness_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/poll_richness_boot-1.png" width="672" /></p>
<p>In the cases where we have used a single row as a reference point, this shows up as a sharp spike at 0.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(diff_poll_richness_boot2)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/diff_beetle_richness_boot2-1.png" width="672" /></p>
<p>More plotting options are available, see the <code>Norimon</code> indicator workflow vignette.</p>
</div>
<div id="map-plots" class="section level4" number="1.8.1.5">
<h4><span class="header-section-number">1.8.1.5</span> Map plots</h4>
<p>In addition to the density plots above, we can also display the values geographically. The <code>map_plot()</code> function takes a <code>boot_stat</code> object and plots the values according to its region names.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map_plot</span>(poll_richness_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/map_plot_cutout-1.png" width="672" /></p>
<p>By default, it only shows the regions with data, but this can be overridden manually:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">map_plot</span>(poll_richness_boot,</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">whole_country =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/map_plot_whole-1.png" width="672" /></p>
<p>We can also choose a different palette, for example from the NinaR package, and visualize the uncertainty by setting the transparency of the colors from the bootstrap standard deviations. By the way, most of these functions can be piped as well:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>diff_poll_richness_boot <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">map_plot</span>(<span class="at">palette =</span> <span class="st">&quot;orange-green&quot;</span>,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">whole_country =</span> <span class="cn">FALSE</span>,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">alpha_from_sd =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/diff_beetle_richness_boot_map-1.png" width="672" /></p>
</div>
<div id="year-rolling-window-summaries" class="section level4" number="1.8.1.6">
<h4><span class="header-section-number">1.8.1.6</span> 5 year rolling window summaries</h4>
<p>As described above, to take the staggered 5 year survey scheme into account, it makes sense to summarize the state over a whole 5 year period. This does not need to adhere to predefined periods, since any 5 year long period will result in a complete data set. We can therefore calculate 5 year long summaries in rolling windows, where each focal year is surrounded by +- 2 years. The bootstrap_value does this by default, if we don’t specify <code>rolling_year_average = FALSE</code></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>poll_richness_boot_5_year <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(poll_loc_year,</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">value =</span> no_species,</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>                                                 <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">rolling_year_window =</span> <span class="cn">TRUE</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>                                      )</span></code></pre></div>
<p>Since we are only on the third year in the survey scheme however, the rolling windows are too large to show any differences between years. This will change, starting with the 2023 year data.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(poll_richness_boot_5_year)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>This concludes the tour of the <code>Norimon</code> functionality.</p>
</div>
</div>
<div id="calculation-principles-for-nbbm-indicators" class="section level3" number="1.8.2">
<h3><span class="header-section-number">1.8.2</span> Calculation principles for NBBM indicators</h3>
<p>These indicators were developed for the Nature Index of Norway, back in 2010 by Ola Diserud and Sandra Öberg (Öberg et. al 2011), and are calculated routinely on a yearly basis (see Åström et. al 2022 for the latest report).</p>
<p>These indicators summarize the state of a bumblebee or butterfly community by estimating the difference from a reference community. Formally, the community indicator (CI) is expressed as the relative change of the community from a state of reference (SR), were the change is calculated by a state of change (SC).</p>
<p><span class="math display">\[CI = \frac{SR-SC}{SR}\]</span>
The state of reference (SR) represents a community of species that can be expected to be observed in a given habitat type and region. It is calculated by assigning each species to class of expected commonality: common (C), sporadic (S) and rare (R) species. This classification is done by expert opinion, informed by known present and past species distributions. Note that this state of reference only contains a subset of highlighted species, making up a historical reference community. Potential observations of “new” species therefore don’t inform the indicator value at all. Common species are expected to be observed in at least 5 % of the surveyed transects in a habitat type and region. Sporadic species similarly are defined as having a presence below 5 % but above 1 %. Rare species are seen in no more than 1 % of the transects, and lastly, species not seen in any transect are assumed as lost (L) for the purpose of this calculation.</p>
<p>Each commonality class gets its own weight, so that common species inform the reference state more than sporadic species, followed by rare species. The state of reference (SR) is thus a value where each species adds to the state according to its commonality. Formally, it is defined as:</p>
<p><span class="math display">\[SR = n_C * w_{C, SR} + n_S * w_{S, SR} + n_R * w_{R, SR} = \sum_{i = (C, S, R)} n_i * w_{i, SR}, \]</span>
where <span class="math inline">\(n_i\)</span> is the number of species in a commonality class (C, S, R), and the weights <span class="math inline">\([w_{C, SR}, w_{S, SR}, w_{R, SR}]\)</span> specify their respective contribution to the reference state (SR). The weights used are <span class="math inline">\([w_{C, SR}, w_{S, SR}, w_{R, SR}] = [1.0, 0.75, 0.50]\)</span>, i.e. a sporadic species has 75 % the weight of a common species, and a rare species has 50 % the weight of a common species.</p>
<p>The state of change (SC) is calculated as
<span class="math display">\[SC = n_{CS} * w_{CS} + n_{CR} * w_{CR} + n_{CL} * w_{CL} + n_{SR} * w_{SR} + n_{SL} * w_{SL} + n_{RL} * w_{RL}, \]</span>
where <span class="math inline">\(n_{CS}\)</span> is the number of expected common species (C) that is observed sporadically (S), and <span class="math inline">\(w_{CS}\)</span> is the weight of this change. Similarly, <span class="math inline">\(n_{CR}\)</span> is the number of expected species that is observed rarely, <span class="math inline">\(n_{RL}\)</span> is the number of rare species that is lost, and so on. Potential increases in observation rates do not inform the indicator. Changes for common species are weighted more heavily than changes for less common species, and larger decreases are weighted more heavily than smaller decreases, such that <span class="math inline">\([w_{CS}, w_{CR}, w_{CL}, w_{SR}, w_{SL}, w_{RL}]= [0.50, 0.75, 1.00, 0.50, 0.75, 0.50]\)</span>.</p>
<p>The indicator values are estimated with uncertainty, by bootstrapping the set of localities within a region, and calculating the indicator for each of 9999 such samples.</p>
<div id="calculation-example" class="section level4" number="1.8.2.1">
<h4><span class="header-section-number">1.8.2.1</span> Calculation example</h4>
<p>These calculations with bootstrap sampling are implemented in the R-package <code>bombLepiSurv</code> (<a href="https://github.com/jenast/bombLepiSurv">Bombus and Lepidoptera Survey</a>). We briefly show this functionality with bumblebees in semi-natural lands as an example.</p>
<p>We connect to the internal database with a convenience function.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>bombLepiSurv<span class="sc">::</span><span class="fu">humlesommerfConnect</span>()</span></code></pre></div>
<p>We can fetch all observations of bumblebees in semi-natural land with the <code>getAllData</code> function. Species names are here shown in Norwegian.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>allBombusGrassland2022 <span class="ot">&lt;-</span> <span class="fu">getAllData</span>(<span class="at">type =</span> <span class="st">&quot;bumblebees&quot;</span>,</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">habitat =</span> <span class="st">&quot;gressmark&quot;</span>,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">language =</span> <span class="st">&quot;norsk&quot;</span>)</span></code></pre></div>
<p>A summary of the records is most easily plotted by <code>surveyBarPlot</code>:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">surveyBarplot</span>(allBombusGrassland2022)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p>Note that for calculating the indicator values, we only consider the transects that were surveyed for the standard 3 times per year, to cover the phenology of the entire season without bias. Data of these complete survey rounds can be fetched from the database by the function <code>getComplData</code>, which fetches data from one habitat type and region at a time.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>bombus_trond_2022 <span class="ot">&lt;-</span> <span class="fu">getComplData</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">region_short =</span> <span class="st">&quot;Trond&quot;</span>,</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">year =</span> <span class="dv">2022</span>)</span></code></pre></div>
<p>The observed relative to the expected occurrences can be visualized through the <code>plotArt</code> function. This requires a reference community, which can be fetched through the <code>getExpValues</code> function.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>exp_bombus_trond <span class="ot">&lt;-</span> <span class="fu">getExpValues</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">region_short =</span> <span class="st">&quot;Trond&quot;</span>,</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>)</span></code></pre></div>
<p>Species in green bars are species that are expected to be common, who need to reach the green areas in the plot not to decrease the indicator value (from 1). Similarly with the blue (sporadic species), and red (rare species). We see in this case that <em>Bombus soroeensis</em> is expected to be a sporadic species, but that it was only observed rarely.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotArt</span>(bombus_trond_2022,</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>        exp_bombus_trond)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<p>The actual calculation and bootstrapping of the indicator is done in the function <code>calcInd</code>. This function calls the <code>getComplData</code> and <code>getExpectedValues</code> functions internally. It also requires the weights for the commonality classes in the reference community (SR), and the weights for the changes in state (CS). This is fetched internally via the functions <code>getAmountWeights</code> and <code>getDiffWeights</code>, respectively. Here the classes are coded in Norwegian (v = common, m = sporadic, s = rare).</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getDiffWeights</span>()</span></code></pre></div>
<pre><code>##      v    m   s
## v 0.00 0.00 0.0
## m 0.50 0.00 0.0
## s 0.75 0.50 0.0
## t 1.00 0.75 0.5</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getAmountWeights</span>()</span></code></pre></div>
<pre><code>##   v    m   s
## 1 1 0.75 0.5</code></pre>
<p>Lastly, we specify the number of samples for the bootstrap. To speed up, only 999 samples are used here.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>nIter <span class="ot">=</span> <span class="dv">999</span></span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>hInd2022TrondGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Trond&quot;</span>,</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<p>The result is an object of class “comm_index” (community index). It comes with some (still rudimentary) print and plotting functions, showing the point estimate and the limits of a 95% confidence interval. Due to the limited number of species and the fixed “weights” of each species and state of change, the indicator calculation returns a distribution of discrete values. Therefore it can happen that the 95% and the 90% confidence intervals are the same, as in this case.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>hInd2022TrondGress</span></code></pre></div>
<pre><code>## $`Community indicator estimates, with percentiles`
##           2.5%             5% Point estimate            95%          97.5% 
##      0.5625000      0.5833333      0.7083333      0.7500000      0.7500000</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(hInd2022TrondGress)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<p>Plotting functions for a series of indicator values is shown below in the indicator calculations.</p>
</div>
</div>
<div id="data-sets" class="section level3" number="1.8.3">
<h3><span class="header-section-number">1.8.3</span> Data sets</h3>
<p>There are a few different ways to access the data required for these indicators. Both the NMBB and the NorIns project store their data in an internal database at NINA. They both also export most of their data to GBIF, but those exports need to be restructured before they can be processed in the following scripts.</p>
<div id="norins-data" class="section level4" number="1.8.3.1">
<h4><span class="header-section-number">1.8.3.1</span> NorIns data</h4>
<p>The <code>Norimon</code> package has convenience functions to fetch data from the database, housed internally at NINA. This database is currently not available outside NINA, but we will implement a solution for this. Either we will make the database externally available, or we will provide an alternative route to the data from the GBIF export. For now, we will fetch the data through the <code>Norimon</code> functions.</p>
</div>
<div id="nmbb-data" class="section level4" number="1.8.3.2">
<h4><span class="header-section-number">1.8.3.2</span> NMBB data</h4>
<p>The data for the bumblebee and butterfly indicators can be accessed through the R-package <code>bombLepiSurv</code>.</p>
</div>
<div id="regions" class="section level4" number="1.8.3.3">
<h4><span class="header-section-number">1.8.3.3</span> Regions</h4>
<!-- In case you need to map the indicator value to regions, you can do that here. Remove this chapter if not relevant. -->
<p>We here show how to import a shape file with the regional delineation. The indicators associated with the NorIns project can be attributed to the 5 country regions of Norway. Currently however, the data program only covers 4 of the 5 regions.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">connect_to_insect_db</span>()</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>norway_regions <span class="ot">&lt;-</span> Norimon<span class="sc">::</span><span class="fu">get_map</span>()</span></code></pre></div>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>norway_reg_NorIns <span class="ot">&lt;-</span> norway_regions <span class="sc">%&gt;%</span> </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region <span class="sc">!=</span> <span class="st">&quot;Vestlandet&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(region) <span class="sc">%&gt;%</span> </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(region) <span class="sc">%&gt;%</span> </span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">geom =</span> <span class="fu">st_union</span>(geom))</span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tm_shape</span>(norway_reg_NorIns) <span class="sc">+</span> </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_polygons</span>(<span class="at">col =</span> <span class="st">&quot;region&quot;</span>) </span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-40-1.png" width="672" /></p>
<p>The indicators from the NMBB program are similarly connected to regions, with the exception that the south-east region only covers the old counties of Vestfold and Østfold, and that Nord-Norge isn’t covered yet.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>nbbm_norway_regions <span class="ot">&lt;-</span> bombLepiSurv<span class="sc">::</span><span class="fu">get_map</span>()</span></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>norway_reg_NBBM <span class="ot">&lt;-</span> nbbm_norway_regions <span class="sc">%&gt;%</span> </span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Øst&quot;</span>, <span class="st">&quot;Sørlandet&quot;</span>, <span class="st">&quot;Trøndelag&quot;</span>, <span class="st">&quot;Vestlandet&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(region) <span class="sc">%&gt;%</span> </span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(region) <span class="sc">%&gt;%</span> </span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">geom =</span> <span class="fu">st_union</span>(geom))</span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tm_shape</span>(norway_reg_NBBM) <span class="sc">+</span> </span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_polygons</span>(<span class="at">col =</span> <span class="st">&quot;region&quot;</span>) </span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
</div>
</div>
<div id="calculation-of-norins-indicators" class="section level3" number="1.8.4">
<h3><span class="header-section-number">1.8.4</span> Calculation of NorIns indicators</h3>
<p>Here we calculate the indicators in abbreviated form, following the general framework outlined above.</p>
<div id="biomass-of-flying-insects-in-semi-natural-land" class="section level4" number="1.8.4.1">
<h4><span class="header-section-number">1.8.4.1</span> Biomass of flying insects in semi-natural land</h4>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>biomass_sn <span class="ot">&lt;-</span> <span class="fu">get_biomass</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">subset_region =</span> <span class="cn">NULL</span>,</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>biomass_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> biomass_sn,</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">value =</span> avg_wet_weight,</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="st">&quot;region_name&quot;</span>)</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>                                   )</span></code></pre></div>
<p>Compare values to a reference state. The reference state is here set uniformly for all regions, with a made-up value just to be able to continue the calculations. Alternatively, we could calculate the indicator values for each region separately, compare to individual reference states, and then put it all together again.</p>
<p>We here use a single value as a reference state and normalize the values at the same time, using the <code>/</code> function.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>biomass_ref <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>biomass_sn_diff <span class="ot">&lt;-</span> biomass_sn_boot <span class="sc">/</span> biomass_ref</span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>biomass_sn_diff</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name avg_wet_weight boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;                &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet            0.470  0.0648         0.345          0.594
## 2  2020 Trøndelag            0.743  0.0700         0.614          0.886
## 3  2020 Østlandet            0.757  0.0496         0.659          0.850
## 4  2021 Sørlandet            0.470  0.0651         0.347          0.606
## 5  2021 Trøndelag            0.746  0.0702         0.607          0.882
## 6  2021 Østlandet            0.755  0.0498         0.658          0.856
## 7  2022 Sørlandet            0.466  0.0664         0.339          0.601
## 8  2022 Trøndelag            0.752  0.0719         0.611          0.883
## 9  2022 Østlandet            0.759  0.0522         0.663          0.868</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(biomass_sn_diff)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-48-1.png" width="672" /></p>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>biomass_sn_to_exp <span class="ot">&lt;-</span> biomass_sn_diff<span class="sc">$</span>bootstrap_summary</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>biomass_sn_to_exp <span class="ot">&lt;-</span> biomass_sn_to_exp <span class="sc">%&gt;%</span> </span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> boot_value,</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> boot_sd,</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> boot_lower2<span class="fl">.5</span>,</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> boot_upper97<span class="fl">.5</span>)</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>biomass_sn_to_exp <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(biomass_sn_to_exp,</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>biomass_sn_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 6 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 7
##   region                                                                                  geom  year ind_value ind_sd ind_lower2.5 ind_upper97.5
##   &lt;chr&gt;                                                                     &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2020     0.470 0.0648        0.345         0.594
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2021     0.470 0.0651        0.347         0.606
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2022     0.466 0.0664        0.339         0.601
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2020     0.743 0.0700        0.614         0.886
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2021     0.746 0.0702        0.607         0.882
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2022     0.752 0.0719        0.611         0.883
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2020     0.757 0.0496        0.659         0.850
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2021     0.755 0.0498        0.658         0.856
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2022     0.759 0.0522        0.663         0.868</code></pre>
</div>
<div id="species-richness-of-flying-insects-in-semi-natural-land" class="section level4" number="1.8.4.2">
<h4><span class="header-section-number">1.8.4.2</span> Species richness of flying insects in semi-natural land</h4>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>richness_sn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_region =</span> <span class="cn">NULL</span>,</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>richness_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> richness_sn,</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">value =</span> no_species,</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="st">&quot;region_name&quot;</span>)</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>                                   )</span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>richness_ref <span class="ot">&lt;-</span> <span class="dv">4000</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>richness_sn_diff <span class="ot">&lt;-</span> richness_sn_boot <span class="sc">/</span> richness_ref</span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>richness_sn_diff</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet        0.545  0.0663         0.424          0.680
## 2  2020 Trøndelag        0.664  0.0270         0.608          0.716
## 3  2020 Østlandet        0.705  0.0363         0.636          0.773
## 4  2021 Sørlandet        0.545  0.0653         0.423          0.674
## 5  2021 Trøndelag        0.662  0.0286         0.606          0.718
## 6  2021 Østlandet        0.703  0.0356         0.639          0.774
## 7  2022 Sørlandet        0.548  0.0658         0.423          0.681
## 8  2022 Trøndelag        0.665  0.0282         0.609          0.720
## 9  2022 Østlandet        0.705  0.0355         0.637          0.780</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(richness_sn_diff)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-54-1.png" width="672" /></p>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>richness_sn_to_exp <span class="ot">&lt;-</span> richness_sn_diff<span class="sc">$</span>bootstrap_summary</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>richness_sn_to_exp <span class="ot">&lt;-</span> richness_sn_to_exp <span class="sc">%&gt;%</span> </span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> boot_value,</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> boot_sd,</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> boot_lower2<span class="fl">.5</span>,</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> boot_upper97<span class="fl">.5</span>)</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>richness_sn_to_exp <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(richness_sn_to_exp,</span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a>richness_sn_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 6 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 7
##   region                                                                                  geom  year ind_value ind_sd ind_lower2.5 ind_upper97.5
##   &lt;chr&gt;                                                                     &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2020     0.545 0.0663        0.424         0.680
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2021     0.545 0.0653        0.423         0.674
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2022     0.548 0.0658        0.423         0.681
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2020     0.664 0.0270        0.608         0.716
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2021     0.662 0.0286        0.606         0.718
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2022     0.665 0.0282        0.609         0.720
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2020     0.705 0.0363        0.636         0.773
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2021     0.703 0.0356        0.639         0.774
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2022     0.705 0.0355        0.637         0.780</code></pre>
</div>
<div id="species-richness-of-pollinators-in-semi-natural-land" class="section level4" number="1.8.4.3">
<h4><span class="header-section-number">1.8.4.3</span> Species richness of pollinators in semi-natural land</h4>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>pollinator_fam <span class="ot">&lt;-</span> <span class="fu">get_pollinators</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(family_latin) <span class="sc">%&gt;%</span> </span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>pollinators_sn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">subset_families =</span> pollinator_fam,</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>pollinators_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> pollinators_sn,</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">value =</span> no_species,</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>                                                  <span class="st">&quot;region_name&quot;</span>)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>                                       )</span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>pollinators_ref <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>pollinators_sn_diff <span class="ot">&lt;-</span> pollinators_sn_boot <span class="sc">/</span> pollinators_ref</span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>pollinators_sn_diff</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet        0.602  0.117          0.372          0.83 
## 2  2020 Trøndelag        0.765  0.0448         0.678          0.852
## 3  2020 Østlandet        0.971  0.0406         0.861          1    
## 4  2021 Sørlandet        0.595  0.116          0.36           0.828
## 5  2021 Trøndelag        0.766  0.0427         0.68           0.845
## 6  2021 Østlandet        0.973  0.0404         0.862          1    
## 7  2022 Sørlandet        0.595  0.108          0.384          0.802
## 8  2022 Trøndelag        0.765  0.0429         0.67           0.842
## 9  2022 Østlandet        0.969  0.0426         0.859          1</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pollinators_sn_diff)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>pollinators_sn_to_exp <span class="ot">&lt;-</span> pollinators_sn_diff<span class="sc">$</span>bootstrap_summary</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>pollinators_sn_to_exp <span class="ot">&lt;-</span> pollinators_sn_to_exp <span class="sc">%&gt;%</span> </span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> boot_value,</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> boot_sd,</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> boot_lower2<span class="fl">.5</span>,</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> boot_upper97<span class="fl">.5</span>)</span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>pollinators_sn_to_exp <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(pollinators_sn_to_exp,</span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a>pollinators_sn_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 6 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 7
##   region                                                                                  geom  year ind_value ind_sd ind_lower2.5 ind_upper97.5
##   &lt;chr&gt;                                                                     &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2020     0.602 0.117         0.372         0.83 
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2021     0.595 0.116         0.36          0.828
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2022     0.595 0.108         0.384         0.802
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2020     0.765 0.0448        0.678         0.852
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2021     0.766 0.0427        0.68          0.845
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2022     0.765 0.0429        0.67          0.842
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2020     0.971 0.0406        0.861         1    
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2021     0.973 0.0404        0.862         1    
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2022     0.969 0.0426        0.859         1</code></pre>
</div>
<div id="species-richness-of-dung-associated-insects" class="section level4" number="1.8.4.4">
<h4><span class="header-section-number">1.8.4.4</span> Species richness of dung associated insects</h4>
<p>Note that the species list will probably be expanded, after a taxonomic review.</p>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>dung_fam <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Muscidae&quot;</span>) </span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>dung_sn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">subset_families =</span> dung_fam,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<p>[This will be amended with data based on genus and species lists. Need to use the lowest taxonomic resolution. Will implement a species lookup list in the database]</p>
<p>Temporary…put these in the DB. Also find all species of Muscidae and append them.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>coleoptera_dung_species <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Trox sabulosus&quot;</span>,</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Trox scaber&quot;</span>,</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Geotrupes spiniger&quot;</span>,</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Geotrupes stercorarius&quot;</span>,</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Trypocopris vernalis&quot;</span>,</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Anoplotrupes stercorosus&quot;</span>,</span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aegialia arenaria&quot;</span>,</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aegialia rufa&quot;</span>,</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aegialia sabuleti&quot;</span>,</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius ater&quot;</span>,</span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius borealis&quot;</span>,</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius brevis&quot;</span>,</span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius coenosus&quot;</span>,</span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius conspurcatus&quot;</span>,</span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius contaminatus&quot;</span>,</span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius depressus&quot;</span>,</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius distinctus&quot;</span>,</span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius erraticus&quot;</span>,</span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius fasciatus&quot;</span>,</span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius fimetarius&quot;</span>,</span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius foetens&quot;</span>,</span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius fossor&quot;</span>,</span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius granarius&quot;</span>,</span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius haemorrhoidalis&quot;</span>,</span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius ictericus&quot;</span>,</span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius lapponum&quot;</span>,</span>
<span id="cb78-27"><a href="#cb78-27" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius luridus&quot;</span>,</span>
<span id="cb78-28"><a href="#cb78-28" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius merdarius&quot;</span>,</span>
<span id="cb78-29"><a href="#cb78-29" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius nemoralis &quot;</span>,</span>
<span id="cb78-30"><a href="#cb78-30" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius niger&quot;</span>,</span>
<span id="cb78-31"><a href="#cb78-31" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius paykulli&quot;</span>,</span>
<span id="cb78-32"><a href="#cb78-32" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius pedellus&quot;</span>,</span>
<span id="cb78-33"><a href="#cb78-33" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius piceus&quot;</span>,</span>
<span id="cb78-34"><a href="#cb78-34" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius plagiatus&quot;</span>,</span>
<span id="cb78-35"><a href="#cb78-35" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius porcus&quot;</span>,</span>
<span id="cb78-36"><a href="#cb78-36" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius prodromus&quot;</span>,</span>
<span id="cb78-37"><a href="#cb78-37" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius punctatosulcatus&quot;</span>,</span>
<span id="cb78-38"><a href="#cb78-38" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius pusillus&quot;</span>,</span>
<span id="cb78-39"><a href="#cb78-39" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius rufipes&quot;</span>,</span>
<span id="cb78-40"><a href="#cb78-40" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius rufus&quot;</span>,</span>
<span id="cb78-41"><a href="#cb78-41" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius sordidus&quot;</span>,</span>
<span id="cb78-42"><a href="#cb78-42" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius sphacelatus&quot;</span>,</span>
<span id="cb78-43"><a href="#cb78-43" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius sticticus&quot;</span>,</span>
<span id="cb78-44"><a href="#cb78-44" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Aphodius subterraneus&quot;</span>,</span>
<span id="cb78-45"><a href="#cb78-45" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Euheptaulacus villosus&quot;</span>,</span>
<span id="cb78-46"><a href="#cb78-46" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Psammodius asper&quot;</span>,</span>
<span id="cb78-47"><a href="#cb78-47" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Onthophagusjoannae&quot;</span>,</span>
<span id="cb78-48"><a href="#cb78-48" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Onthophagus nuchicomis&quot;</span>,</span>
<span id="cb78-49"><a href="#cb78-49" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Onthophagus fracticornis&quot;</span>,</span>
<span id="cb78-50"><a href="#cb78-50" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Saprinus rugifer&quot;</span>,</span>
<span id="cb78-51"><a href="#cb78-51" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Saprinus planiusculus&quot;</span>,</span>
<span id="cb78-52"><a href="#cb78-52" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Saprinus semistriatus &quot;</span>,</span>
<span id="cb78-53"><a href="#cb78-53" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Saprinus aeneus&quot;</span>,</span>
<span id="cb78-54"><a href="#cb78-54" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hypocaccus rugiceps&quot;</span>,</span>
<span id="cb78-55"><a href="#cb78-55" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hypocaccus metallicus&quot;</span>,</span>
<span id="cb78-56"><a href="#cb78-56" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hypocaccus rugifrons&quot;</span>,</span>
<span id="cb78-57"><a href="#cb78-57" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus brunneus&quot;</span>,</span>
<span id="cb78-58"><a href="#cb78-58" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus striola&quot;</span>,</span>
<span id="cb78-59"><a href="#cb78-59" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus merdarius&quot;</span>,</span>
<span id="cb78-60"><a href="#cb78-60" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus obscurus&quot;</span>,</span>
<span id="cb78-61"><a href="#cb78-61" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus ventralis&quot;</span>,</span>
<span id="cb78-62"><a href="#cb78-62" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus neglectus&quot;</span>,</span>
<span id="cb78-63"><a href="#cb78-63" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus purpurascens&quot;</span>,</span>
<span id="cb78-64"><a href="#cb78-64" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Margarinotus carbonarius&quot;</span>,</span>
<span id="cb78-65"><a href="#cb78-65" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hister unicolor&quot;</span>,</span>
<span id="cb78-66"><a href="#cb78-66" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Hister bissexstriatus&quot;</span>,</span>
<span id="cb78-67"><a href="#cb78-67" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Histerfunestus&quot;</span>,</span>
<span id="cb78-68"><a href="#cb78-68" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Atholus bimaculatus&quot;</span>,</span>
<span id="cb78-69"><a href="#cb78-69" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Atholus duodecimstriatus&quot;</span>,</span>
<span id="cb78-70"><a href="#cb78-70" aria-hidden="true" tabindex="-1"></a><span class="st">&quot;Atholus corvinus&quot;</span>)</span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>dung_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> dung_sn,</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">value =</span> no_species,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>                                                  <span class="st">&quot;region_name&quot;</span>)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>                                       )</span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>dung_ref <span class="ot">&lt;-</span> <span class="dv">90</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>dung_sn_diff <span class="ot">&lt;-</span> dung_sn_boot <span class="sc">/</span> dung_ref</span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>dung_sn_diff</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name no_species boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet        0.677  0.0676         0.553          0.808
## 2  2020 Trøndelag        0.926  0.0360         0.853          0.989
## 3  2020 Østlandet        0.846  0.0364         0.777          0.916
## 4  2021 Sørlandet        0.676  0.0713         0.556          0.83 
## 5  2021 Trøndelag        0.925  0.0370         0.845          0.997
## 6  2021 Østlandet        0.847  0.0370         0.772          0.921
## 7  2022 Sørlandet        0.667  0.0671         0.539          0.806
## 8  2022 Trøndelag        0.925  0.0347         0.85           0.988
## 9  2022 Østlandet        0.846  0.0367         0.776          0.922</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dung_sn_diff)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-68-1.png" width="672" /></p>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>dung_sn_to_exp <span class="ot">&lt;-</span> dung_sn_diff<span class="sc">$</span>bootstrap_summary</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>dung_sn_to_exp <span class="ot">&lt;-</span> dung_sn_to_exp <span class="sc">%&gt;%</span> </span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> boot_value,</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> boot_sd,</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> boot_lower2<span class="fl">.5</span>,</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> boot_upper97<span class="fl">.5</span>)</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>dung_sn_to_exp <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(dung_sn_to_exp,</span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>dung_sn_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 6 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 7
##   region                                                                                  geom  year ind_value ind_sd ind_lower2.5 ind_upper97.5
##   &lt;chr&gt;                                                                     &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2020     0.677 0.0676        0.553         0.808
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2021     0.676 0.0713        0.556         0.83 
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2022     0.667 0.0671        0.539         0.806
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2020     0.926 0.0360        0.853         0.989
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2021     0.925 0.0370        0.845         0.997
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2022     0.925 0.0347        0.85          0.988
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2020     0.846 0.0364        0.777         0.916
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2021     0.847 0.0370        0.772         0.921
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2022     0.846 0.0367        0.776         0.922</code></pre>
</div>
<div id="relationship-between-symphyta-and-parasitica-in-semi-natural-land" class="section level4" number="1.8.4.5">
<h4><span class="header-section-number">1.8.4.5</span> Relationship between Symphyta and Parasitica in semi-natural land</h4>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>symphyta_fam <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Argidae&quot;</span>, <span class="st">&quot;Cephidae&quot;</span>, <span class="st">&quot;Cimbicidae&quot;</span>, </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;Diprionidae&quot;</span>, <span class="st">&quot;Orussidae&quot;</span>, <span class="st">&quot;Pamphiliidae&quot;</span>, </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;Pergidae&quot;</span>, <span class="st">&quot;Siricidae&quot;</span>, <span class="st">&quot;Anaxyelidae&quot;</span>,</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>                  <span class="st">&quot;Tenthredinidae&quot;</span>, <span class="st">&quot;Xiphydriidae&quot;</span>, <span class="st">&quot;Xyelidae&quot;</span>) </span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>symphyta_sn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_families =</span> symphyta_fam,</span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>symphyta_sn</span></code></pre></div>
<pre><code>## # A tibble: 60 × 8
##     year locality    habitat_type region_name no_species shannon_div mean_no_asv_per_species GDE_by_asv
##    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;        &lt;chr&gt;            &lt;int&gt;       &lt;dbl&gt;                   &lt;dbl&gt;      &lt;dbl&gt;
##  1  2020 Semi-nat_01 Semi-nat     Østlandet           38        31.9                    1.66       0.84
##  2  2020 Semi-nat_02 Semi-nat     Østlandet           16        15.7                    1.06       0.98
##  3  2020 Semi-nat_03 Semi-nat     Østlandet           13        12.3                    1.23       0.95
##  4  2020 Semi-nat_04 Semi-nat     Østlandet           39        34.6                    1.38       0.89
##  5  2020 Semi-nat_05 Semi-nat     Østlandet           65        58.5                    1.58       0.9 
##  6  2020 Semi-nat_06 Semi-nat     Østlandet           22        20.0                    1.14       0.91
##  7  2020 Semi-nat_07 Semi-nat     Østlandet           36        33.2                    1.31       0.92
##  8  2020 Semi-nat_08 Semi-nat     Østlandet           46        40.4                    1.48       0.88
##  9  2020 Semi-nat_09 Semi-nat     Østlandet           21        18.2                    1.62       0.86
## 10  2020 Semi-nat_10 Semi-nat     Østlandet           68        56.9                    1.62       0.84
## # ℹ 50 more rows</code></pre>
<p>[Note, this list needs to be checked over by e.g. Arnstein.]</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>parasitica_fam <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Braconidae&quot;</span>, <span class="st">&quot;Ichneumonidae&quot;</span>, <span class="st">&quot;Chalcididae&quot;</span>, </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Eulophidae&quot;</span>, <span class="st">&quot;Pteromalidae&quot;</span>, <span class="st">&quot;Aphelinidae&quot;</span>, </span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Scelionidae&quot;</span>, <span class="st">&quot;Eupelmidae&quot;</span>, <span class="st">&quot;Encyrtidae&quot;</span>, </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Mymaridae&quot;</span>, <span class="st">&quot;Diapriidae&quot;</span>, <span class="st">&quot;Bethylidae&quot;</span>, </span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Evaniidae&quot;</span>, <span class="st">&quot;Ceraphronidae&quot;</span>, <span class="st">&quot;Torymidae&quot;</span>, </span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Dryinidae&quot;</span>, <span class="st">&quot;Eucharitidae&quot;</span>, <span class="st">&quot;Mymarommatidae&quot;</span>, </span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Orussidae&quot;</span>, <span class="st">&quot;Megaspilidae&quot;</span>, <span class="st">&quot;Stephanidae&quot;</span>, </span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Trigonalidae&quot;</span>, <span class="st">&quot;Platygastridae&quot;</span>, <span class="st">&quot;Aulacidae&quot;</span>, </span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Gasteruptiidae&quot;</span>, <span class="st">&quot;Rhopalosomatidae&quot;</span>, <span class="st">&quot;Larridae&quot;</span>, </span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;Agaonidae&quot;</span>, <span class="st">&quot;Pompilidae&quot;</span>, <span class="st">&quot;Bradynobaenidae&quot;</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>                    )</span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>parasitica_sn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_families =</span> parasitica_fam,</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>parasitica_sn</span></code></pre></div>
<pre><code>## # A tibble: 60 × 8
##     year locality    habitat_type region_name no_species shannon_div mean_no_asv_per_species GDE_by_asv
##    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;        &lt;chr&gt;            &lt;int&gt;       &lt;dbl&gt;                   &lt;dbl&gt;      &lt;dbl&gt;
##  1  2020 Semi-nat_01 Semi-nat     Østlandet          480        413.                    1.43       0.86
##  2  2020 Semi-nat_02 Semi-nat     Østlandet          112        107.                    1.12       0.95
##  3  2020 Semi-nat_03 Semi-nat     Østlandet          119        113.                    1.14       0.95
##  4  2020 Semi-nat_04 Semi-nat     Østlandet          369        316.                    1.47       0.86
##  5  2020 Semi-nat_05 Semi-nat     Østlandet          642        526.                    1.64       0.82
##  6  2020 Semi-nat_06 Semi-nat     Østlandet          190        177.                    1.19       0.93
##  7  2020 Semi-nat_07 Semi-nat     Østlandet          253        233.                    1.23       0.92
##  8  2020 Semi-nat_08 Semi-nat     Østlandet          323        292.                    1.27       0.9 
##  9  2020 Semi-nat_09 Semi-nat     Østlandet          216        187.                    1.38       0.86
## 10  2020 Semi-nat_10 Semi-nat     Østlandet          531        463.                    1.41       0.87
## # ℹ 50 more rows</code></pre>
<p>Divide the richness of Parasitica by Symphyta. [Could make a function for this later on.]</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>par_sym <span class="ot">&lt;-</span> symphyta_sn <span class="sc">%&gt;%</span> </span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(parasitica_sn,</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span> <span class="ot">=</span><span class="st">&quot;year&quot;</span>,</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;locality&quot;</span> <span class="ot">=</span> <span class="st">&quot;locality&quot;</span>,</span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;habitat_type&quot;</span> <span class="ot">=</span> <span class="st">&quot;habitat_type&quot;</span>,</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;region_name&quot;</span> <span class="ot">=</span> <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">suffix =</span> <span class="fu">c</span>(<span class="st">&quot;_sym&quot;</span>, <span class="st">&quot;_par&quot;</span>)</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>       ) <span class="sc">%&gt;%</span> </span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">par_per_sym_richn =</span> no_species_par <span class="sc">/</span> no_species_sym)</span></code></pre></div>
<p>Bootstrap the fraction of Parasitica to Symphyta.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>par_sym_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(par_sym,</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">value =</span> par_per_sym_richn,</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>, </span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>                                           <span class="st">&quot;region_name&quot;</span>))</span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>par_sym_ref <span class="ot">&lt;-</span> <span class="dv">40</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>par_sym_sn_diff <span class="ot">&lt;-</span> par_sym_sn_boot <span class="sc">/</span> par_sym_ref</span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>par_sym_sn_diff</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name par_per_sym_richn boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;                   &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet               0.826  0.164          0.537          1    
## 2  2020 Trøndelag               0.732  0.0743         0.593          0.880
## 3  2020 Østlandet               0.447  0.0420         0.366          0.533
## 4  2021 Sørlandet               0.823  0.167          0.526          1    
## 5  2021 Trøndelag               0.733  0.0788         0.594          0.903
## 6  2021 Østlandet               0.448  0.0427         0.365          0.533
## 7  2022 Sørlandet               0.818  0.168          0.530          1    
## 8  2022 Trøndelag               0.740  0.0790         0.595          0.908
## 9  2022 Østlandet               0.447  0.0412         0.364          0.527</code></pre>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(par_sym_sn_diff)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-76-1.png" width="672" /></p>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>par_sym_sn_exp <span class="ot">&lt;-</span> par_sym_sn_diff<span class="sc">$</span>bootstrap_summary</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>par_sym_sn_exp <span class="ot">&lt;-</span> par_sym_sn_exp <span class="sc">%&gt;%</span> </span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> boot_value,</span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> boot_sd,</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> boot_lower2<span class="fl">.5</span>,</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> boot_upper97<span class="fl">.5</span>)</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>par_sym_sn_exp <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(par_sym_sn_exp,</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>par_sym_sn_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 6 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 7
##   region                                                                                  geom  year ind_value ind_sd ind_lower2.5 ind_upper97.5
##   &lt;chr&gt;                                                                     &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2020     0.826 0.164         0.537         1    
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2021     0.823 0.167         0.526         1    
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2022     0.818 0.168         0.530         1    
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2020     0.732 0.0743        0.593         0.880
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2021     0.733 0.0788        0.594         0.903
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2022     0.740 0.0790        0.595         0.908
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2020     0.447 0.0420        0.366         0.533
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2021     0.448 0.0427        0.365         0.533
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2022     0.447 0.0412        0.364         0.527</code></pre>
</div>
<div id="number-of-alien-species-in-semi-natural-land" class="section level4" number="1.8.4.6">
<h4><span class="header-section-number">1.8.4.6</span> Number of alien species in semi-natural land</h4>
<p>Here we measure the prevalence of alien species in the trap catches. We present two options for calculating the indicator, as it is not clear at this point which of them is superior.</p>
<p>The simplest approach is to just sum up the number of alien species present in a region, i.e. the species richness of alien species. This can be related to the total species richness of the community, and expressed as the proportion of alien species in the community. The reference state of alien species is zero by definition, since we don’t want any alien species. The value for good ecological condition could be set at a low (but arbitrary) proportion of alien species in the community. This is done for alien plant species where the value for good ecological condition is set to 0.05 (&lt; 5 % of the plant community should be alien). For a hyper-diverse community such as insects, this is probably too high a value. In a community of say 20 000 insect species, 5 % would mean 1 000 alien species, which is hard to see as a good condition. It is not clear to us at present what level to set here, but we exemplify the calculations with a good condition set to 0.1 % (&lt; 20 species in a 20 000 species community). It is also likely that the species richness of the entire community will increase, perhaps substantially, with increased sampling. It is not clear if the number of detected native and alien species will increase at the same rate, so it may be that the proportion of alien species will change, perhaps substantially, with increased sampling effort. It may be difficult therefore to interpret this frequency of alien insect species.</p>
<p>In addition, this approach will not take into account the spread of the species, a measure that is especially relevant for rare but increasing species. A species found only once will count as much as a species found in every locality. To take into account the prevalence of the species, we can observe the proportion of localities each alien species is observed in. These proportions could then be summed together for at total score of alien species. Here, a species found in all localities will count much more, than a species only found once.</p>
<p>For each alien species, the observation frequency is calculated as the number of localities the species is observed in, divided by the total amount of localities in a geographical region and time period. <span class="math display">\[p_i = \frac{1}{N} \sum_{i=1, j=1}^{N}{obs_{ij}}\]</span>, where <span class="math inline">\(obs_ij\)</span> is an observation of species <span class="math inline">\(i\)</span> in locality <span class="math inline">\(j\)</span>, and <span class="math inline">\(N\)</span> is the total number of possible observations (localities in the region and time period).
These proportions are combined into a common score for all alien species in <span class="math display">\[alien~score =\sum_{i=1}^{S}{~p_{i}}\]</span>, where <span class="math inline">\(p_i\)</span> is the observation frequency of species <span class="math inline">\(i\)</span>, and <span class="math inline">\(S\)</span> is the set of all observed alien species. The uncertainty of both types of indicators is calculated by bootstrapping the set of localities within a temporal and spatial scope, similarly to the other indicators. If all alien species are present in all localities, the indicator value simply becomes <span class="math inline">\(S\)</span>. If the species is not present in all localities, it’s influence on the indicator is weighted down linearly with it presence frequency.</p>
<p><strong>Fetch the species list.</strong></p>
<p>The alien species observations are stored in a materialized view for convenience. This is simply a list of observations of the alien species of categories LO, PH, HI, or SE, together with the year localities and locality samplings they were observed in. For the purposes of this calculation, we will only consider the year localities (and not the prevalence og alien species within locality and season). So the first thing to do is to fetch this list, and aggregate it to the year locality level. This summary is based on the 2023 of the alien species evaluation, and we only consider species that are identified with identification confidence “Moderate” or “High” (not Low or Poor). In addition, we filter out the forest sites.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>alien_obs <span class="ot">&lt;-</span> <span class="fu">tbl</span>(con,</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">Id</span>(<span class="at">schema =</span> <span class="st">&quot;views&quot;</span>,</span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">table =</span> <span class="st">&quot;alien_obs&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(identification_confidence <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;MODERATE&quot;</span>, <span class="st">&quot;HIGH&quot;</span>),</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a>         <span class="fu">grepl</span>(<span class="st">&quot;Semi-nat&quot;</span>, locality))</span></code></pre></div>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>alien_obs_ls <span class="ot">&lt;-</span> alien_obs <span class="sc">%&gt;%</span> </span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">as.integer</span>(<span class="fu">n_distinct</span>(ls_id))) <span class="sc">%&gt;%</span> </span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>alien_obs_yl <span class="ot">&lt;-</span> alien_obs <span class="sc">%&gt;%</span> </span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">as.integer</span>(<span class="fu">n_distinct</span>(yl_id))) <span class="sc">%&gt;%</span> </span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>no_alien_spec <span class="ot">&lt;-</span> alien_obs <span class="sc">%&gt;%</span> </span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">as.integer</span>(<span class="fu">n_distinct</span>(species_latin_fixed))) <span class="sc">%&gt;%</span> </span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>no_alien_cat <span class="ot">&lt;-</span> alien_obs <span class="sc">%&gt;%</span> </span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">as.integer</span>(<span class="fu">n_distinct</span>(risikokategori_2023))) <span class="sc">%&gt;%</span> </span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>()</span></code></pre></div>
<p>We can have quick look at some statistics of the data. We see that the project thus far has identified 15, representing 3 risk-categories, in (all) 60 semi-natural localities.</p>
<p>To continue, we aggregate the presences of alien species to locality years, meaning in recording in how many localities each alien species were detected at least once.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>alien_obs_yl <span class="ot">&lt;-</span> alien_obs <span class="sc">%&gt;%</span> </span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(yl_id,</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>           locality,</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>           year,</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>           region_name,</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>           risikokategori_2023,</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>           species_latin_fixed) <span class="sc">%&gt;%</span> </span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">present =</span> <span class="fu">sum</span>(present) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(locality, species_latin_fixed) <span class="sc">%&gt;%</span> </span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">collect</span>(<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &quot;yl_id&quot;, &quot;locality&quot;, &quot;year&quot;, &quot;region_name&quot;, and &quot;risikokategori_2023&quot;. You can override using the `.groups`
## argument.</code></pre>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>alien_obs_yl</span></code></pre></div>
<pre><code>## # A tibble: 900 × 7
## # Groups:   yl_id, locality, year, region_name, risikokategori_2023 [180]
##    yl_id                                locality     year region_name risikokategori_2023 species_latin_fixed       present
##    &lt;chr&gt;                                &lt;chr&gt;       &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;               &lt;chr&gt;                     &lt;lgl&gt;  
##  1 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Acrotona pseudotenera     FALSE  
##  2 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   PH                  Acrotrichis insularis     FALSE  
##  3 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Cartodere constricta      FALSE  
##  4 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Dacnusa sibirica          FALSE  
##  5 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   PH                  Deraeocoris lutescens     FALSE  
##  6 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Eurhadinoceraea ventralis FALSE  
##  7 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Haematobosca stimulans    FALSE  
##  8 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Omalium rugatum           FALSE  
##  9 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   LO                  Omonadus floralis         FALSE  
## 10 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   SE                  Opilio canestrinii        FALSE  
## # ℹ 890 more rows</code></pre>
<p>Next step is to bootstrap the alien observations to get the uncertainty measurement. This has to be done in a slightly different way than for the aggregated diversity or biomass values that we have used for the other indicators. Here we have a set of alien species presences for a locality, and when we bootstrap the localities, all these presences should go together. So we first have to take bootstrap samples of sets of localities (with replacement). Then we calculate the presence (or presence frequency) of each alien species <span class="math inline">\(p_i\)</span>, before we can summarize all of these into the indicator value <span class="math inline">\(H_alien\)</span>. This is implemented in a separate function <code>bootstrap_alien_obs</code> in the <code>Norimon</code> package, were we can optionally either sum the species numbers, or sum the species presence frequencies. Note that for all alien species recorded in at least one locality, we have recorded the presences and absences in all localities. This is needed for his function to work. This function is more computer intensive than the bootstrap of already aggregated values, so we employ parallelization through the <code>parallel</code> package internally.</p>
<p><strong>We first calculate the indicator based on the the number of alien species</strong></p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>alien_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_alien_obs</span>(alien_obs_yl,</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">rolling_year_window =</span> <span class="cn">TRUE</span>,</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">only_no_spec =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(alien_sn_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<p>The jagged lines in the density plot comes from the discrete values of number of species. Østlandet has the most alien species, followed by Sørlandet, and lastly Trøndelag.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ts_plot</span>(alien_sn_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-84-1.png" width="672" /></p>
<p>As explained above, these values could be expressed as a proportion of the the total species richness. Since the total species richness is different for the different regions, we (currently) have to do this separately for each region.</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>alien_obs_yl_sor <span class="ot">&lt;-</span> alien_obs_yl <span class="sc">%&gt;%</span> </span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region_name <span class="sc">==</span> <span class="st">&quot;Sørlandet&quot;</span>)</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>alien_obs_yl_ost <span class="ot">&lt;-</span> alien_obs_yl <span class="sc">%&gt;%</span> </span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region_name <span class="sc">==</span> <span class="st">&quot;Østlandet&quot;</span>)</span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>alien_obs_yl_trond <span class="ot">&lt;-</span> alien_obs_yl <span class="sc">%&gt;%</span> </span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region_name <span class="sc">==</span> <span class="st">&quot;Trøndelag&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>alien_sn_sor_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_alien_obs</span>(alien_obs_yl_sor,</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">rolling_year_window =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">only_no_spec =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>alien_sn_ost_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_alien_obs</span>(alien_obs_yl_ost,</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">rolling_year_window =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">only_no_spec =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a>alien_sn_trond_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_alien_obs</span>(alien_obs_yl_trond,</span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb108-14"><a href="#cb108-14" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">rolling_year_window =</span> <span class="cn">TRUE</span>,</span>
<span id="cb108-15"><a href="#cb108-15" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">only_no_spec =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Next, we fetch a record of the species richness found so far, in each region.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>region_richn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">agg_level =</span> <span class="st">&quot;region_habitat&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>region_richn</span></code></pre></div>
<pre><code>## # A tibble: 4 × 6
##   habitat_type region_name no_species shannon_div mean_no_asv_per_species GDE_by_asv
##   &lt;chr&gt;        &lt;chr&gt;            &lt;int&gt;       &lt;dbl&gt;                   &lt;dbl&gt;      &lt;dbl&gt;
## 1 Forest       Østlandet        13726       4924.                    4.87      0.359
## 2 Semi-nat     Sørlandet         6679       3204.                    2.6       0.480
## 3 Semi-nat     Trøndelag        10063       3359.                    4.26      0.334
## 4 Semi-nat     Østlandet        15515       5716.                    4.91      0.368</code></pre>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>region_richn_sor <span class="ot">&lt;-</span> region_richn <span class="sc">%&gt;%</span> </span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region_name <span class="sc">==</span> <span class="st">&quot;Sørlandet&quot;</span>,</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>         habitat_type <span class="sc">==</span> <span class="st">&quot;Semi-nat&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(no_species) <span class="sc">%&gt;%</span> </span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>()</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>region_richn_ost <span class="ot">&lt;-</span> region_richn <span class="sc">%&gt;%</span> </span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region_name <span class="sc">==</span> <span class="st">&quot;Østlandet&quot;</span>,</span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>         habitat_type <span class="sc">==</span> <span class="st">&quot;Semi-nat&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(no_species) <span class="sc">%&gt;%</span> </span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>()</span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>region_richn_trond <span class="ot">&lt;-</span> region_richn <span class="sc">%&gt;%</span> </span>
<span id="cb112-16"><a href="#cb112-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region_name <span class="sc">==</span> <span class="st">&quot;Trøndelag&quot;</span>,</span>
<span id="cb112-17"><a href="#cb112-17" aria-hidden="true" tabindex="-1"></a>         habitat_type <span class="sc">==</span> <span class="st">&quot;Semi-nat&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb112-18"><a href="#cb112-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(no_species) <span class="sc">%&gt;%</span> </span>
<span id="cb112-19"><a href="#cb112-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb112-20"><a href="#cb112-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.numeric</span>()</span></code></pre></div>
<p>Next, we scale each indicator to the total number of observed species.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>alien_sn_sor_prop <span class="ot">&lt;-</span> alien_sn_sor_boot <span class="sc">/</span> region_richn_sor</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>alien_sn_ost_prop <span class="ot">&lt;-</span> alien_sn_ost_boot <span class="sc">/</span> region_richn_ost</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>alien_sn_trond_prop <span class="ot">&lt;-</span> alien_sn_trond_boot <span class="sc">/</span> region_richn_trond</span></code></pre></div>
<p>We could scale these values with a reference value of 0, and a breaking point of 0.001. eaTools has a function for that, but for this to work, we currently can only operate on the aggregate values of the indicator (not the bootstrap samples).</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>alien_sn_prop <span class="ot">&lt;-</span> alien_sn_sor_prop[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(alien_sn_ost_prop[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(alien_sn_trond_prop[[<span class="dv">1</span>]]) <span class="sc">%&gt;%</span> </span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">alien_score =</span> boot_value)</span></code></pre></div>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>alien_sn_prop_sf <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(alien_sn_prop,</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>alien_sn_prop_sf<span class="sc">$</span>alien_score_scaled <span class="ot">&lt;-</span> eaTools<span class="sc">::</span><span class="fu">ea_normalise</span>(alien_sn_prop_sf, </span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">vector =</span> <span class="st">&quot;alien_score&quot;</span>, </span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">upper_reference_level =</span> <span class="fl">0.01</span>,</span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">lower_reference_level =</span> <span class="dv">0</span>,</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">break_point =</span> <span class="fl">0.001</span>, </span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">reverse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">plot =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>This calculation is then an the sf-object can be exported.</p>
<p><strong>Secondly, we calculate the indicator based on the distribution of the alien species</strong></p>
<p>We call this alien_H, since the calculation resembles Hill numbers from calculations of Shannon information.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>alien_H_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_alien_obs</span>(alien_obs_yl,</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">rolling_year_window =</span> <span class="cn">TRUE</span>,</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">only_no_spec =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>alien_H_boot</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name alien_score boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;             &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet         0.9     0.242         0.5             1.4 
## 2  2020 Trøndelag         0.699   0.161         0.4             1   
## 3  2020 Østlandet         0.941   0.161         0.633           1.27
## 4  2021 Sørlandet         0.910   0.244         0.5             1.4 
## 5  2021 Trøndelag         0.696   0.163         0.4             1.05
## 6  2021 Østlandet         0.942   0.152         0.633           1.23
## 7  2022 Sørlandet         0.908   0.256         0.4             1.4 
## 8  2022 Trøndelag         0.700   0.159         0.4             1.05
## 9  2022 Østlandet         0.926   0.161         0.633           1.27</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(alien_H_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-96-1.png" width="672" />
Interstingly, by this calculation method, the different regions have much more similar values. Specifically, the southern region has comparable values to the eastern region, while Trøndelag has lower values. This indicates that region Østlandet has markedly higher number of alien species than the Southern region, but with generally more limited distribution. While in comparison, the alien species in the south are more frequently distributed throughout the sampling localities.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ts_plot</span>(alien_H_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-97-1.png" width="672" /></p>
<p>This indicator also has a natural reference value, with zero presences of alien species, meaning an alien score of 0. We could scale this indicator similarly as with the indicator based on the pure number of alien species. Similar to that case, there is no upper limit to how many alien species there can be, but we set a high enough value, and use a breaking point for the scaling, this upper end point has limited influence.</p>
<p>We don’t have enough experience to determine an appropriate breakpoint for the scaling at this point, or an upper value, but exemplify the calculation with some subjective values</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>alien_H_boot_sf <span class="ot">&lt;-</span> alien_H_boot[[<span class="dv">1</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">alien_score =</span> boot_value)</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>alien_H_boot_sf <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(alien_H_boot_sf,</span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>alien_H_boot_sf<span class="sc">$</span>alien_score_scaled <span class="ot">&lt;-</span> eaTools<span class="sc">::</span><span class="fu">ea_normalise</span>(alien_H_boot_sf,</span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">vector =</span> <span class="st">&quot;alien_score&quot;</span>, </span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">upper_reference_level =</span> <span class="dv">10</span>,</span>
<span id="cb122-15"><a href="#cb122-15" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">lower_reference_level =</span> <span class="dv">0</span>,</span>
<span id="cb122-16"><a href="#cb122-16" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">break_point =</span> <span class="dv">2</span>, </span>
<span id="cb122-17"><a href="#cb122-17" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">reverse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb122-18"><a href="#cb122-18" aria-hidden="true" tabindex="-1"></a>                                                            <span class="at">plot =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>This is then ready for export.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>alien_H_boot_sf</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 4 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 5
##   region                                                                                       geom  year alien_score alien_score_scaled
## * &lt;chr&gt;                                                                          &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;       &lt;dbl&gt;              &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2020       0.9                0.82 
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2021       0.910              0.818
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2022       0.908              0.818
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030, 2...  2020       0.699              0.860
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030, 2...  2021       0.696              0.861
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030, 2...  2022       0.700              0.860
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581, 1...  2020       0.941              0.812
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581, 1...  2021       0.942              0.812
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581, 1...  2022       0.926              0.815</code></pre>
</div>
<div id="genetic-diversity-evenness-of-flying-insects-in-semi-natural-land" class="section level4" number="1.8.4.7">
<h4><span class="header-section-number">1.8.4.7</span> Genetic diversity evenness of flying insects in semi-natural land</h4>
<p>This calculation is not fully implemented yet, and the script shown here functions as a placeholder, awaiting a proper routines. Here, the intraspecific variation is based simply on the number of unique genetic variants (ASVs), and their relative abundance measured as number of DNA-reads in the samples. But to do this properly, we need to use the full COI sequence to calculate the true genetic diversity (GD), and from there the mean diversity (GDM) and the evenness (GDE) of a set of species in a spatial and temporal unit (see Connor et al. 2023). This work is on the way.</p>
<p><strong>Update</strong>, Marie now have calculated the GDM manually, for all species on the year locality level. The GDE still remains to be calculated.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>gdm <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;GDM_year_locality.rds&quot;</span>)</span></code></pre></div>
<p>The <code>gdm</code> contains the mean genetic diversity for the species found in every year locality combination. This is calculated after first filtering on xxx, xxxx.</p>
<p>The <code>gdm</code> object contains the year_locality_ids as a key, which we can join to the relevant metadata, e.g. through the alien observation object used above.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>gdm_yl <span class="ot">&lt;-</span> alien_obs_yl  <span class="sc">%&gt;%</span> </span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(yl_id, locality, year, region_name) <span class="sc">%&gt;%</span> </span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gdm, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;yl_id&quot;</span><span class="ot">=</span> <span class="st">&quot;year_locality_id&quot;</span>))</span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>gdm_yl</span></code></pre></div>
<pre><code>## # A tibble: 60 × 5
##    yl_id                                locality     year region_name     GDM
##    &lt;chr&gt;                                &lt;chr&gt;       &lt;int&gt; &lt;chr&gt;         &lt;dbl&gt;
##  1 a3f86975-b0c2-4200-9860-e64d22f2d40f Semi-nat_01  2020 Østlandet   0.00723
##  2 11b4435f-8cc9-4ba2-9d65-b0ea8240bb49 Semi-nat_02  2020 Østlandet   0.00947
##  3 0b2956ca-3449-47b8-b932-98979ac6d8d9 Semi-nat_03  2020 Østlandet   0.00883
##  4 f0412a3b-7864-472d-a834-e5e2cced1973 Semi-nat_04  2020 Østlandet   0.00618
##  5 b462682b-d037-43fd-b1e3-86a5259720b3 Semi-nat_05  2020 Østlandet   0.00973
##  6 c76f63db-56a2-47c4-b2a7-6d0f78848850 Semi-nat_06  2020 Østlandet   0.00950
##  7 0256d2aa-07f1-45ce-a33d-1e56fb44a995 Semi-nat_07  2020 Østlandet   0.00613
##  8 271cb773-e404-46a7-a7e0-d6991f593863 Semi-nat_08  2020 Østlandet   0.00999
##  9 9768a7ca-0bc8-43bf-945e-f4ee5b8ed602 Semi-nat_09  2020 Østlandet   0.0111 
## 10 6a19c4e6-8226-432c-9444-339668ba8324 Semi-nat_10  2020 Østlandet   0.00840
## # ℹ 50 more rows</code></pre>
<p>We can explore how the mean genetic diversity relates to the total species richness of the localities.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a>richness_yl <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">agg_level =</span> <span class="st">&quot;year_locality&quot;</span>,</span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>richness_gdm_join <span class="ot">&lt;-</span> richness_yl <span class="sc">%&gt;%</span> </span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gdm_yl,</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;locality&quot;</span> <span class="ot">=</span> <span class="st">&quot;locality&quot;</span>,</span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;year&quot;</span> <span class="ot">=</span> <span class="st">&quot;year&quot;</span>,</span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;region_name&quot;</span> <span class="ot">=</span> <span class="st">&quot;region_name&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(richness_gdm_join) <span class="sc">+</span></span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> no_species,</span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> GDM,</span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> region_name)) <span class="sc">+</span></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">x =</span> no_species,</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> GDM,</span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> region_name),</span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> lm)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-104-1.png" width="672" />
Interestingly, and perhaps reassuringly, the number of species doesn’t seem to be related to the average genetic diversity. We can look at the mean number of ASVs as well.</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(richness_gdm_join) <span class="sc">+</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_no_asv_per_species,</span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> GDM,</span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> region_name)) <span class="sc">+</span></span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_no_asv_per_species,</span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> GDM,</span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> region_name),</span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> lm)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-105-1.png" width="672" />
While the number of ASVs (genetic variants) and the number of species per locality show more clear relationships.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(richness_gdm_join) <span class="sc">+</span></span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> no_species,</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> mean_no_asv_per_species,</span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> region_name)) <span class="sc">+</span></span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">x =</span> no_species,</span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">y =</span> mean_no_asv_per_species,</span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> region_name),</span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">method =</span> lm)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-106-1.png" width="672" /></p>
<p>We can bootstrap the mean genetic diversity like with the other indicators.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>gdm_yl_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> gdm_yl,</span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">value =</span> GDM,</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="st">&quot;region_name&quot;</span>)</span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>                                   )</span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>gdm_yl_boot</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name     GDM  boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;         &lt;dbl&gt;    &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet   0.00892 0.000557       0.00796        0.0101 
## 2  2020 Trøndelag   0.00841 0.000379       0.00764        0.00915
## 3  2020 Østlandet   0.00882 0.000252       0.00832        0.00931
## 4  2021 Sørlandet   0.00890 0.000562       0.00790        0.0102 
## 5  2021 Trøndelag   0.00840 0.000371       0.00765        0.00911
## 6  2021 Østlandet   0.00884 0.000265       0.00832        0.00937
## 7  2022 Sørlandet   0.00892 0.000583       0.00792        0.0102 
## 8  2022 Trøndelag   0.00841 0.000358       0.00772        0.00909
## 9  2022 Østlandet   0.00883 0.000264       0.00828        0.00932</code></pre>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(gdm_yl_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-108-1.png" width="672" /></p>
<p>So far, Trøndelag seem to have lower genetic diversity than Sørlandet and Østlandet.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ts_plot</span>(gdm_yl_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-109-1.png" width="672" /></p>
<p>Although, as with the other indicators, we see variation between years. This can be seen if we do not bootstrap on a rolling 5 year window.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>gdm_yl_yearly_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> gdm_yl,</span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">value =</span> GDM,</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="st">&quot;region_name&quot;</span>),</span>
<span id="cb140-5"><a href="#cb140-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">rolling_year_window =</span> <span class="cn">FALSE</span></span>
<span id="cb140-6"><a href="#cb140-6" aria-hidden="true" tabindex="-1"></a>                                   )</span></code></pre></div>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ts_plot</span>(gdm_yl_yearly_boot)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-111-1.png" width="672" /></p>
<p>Below is some placeholder calculation, in wait of the proper GDE calculations.</p>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>richness_sn <span class="ot">&lt;-</span> <span class="fu">get_observations</span>(<span class="at">subset_year =</span> <span class="dv">2020</span><span class="sc">:</span><span class="dv">2022</span>,</span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">subset_habitat =</span> <span class="st">&quot;Semi-nat&quot;</span>)</span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>gde_sn_boot <span class="ot">&lt;-</span> <span class="fu">bootstrap_value</span>(<span class="at">df =</span> richness_sn,</span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">value =</span> GDE_by_asv,</span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">groups =</span> <span class="fu">c</span>(<span class="st">&quot;year&quot;</span>,</span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>                                              <span class="st">&quot;region_name&quot;</span>)</span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a>                                   )</span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>gde_sn_ref <span class="ot">&lt;-</span> <span class="fl">0.7</span></span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>gde_sn_diff <span class="ot">&lt;-</span> gde_sn_boot <span class="sc">/</span> gde_sn_ref</span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>gde_sn_diff</span></code></pre></div>
<pre><code>## # A tibble: 9 × 6
##    year region_name GDE_by_asv boot_sd boot_lower2.5 boot_upper97.5
##   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;          &lt;dbl&gt;
## 1  2020 Sørlandet        0.810  0.0290         0.754          0.867
## 2  2020 Trøndelag        0.750  0.0163         0.719          0.783
## 3  2020 Østlandet        0.900  0.0204         0.857          0.939
## 4  2021 Sørlandet        0.810  0.0289         0.753          0.866
## 5  2021 Trøndelag        0.751  0.0167         0.718          0.783
## 6  2021 Østlandet        0.900  0.0194         0.861          0.939
## 7  2022 Sørlandet        0.808  0.0264         0.757          0.86 
## 8  2022 Trøndelag        0.751  0.0170         0.717          0.785
## 9  2022 Østlandet        0.899  0.0203         0.857          0.936</code></pre>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(gde_sn_diff)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/unnamed-chunk-116-1.png" width="672" /></p>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>gde_sn_to_exp <span class="ot">&lt;-</span> gde_sn_diff<span class="sc">$</span>bootstrap_summary</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>gde_sn_to_exp <span class="ot">&lt;-</span> gde_sn_to_exp <span class="sc">%&gt;%</span> </span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(year,</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> region_name,</span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> boot_value,</span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> boot_sd,</span>
<span id="cb148-8"><a href="#cb148-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> boot_lower2<span class="fl">.5</span>,</span>
<span id="cb148-9"><a href="#cb148-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> boot_upper97<span class="fl">.5</span>)</span>
<span id="cb148-10"><a href="#cb148-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-11"><a href="#cb148-11" aria-hidden="true" tabindex="-1"></a>gde_sn_to_exp <span class="ot">&lt;-</span> norway_reg_NorIns  <span class="sc">%&gt;%</span> </span>
<span id="cb148-12"><a href="#cb148-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(gde_sn_to_exp,</span>
<span id="cb148-13"><a href="#cb148-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb148-14"><a href="#cb148-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb148-15"><a href="#cb148-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-16"><a href="#cb148-16" aria-hidden="true" tabindex="-1"></a>gde_sn_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 9 features and 6 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 9 × 7
##   region                                                                                  geom  year ind_value ind_sd ind_lower2.5 ind_upper97.5
##   &lt;chr&gt;                                                                     &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
## 1 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2020     0.810 0.0290        0.754         0.867
## 2 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2021     0.810 0.0289        0.753         0.866
## 3 Sørlandet (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740…  2022     0.808 0.0264        0.757         0.86 
## 4 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2020     0.750 0.0163        0.719         0.783
## 5 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2021     0.751 0.0167        0.718         0.783
## 6 Trøndelag (((227783 7068646, 227876 7069284, 227998 7069292, 228247 7069134, 228869 7069030…  2022     0.751 0.0170        0.717         0.785
## 7 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2020     0.900 0.0204        0.857         0.939
## 8 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2021     0.900 0.0194        0.861         0.939
## 9 Østlandet (((179915 6537265, 180246 6536997, 180600 6536671, 181028 6536601, 181197 6536581…  2022     0.899 0.0203        0.857         0.936</code></pre>
</div>
</div>
<div id="calculation-of-nbbm-indicators" class="section level3" number="1.8.5">
<h3><span class="header-section-number">1.8.5</span> Calculation of NBBM indicators</h3>
</div>
<div id="bumblebees-in-open-landscapes" class="section level3" number="1.8.6">
<h3><span class="header-section-number">1.8.6</span> Bumblebees in open landscapes</h3>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(bombLepiSurv)</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a><span class="fu">humlesommerfConnect</span>(<span class="at">host =</span> <span class="st">&quot;ninradardata01.nina.no&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>allBombusGressmark2022 <span class="ot">&lt;-</span> <span class="fu">getAllData</span>(<span class="at">type =</span> <span class="st">&quot;bumblebees&quot;</span>,</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">habitat =</span> <span class="st">&quot;gressmark&quot;</span>,</span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">language =</span> <span class="st">&quot;norsk&quot;</span>)</span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a><span class="fu">surveyBarplot</span>(allBombusGressmark2022)</span></code></pre></div>
<div class="figure">
<img src="DRAFT_insectIndicators_files/figure-html/bombusGressBarplot-1.png" alt="Bumblebees in grasslands 2021" width="672" />
<p class="caption">
(#fig:bombusGressBarplot)Bumblebees in grasslands 2021
</p>
</div>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a>nIter <span class="ot">=</span> <span class="dv">999</span></span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a>hInd2022TrondGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Trond&quot;</span>,</span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb152-7"><a href="#cb152-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb152-8"><a href="#cb152-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a>hInd2022SorGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Sor&quot;</span>,</span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>hInd2022OstGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Ost&quot;</span>,</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a>hInd2022VestGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Humler&quot;</span>,</span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Vest&quot;</span>,</span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
</div>
<div id="butterflies-in-open-landscapes" class="section level3" number="1.8.7">
<h3><span class="header-section-number">1.8.7</span> Butterflies in open landscapes</h3>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>nIter <span class="ot">=</span> <span class="dv">999</span></span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a>sInd2022TrondGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Sommerfugl&quot;</span>,</span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Trond&quot;</span>,</span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>sInd2022SorGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Sommerfugl&quot;</span>,</span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Sor&quot;</span>,</span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>sInd2022OstGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Sommerfugl&quot;</span>,</span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Ost&quot;</span>,</span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a>sInd2022VestGress <span class="ot">&lt;-</span> <span class="fu">calcInd</span>(<span class="at">type =</span> <span class="st">&quot;Sommerfugl&quot;</span>,</span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">region_short =</span> <span class="st">&quot;Vest&quot;</span>,</span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">habitat =</span> <span class="st">&quot;Gressmark&quot;</span>,</span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">year =</span> <span class="dv">2022</span>,</span>
<span id="cb159-5"><a href="#cb159-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">nIter =</span> nIter,</span>
<span id="cb159-6"><a href="#cb159-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">save.draws =</span> T)</span></code></pre></div>
<p>The calculations for earlier years are already present as datasets in the R-package and can be fetched individually. Once this is done, we can plot the timeseries of values using the function <code>indPlot</code>.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022VestGress&quot;</span>)</span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022VestSkog&quot;</span>)</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022TrondGress&quot;</span>)</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022TrondSkog&quot;</span>)</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022OstGress&quot;</span>)</span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022OstSkog&quot;</span>)</span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022SorGress&quot;</span>)</span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2022SorSkog&quot;</span>)</span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-10"><a href="#cb160-10" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2021TrondGress&quot;</span>)</span>
<span id="cb160-11"><a href="#cb160-11" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2021TrondSkog&quot;</span>)</span>
<span id="cb160-12"><a href="#cb160-12" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2021OstGress&quot;</span>)</span>
<span id="cb160-13"><a href="#cb160-13" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2021OstSkog&quot;</span>)</span>
<span id="cb160-14"><a href="#cb160-14" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2021SorGress&quot;</span>)</span>
<span id="cb160-15"><a href="#cb160-15" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2021SorSkog&quot;</span>)</span>
<span id="cb160-16"><a href="#cb160-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-17"><a href="#cb160-17" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2020TrondGress&quot;</span>)</span>
<span id="cb160-18"><a href="#cb160-18" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2020TrondSkog&quot;</span>)</span>
<span id="cb160-19"><a href="#cb160-19" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2020OstGress&quot;</span>)</span>
<span id="cb160-20"><a href="#cb160-20" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2020OstSkog&quot;</span>)</span>
<span id="cb160-21"><a href="#cb160-21" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2020SorGress&quot;</span>)</span>
<span id="cb160-22"><a href="#cb160-22" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2020SorSkog&quot;</span>)</span>
<span id="cb160-23"><a href="#cb160-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-24"><a href="#cb160-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-25"><a href="#cb160-25" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2019TrondGress&quot;</span>)</span>
<span id="cb160-26"><a href="#cb160-26" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2019TrondSkog&quot;</span>)</span>
<span id="cb160-27"><a href="#cb160-27" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2019OstGress&quot;</span>)</span>
<span id="cb160-28"><a href="#cb160-28" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2019OstSkog&quot;</span>)</span>
<span id="cb160-29"><a href="#cb160-29" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2019SorGress&quot;</span>)</span>
<span id="cb160-30"><a href="#cb160-30" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2019SorSkog&quot;</span>)</span>
<span id="cb160-31"><a href="#cb160-31" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2018TrondGress&quot;</span>)</span>
<span id="cb160-32"><a href="#cb160-32" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2018TrondSkog&quot;</span>)</span>
<span id="cb160-33"><a href="#cb160-33" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2018OstGress&quot;</span>)</span>
<span id="cb160-34"><a href="#cb160-34" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2018OstSkog&quot;</span>)</span>
<span id="cb160-35"><a href="#cb160-35" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2018SorGress&quot;</span>)</span>
<span id="cb160-36"><a href="#cb160-36" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2018SorSkog&quot;</span>)</span>
<span id="cb160-37"><a href="#cb160-37" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2017TrondGress&quot;</span>)</span>
<span id="cb160-38"><a href="#cb160-38" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2017TrondSkog&quot;</span>)</span>
<span id="cb160-39"><a href="#cb160-39" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2017OstGress&quot;</span>)</span>
<span id="cb160-40"><a href="#cb160-40" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2017OstSkog&quot;</span>)</span>
<span id="cb160-41"><a href="#cb160-41" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2017SorGress&quot;</span>)</span>
<span id="cb160-42"><a href="#cb160-42" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2017SorSkog&quot;</span>)</span>
<span id="cb160-43"><a href="#cb160-43" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2016TrondGress&quot;</span>)</span>
<span id="cb160-44"><a href="#cb160-44" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2016TrondSkog&quot;</span>)</span>
<span id="cb160-45"><a href="#cb160-45" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2016OstGress&quot;</span>)</span>
<span id="cb160-46"><a href="#cb160-46" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2016OstSkog&quot;</span>)</span>
<span id="cb160-47"><a href="#cb160-47" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2016SorGress&quot;</span>)</span>
<span id="cb160-48"><a href="#cb160-48" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2016SorSkog&quot;</span>)</span>
<span id="cb160-49"><a href="#cb160-49" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2015TrondGress&quot;</span>)</span>
<span id="cb160-50"><a href="#cb160-50" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2015TrondSkog&quot;</span>)</span>
<span id="cb160-51"><a href="#cb160-51" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2015OstGress&quot;</span>)</span>
<span id="cb160-52"><a href="#cb160-52" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2015OstSkog&quot;</span>)</span>
<span id="cb160-53"><a href="#cb160-53" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2015SorGress&quot;</span>)</span>
<span id="cb160-54"><a href="#cb160-54" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2015SorSkog&quot;</span>)</span>
<span id="cb160-55"><a href="#cb160-55" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2014TrondGress&quot;</span>)</span>
<span id="cb160-56"><a href="#cb160-56" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2014TrondSkog&quot;</span>)</span>
<span id="cb160-57"><a href="#cb160-57" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2014OstGress&quot;</span>)</span>
<span id="cb160-58"><a href="#cb160-58" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2014OstSkog&quot;</span>)</span>
<span id="cb160-59"><a href="#cb160-59" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2014SorGress&quot;</span>)</span>
<span id="cb160-60"><a href="#cb160-60" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2014SorSkog&quot;</span>)</span>
<span id="cb160-61"><a href="#cb160-61" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2013TrondGress&quot;</span>)</span>
<span id="cb160-62"><a href="#cb160-62" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2013TrondSkog&quot;</span>)</span>
<span id="cb160-63"><a href="#cb160-63" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2013OstGress&quot;</span>)</span>
<span id="cb160-64"><a href="#cb160-64" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2013OstSkog&quot;</span>)</span>
<span id="cb160-65"><a href="#cb160-65" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2013SorGress&quot;</span>)</span>
<span id="cb160-66"><a href="#cb160-66" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2013SorSkog&quot;</span>)</span>
<span id="cb160-67"><a href="#cb160-67" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2012TrondGress&quot;</span>)</span>
<span id="cb160-68"><a href="#cb160-68" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2012TrondSkog&quot;</span>)</span>
<span id="cb160-69"><a href="#cb160-69" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2012OstGress&quot;</span>)</span>
<span id="cb160-70"><a href="#cb160-70" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2012OstSkog&quot;</span>)</span>
<span id="cb160-71"><a href="#cb160-71" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2011OstGress&quot;</span>)</span>
<span id="cb160-72"><a href="#cb160-72" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2011OstSkog&quot;</span>)</span>
<span id="cb160-73"><a href="#cb160-73" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2010OstGress&quot;</span>)</span>
<span id="cb160-74"><a href="#cb160-74" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2010OstSkog&quot;</span>)</span>
<span id="cb160-75"><a href="#cb160-75" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2009OstGress&quot;</span>)</span>
<span id="cb160-76"><a href="#cb160-76" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sInd2009OstSkog&quot;</span>)</span>
<span id="cb160-77"><a href="#cb160-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-78"><a href="#cb160-78" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022VestGress&quot;</span>)</span>
<span id="cb160-79"><a href="#cb160-79" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022VestSkog&quot;</span>)</span>
<span id="cb160-80"><a href="#cb160-80" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022TrondGress&quot;</span>)</span>
<span id="cb160-81"><a href="#cb160-81" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022TrondSkog&quot;</span>)</span>
<span id="cb160-82"><a href="#cb160-82" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022OstGress&quot;</span>)</span>
<span id="cb160-83"><a href="#cb160-83" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022OstSkog&quot;</span>)</span>
<span id="cb160-84"><a href="#cb160-84" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022SorGress&quot;</span>)</span>
<span id="cb160-85"><a href="#cb160-85" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2022SorSkog&quot;</span>)</span>
<span id="cb160-86"><a href="#cb160-86" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2021TrondGress&quot;</span>)</span>
<span id="cb160-87"><a href="#cb160-87" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2021TrondSkog&quot;</span>)</span>
<span id="cb160-88"><a href="#cb160-88" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2021OstGress&quot;</span>)</span>
<span id="cb160-89"><a href="#cb160-89" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2021OstSkog&quot;</span>)</span>
<span id="cb160-90"><a href="#cb160-90" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2021SorGress&quot;</span>)</span>
<span id="cb160-91"><a href="#cb160-91" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2021SorSkog&quot;</span>)</span>
<span id="cb160-92"><a href="#cb160-92" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2020TrondGress&quot;</span>)</span>
<span id="cb160-93"><a href="#cb160-93" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2020TrondSkog&quot;</span>)</span>
<span id="cb160-94"><a href="#cb160-94" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2020OstGress&quot;</span>)</span>
<span id="cb160-95"><a href="#cb160-95" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2020OstSkog&quot;</span>)</span>
<span id="cb160-96"><a href="#cb160-96" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2020SorGress&quot;</span>)</span>
<span id="cb160-97"><a href="#cb160-97" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2020SorSkog&quot;</span>)</span>
<span id="cb160-98"><a href="#cb160-98" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2019TrondGress&quot;</span>)</span>
<span id="cb160-99"><a href="#cb160-99" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2019TrondSkog&quot;</span>)</span>
<span id="cb160-100"><a href="#cb160-100" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2019OstGress&quot;</span>)</span>
<span id="cb160-101"><a href="#cb160-101" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2019OstSkog&quot;</span>)</span>
<span id="cb160-102"><a href="#cb160-102" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2019SorGress&quot;</span>)</span>
<span id="cb160-103"><a href="#cb160-103" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2019SorSkog&quot;</span>)</span>
<span id="cb160-104"><a href="#cb160-104" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2018TrondGress&quot;</span>)</span>
<span id="cb160-105"><a href="#cb160-105" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2018TrondSkog&quot;</span>)</span>
<span id="cb160-106"><a href="#cb160-106" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2018OstGress&quot;</span>)</span>
<span id="cb160-107"><a href="#cb160-107" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2018OstSkog&quot;</span>)</span>
<span id="cb160-108"><a href="#cb160-108" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2018SorGress&quot;</span>)</span>
<span id="cb160-109"><a href="#cb160-109" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2018SorSkog&quot;</span>)</span>
<span id="cb160-110"><a href="#cb160-110" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2017TrondGress&quot;</span>)</span>
<span id="cb160-111"><a href="#cb160-111" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2017TrondSkog&quot;</span>)</span>
<span id="cb160-112"><a href="#cb160-112" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2017OstGress&quot;</span>)</span>
<span id="cb160-113"><a href="#cb160-113" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2017OstSkog&quot;</span>)</span>
<span id="cb160-114"><a href="#cb160-114" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2017SorGress&quot;</span>)</span>
<span id="cb160-115"><a href="#cb160-115" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2017SorSkog&quot;</span>)</span>
<span id="cb160-116"><a href="#cb160-116" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2016TrondGress&quot;</span>)</span>
<span id="cb160-117"><a href="#cb160-117" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2016TrondSkog&quot;</span>)</span>
<span id="cb160-118"><a href="#cb160-118" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2016OstGress&quot;</span>)</span>
<span id="cb160-119"><a href="#cb160-119" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2016OstSkog&quot;</span>)</span>
<span id="cb160-120"><a href="#cb160-120" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2016SorGress&quot;</span>)</span>
<span id="cb160-121"><a href="#cb160-121" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2016SorSkog&quot;</span>)</span>
<span id="cb160-122"><a href="#cb160-122" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2015TrondGress&quot;</span>)</span>
<span id="cb160-123"><a href="#cb160-123" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2015TrondSkog&quot;</span>)</span>
<span id="cb160-124"><a href="#cb160-124" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2015OstGress&quot;</span>)</span>
<span id="cb160-125"><a href="#cb160-125" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2015OstSkog&quot;</span>)</span>
<span id="cb160-126"><a href="#cb160-126" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2015SorGress&quot;</span>)</span>
<span id="cb160-127"><a href="#cb160-127" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2015SorSkog&quot;</span>)</span>
<span id="cb160-128"><a href="#cb160-128" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2014TrondGress&quot;</span>)</span>
<span id="cb160-129"><a href="#cb160-129" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2014TrondSkog&quot;</span>)</span>
<span id="cb160-130"><a href="#cb160-130" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2014OstGress&quot;</span>)</span>
<span id="cb160-131"><a href="#cb160-131" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2014OstSkog&quot;</span>)</span>
<span id="cb160-132"><a href="#cb160-132" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2014SorGress&quot;</span>)</span>
<span id="cb160-133"><a href="#cb160-133" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2014SorSkog&quot;</span>)</span>
<span id="cb160-134"><a href="#cb160-134" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2013TrondGress&quot;</span>)</span>
<span id="cb160-135"><a href="#cb160-135" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2013TrondSkog&quot;</span>)</span>
<span id="cb160-136"><a href="#cb160-136" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2013OstGress&quot;</span>)</span>
<span id="cb160-137"><a href="#cb160-137" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2013OstSkog&quot;</span>)</span>
<span id="cb160-138"><a href="#cb160-138" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2013SorGress&quot;</span>)</span>
<span id="cb160-139"><a href="#cb160-139" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2013SorSkog&quot;</span>)</span>
<span id="cb160-140"><a href="#cb160-140" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2012TrondGress&quot;</span>)</span>
<span id="cb160-141"><a href="#cb160-141" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2012TrondSkog&quot;</span>)</span>
<span id="cb160-142"><a href="#cb160-142" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2012OstGress&quot;</span>)</span>
<span id="cb160-143"><a href="#cb160-143" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2012OstSkog&quot;</span>)</span>
<span id="cb160-144"><a href="#cb160-144" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2011OstGress&quot;</span>)</span>
<span id="cb160-145"><a href="#cb160-145" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2011OstSkog&quot;</span>)</span>
<span id="cb160-146"><a href="#cb160-146" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2010OstGress&quot;</span>)</span>
<span id="cb160-147"><a href="#cb160-147" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2010OstSkog&quot;</span>)</span>
<span id="cb160-148"><a href="#cb160-148" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2009OstGress&quot;</span>)</span>
<span id="cb160-149"><a href="#cb160-149" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;hInd2009OstSkog&quot;</span>)</span></code></pre></div>
<p>Once loaded, the yearly datasets can be organized through the convenience function <code>collectIndData</code>, and plotted with <code>indPlot</code>.</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>hGressIndVals <span class="ot">&lt;-</span> <span class="fu">collectIndData</span>(<span class="at">taxa =</span> <span class="st">&quot;bumblebees&quot;</span>, </span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">habitat =</span> <span class="st">&quot;gressmark&quot;</span>,</span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">years =</span> <span class="dv">2009</span><span class="sc">:</span><span class="dv">2022</span>)</span>
<span id="cb161-4"><a href="#cb161-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-5"><a href="#cb161-5" aria-hidden="true" tabindex="-1"></a><span class="fu">indPlot</span>(<span class="at">input =</span> hGressIndVals,</span>
<span id="cb161-6"><a href="#cb161-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">confLevel =</span> <span class="st">&quot;95%&quot;</span>,</span>
<span id="cb161-7"><a href="#cb161-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">lineWidth =</span> <span class="dv">1</span>,</span>
<span id="cb161-8"><a href="#cb161-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">dodge =</span> <span class="fl">0.5</span>,</span>
<span id="cb161-9"><a href="#cb161-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pointSize =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb161-10"><a href="#cb161-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Indicator values for bumblebees in semi-natural landscapes&quot;</span>) <span class="sc">+</span></span>
<span id="cb161-11"><a href="#cb161-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="dv">2009</span><span class="sc">:</span><span class="dv">2022</span>)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/bombusGressIndPlot-1.png" width="672" /></p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>sGressIndVals <span class="ot">&lt;-</span> <span class="fu">collectIndData</span>(<span class="at">taxa =</span> <span class="st">&quot;butterflies&quot;</span>, </span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a>                                <span class="at">habitat =</span> <span class="st">&quot;gressmark&quot;</span>,</span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">years =</span> <span class="dv">2009</span><span class="sc">:</span><span class="dv">2022</span>)</span>
<span id="cb162-4"><a href="#cb162-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-5"><a href="#cb162-5" aria-hidden="true" tabindex="-1"></a><span class="fu">indPlot</span>(<span class="at">input =</span> sGressIndVals,</span>
<span id="cb162-6"><a href="#cb162-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">confLevel =</span> <span class="st">&quot;95%&quot;</span>,</span>
<span id="cb162-7"><a href="#cb162-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">lineWidth =</span> <span class="dv">1</span>,</span>
<span id="cb162-8"><a href="#cb162-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">dodge =</span> <span class="fl">0.5</span>,</span>
<span id="cb162-9"><a href="#cb162-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">pointSize =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb162-10"><a href="#cb162-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Indicator values for butterflies in semi-natural landscapes&quot;</span>) <span class="sc">+</span></span>
<span id="cb162-11"><a href="#cb162-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="dv">2009</span><span class="sc">:</span><span class="dv">2022</span>)</span></code></pre></div>
<p><img src="DRAFT_insectIndicators_files/figure-html/lepiGressIndPlot-1.png" width="672" /></p>
</div>
</div>
<div id="prepare-export" class="section level2" number="1.9">
<h2><span class="header-section-number">1.9</span> Prepare export</h2>
<p>We here fetch the indicator regions as an sf and join the calculated indicator values.</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>norway_reg_NBBM <span class="ot">&lt;-</span> bombLepiSurv<span class="sc">::</span><span class="fu">get_map</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(region <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Sør&quot;</span>, <span class="st">&quot;Vestlandet&quot;</span>, <span class="st">&quot;Øst&quot;</span>, <span class="st">&quot;Trøndelag&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb163-3"><a href="#cb163-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(region) <span class="sc">%&gt;%</span> </span>
<span id="cb163-4"><a href="#cb163-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(region) <span class="sc">%&gt;%</span> </span>
<span id="cb163-5"><a href="#cb163-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">geom =</span> <span class="fu">st_union</span>(geom))</span></code></pre></div>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a>hGressIndVals_to_exp <span class="ot">&lt;-</span> hGressIndVals <span class="sc">%&gt;%</span> </span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">year =</span> Year,</span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> Region,</span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> Indicator_value,</span>
<span id="cb164-5"><a href="#cb164-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> <span class="cn">NULL</span>,</span>
<span id="cb164-6"><a href="#cb164-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> lower95,</span>
<span id="cb164-7"><a href="#cb164-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> upper95)</span>
<span id="cb164-8"><a href="#cb164-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-9"><a href="#cb164-9" aria-hidden="true" tabindex="-1"></a>hGressIndVals_to_exp <span class="ot">&lt;-</span> norway_reg_NBBM  <span class="sc">%&gt;%</span> </span>
<span id="cb164-10"><a href="#cb164-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(hGressIndVals_to_exp,</span>
<span id="cb164-11"><a href="#cb164-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb164-12"><a href="#cb164-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb164-13"><a href="#cb164-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-14"><a href="#cb164-14" aria-hidden="true" tabindex="-1"></a>hGressIndVals_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 36 features and 5 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 36 × 6
##    region                                                                                    geom  year ind_value ind_lower2.5 ind_upper97.5
##    &lt;chr&gt;                                                                       &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
##  1 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2013     0.596        0.423         0.635
##  2 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2014     0.635        0.5           0.673
##  3 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2015     0.519        0.385         0.558
##  4 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2016     0.635        0.462         0.673
##  5 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2017     0.615        0.423         0.654
##  6 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2018     0.577        0.462         0.615
##  7 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2019     0.519        0.422         0.558
##  8 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2020     0.593        0.444         0.593
##  9 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2021     0.519        0.389         0.593
## 10 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2022     0.593        0.370         0.704
## # ℹ 26 more rows</code></pre>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>sGressIndVals_to_exp <span class="ot">&lt;-</span> sGressIndVals <span class="sc">%&gt;%</span> </span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">year =</span> Year,</span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">region =</span> Region,</span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_value =</span> Indicator_value,</span>
<span id="cb166-5"><a href="#cb166-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_sd =</span> <span class="cn">NULL</span>,</span>
<span id="cb166-6"><a href="#cb166-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_lower2.5 =</span> lower95,</span>
<span id="cb166-7"><a href="#cb166-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ind_upper97.5 =</span> upper95)</span>
<span id="cb166-8"><a href="#cb166-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb166-9"><a href="#cb166-9" aria-hidden="true" tabindex="-1"></a>sGressIndVals_to_exp <span class="ot">&lt;-</span> norway_reg_NBBM  <span class="sc">%&gt;%</span> </span>
<span id="cb166-10"><a href="#cb166-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(sGressIndVals_to_exp,</span>
<span id="cb166-11"><a href="#cb166-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;region&quot;</span> <span class="ot">=</span> <span class="st">&quot;region&quot;</span>),</span>
<span id="cb166-12"><a href="#cb166-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">multiple =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb166-13"><a href="#cb166-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb166-14"><a href="#cb166-14" aria-hidden="true" tabindex="-1"></a>sGressIndVals_to_exp</span></code></pre></div>
<pre><code>## Simple feature collection with 36 features and 5 fields
## Geometry type: MULTIPOLYGON
## Dimension:     XY
## Bounding box:  xmin: -76208 ymin: 6450245 xmax: 468357.5 ymax: 7237323
## Projected CRS: ETRS89 / UTM zone 33N
## # A tibble: 36 × 6
##    region                                                                                    geom  year ind_value ind_lower2.5 ind_upper97.5
##    &lt;chr&gt;                                                                       &lt;MULTIPOLYGON [m]&gt; &lt;int&gt;     &lt;dbl&gt;        &lt;dbl&gt;         &lt;dbl&gt;
##  1 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2013     0.259        0.151         0.295
##  2 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2014     0.374        0.273         0.424
##  3 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2015     0.345        0.230         0.360
##  4 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2016     0.302        0.216         0.353
##  5 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2017     0.345        0.237         0.367
##  6 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2018     0.309        0.201         0.345
##  7 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2019     0.388        0.281         0.417
##  8 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2020     0.419        0.284         0.432
##  9 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2021     0.399        0.270         0.439
## 10 Sør    (((-31605 6513854, -31464 6513743, -31449 6513590, -32255 6512745, -32512 6512740, -...  2022     0.358        0.25          0.399
## # ℹ 26 more rows</code></pre>
<div id="eksport-file-final-product" class="section level3" number="1.9.1">
<h3><span class="header-section-number">1.9.1</span> Eksport file (final product)</h3>
<!-- Export final file. Ideally a georeferenced shape or raster with indicators values (raw and normalised), reference values and errors. -->
<p>TBD. Need to consult with others.</p>
</div>
</div>
</div>
<!--bookdown:body:end-->
  </main>

  <div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page">
      <h2>On this page</h2>
      <div id="book-on-this-page"></div>

      <div class="book-extra">
        <ul class="list-unstyled">
          <li><a id="book-source" href="#">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="#">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
      </div>
    </nav>
  </div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5">
  <div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong></strong>" was written by . </p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
<script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>

</html>
