<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 10 Insect indicators | Indicators for Ecosystem Condition in Norway</title>
<meta name="author" content="Anders L. Kolstad">
<meta name="description" content="Author and date: Jens Åström Marie Davey 2023-12-05   .cl-809c7056{table-layout:auto;}.cl-808f7630{font-family:'DejaVu...">
<meta name="generator" content="bookdown 0.33 with bs4_book()">
<meta property="og:title" content="Chapter 10 Insect indicators | Indicators for Ecosystem Condition in Norway">
<meta property="og:type" content="book">
<meta property="og:url" content="https://ninanor.github.io/ecosystemCondition/index.html/insect-indicators.html">
<meta property="og:image" content="https://ninanor.github.io/ecosystemCondition/index.html/images/Picture1.png">
<meta property="og:description" content="Author and date: Jens Åström Marie Davey 2023-12-05   .cl-809c7056{table-layout:auto;}.cl-808f7630{font-family:'DejaVu...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 10 Insect indicators | Indicators for Ecosystem Condition in Norway">
<meta name="twitter:description" content="Author and date: Jens Åström Marie Davey 2023-12-05   .cl-809c7056{table-layout:auto;}.cl-808f7630{font-family:'DejaVu...">
<meta name="twitter:image" content="https://ninanor.github.io/ecosystemCondition/index.html/images/Picture1.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.2/transition.js"></script><script src="libs/bs3compat-0.4.2/tabs.js"></script><script src="libs/bs3compat-0.4.2/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet">
<script src="libs/tabwid-1.1.3/tabwid.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.6.1/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.27/datatables.js"></script><link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Indicators for Ecosystem Condition in Norway</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">About </a></li>
<li class="book-part">INDICATORS</li>
<li><a class="" href="gjengroing.html"><span class="header-section-number">1</span> Enchroachment</a></li>
<li><a class="" href="nature-index-product.html"><span class="header-section-number">2</span> Nature index product</a></li>
<li><a class="" href="slitasje.html"><span class="header-section-number">3</span> Anthropogenic disturbance to soils and vegetation (ADSV)</a></li>
<li><a class="" href="median-summer-temp.html"><span class="header-section-number">4</span> Median Summer Temperature</a></li>
<li><a class="" href="alien-species.html"><span class="header-section-number">5</span> Alien species</a></li>
<li><a class="" href="functional-plant-indicators-wetland.html"><span class="header-section-number">6</span> Functional Plant Indicators - Wetlands</a></li>
<li><a class="" href="functional-plant-indicators-seminat.html"><span class="header-section-number">7</span> Functional Plant Indicators - Semi-Natural Ecosystems</a></li>
<li><a class="" href="functional-plant-indicators-open.html"><span class="header-section-number">8</span> Functional Plant Indicators - Naturally Open Ecosystems</a></li>
<li><a class="" href="trophic-level-biomass-ratios.html"><span class="header-section-number">9</span> Trophic level biomass ratios</a></li>
<li><a class="active" href="insect-indicators.html"><span class="header-section-number">10</span> Insect indicators</a></li>
<li><a class="" href="breeding-birds.html"><span class="header-section-number">11</span> Breeding bird indicators</a></li>
<li><a class="" href="connectivity.html"><span class="header-section-number">12</span> Connectivity</a></li>
<li class="book-part">R &amp; D</li>
<li><a class="" href="NDVI-indicator-natopen.html"><span class="header-section-number">13</span> NDVI - Naturally Open Ecosystems</a></li>
<li><a class="" href="NDVI-indicator-seminat.html"><span class="header-section-number">14</span> NDVI - Semi-natural Ecosystems</a></li>
<li><a class="" href="NDVI-indicator-wetland.html"><span class="header-section-number">15</span> NDVI - wetland ecosystems</a></li>
<li><a class="" href="flood-frequency.html"><span class="header-section-number">16</span> Flood frequency</a></li>
<li class="book-part">INDICATORS FROM THE ALPINE ASSESSMENT</li>
<li><a class="" href="areal-av-isbreer.html"><span class="header-section-number">17</span> Areal av isbreer</a></li>
<li class="book-part">DEPRECATED</li>
<li><a class="" href="areal-uten-d%C3%B8d-eller-skadet-r%C3%B8sslyng-i-kystlynghei.html"><span class="header-section-number">18</span> Areal uten død eller skadet røsslyng i kystlynghei</a></li>
<li class="book-part">GENERAL TOPICS</li>
<li><a class="" href="naturtype.html"><span class="header-section-number">19</span> Nature types</a></li>
<li><a class="" href="scaling-functions.html"><span class="header-section-number">20</span> Scaling functions</a></li>
<li><a class="" href="homogeneous-impact-areas.html"><span class="header-section-number">21</span> Homogeneous Impact Areas</a></li>
<li><a class="" href="master-grid.html"><span class="header-section-number">22</span> Master grid</a></li>
<li><a class="" href="climate-indicators-explained.html"><span class="header-section-number">23</span> Climate indicators explained</a></li>
<li><a class="" href="oversikt-over-indikatorforslag.html"><span class="header-section-number">24</span> Oversikt over indikatorforslag</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/NINAnor/ecosystemCondition">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="insect-indicators" class="section level1" number="10">
<h1>
<span class="header-section-number">10</span> Insect indicators<a class="anchor" aria-label="anchor" href="#insect-indicators"><i class="fas fa-link"></i></a>
</h1>
<p><br></p>
<p><em>Author and date:</em></p>
<p>Jens Åström
Marie Davey</p>
<p>2023-12-05</p>
<p><br></p>
<!-- Load all you dependencies here -->
<!-- Fill in which ecosystem the indicator belongs to, as well as the ecosystem characteristic it should be linked to. It's OK to use some Norwegian here -->
<div class="tabwid">
<style>.cl-809c7056{table-layout:auto;}.cl-808f7630{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-808f7644{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-809595d8{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-8095bd1a{background-color:rgba(173, 216, 230, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-8095bd2e{background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-8095bd38{background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-8095bd39{background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<div class="inline-table"><table data-quarto-disable-processing="true" class="cl-809c7056">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-8095bd1a"><p class="cl-809595d8"><span class="cl-808f7630">Ecosystem</span></p></th>
<th class="cl-8095bd1a"><p class="cl-809595d8"><span class="cl-808f7630">Økologisk egenskap</span></p></th>
<th class="cl-8095bd1a"><p class="cl-809595d8"><span class="cl-808f7630">ECT class</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-8095bd2e"><p class="cl-809595d8"><span class="cl-808f7644">Semi-naturlig mark</span></p></td>
<td class="cl-8095bd2e"><p class="cl-809595d8"><span class="cl-808f7644">Biologisk mangfold</span></p></td>
<td class="cl-8095bd2e"><p class="cl-809595d8"><span class="cl-808f7644">Functional state characteristics</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-8095bd38"><p class="cl-809595d8"><span class="cl-808f7644">Semi-naturlig mark</span></p></td>
<td class="cl-8095bd38"><p class="cl-809595d8"><span class="cl-808f7644">Funksjonelle grupper innen trofiske nivåer</span></p></td>
<td class="cl-8095bd38"><p class="cl-809595d8"><span class="cl-808f7644">Functional state characteristics</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-8095bd39"><p class="cl-809595d8"><span class="cl-808f7644">Semi-naturlig mark</span></p></td>
<td class="cl-8095bd39"><p class="cl-809595d8"><span class="cl-808f7644">Funksjonelt viktige arter og biofysiske strukturer</span></p></td>
<td class="cl-8095bd39"><p class="cl-809595d8"><span class="cl-808f7644">Functional state characteristics</span></p></td>
</tr>
</tbody>
</table></div>
</div>
<!-- Don't remove these three html lines -->
<br><br><hr>
<div id="introduction-2" class="section level2" number="10.1">
<h2>
<span class="header-section-number">10.1</span> Introduction<a class="anchor" aria-label="anchor" href="#introduction-2"><i class="fas fa-link"></i></a>
</h2>
<p>We here show how to calculate a set of indicators for terrestrial insects in Norway. Two of them are adapted from earlier work on the Nature Index for Norway and use data from the Norwegian monitoring program on bumblebees and butterflies (<strong>NMBB</strong>). The others are developed during 2023, and use data from the Norwegian insect monitoring program (<strong>NorIns</strong>).</p>
<p>This workflow is shortened by putting most of the underlying code in separate R-packages, which are freely available through Github. The relevant packages are <code>bombLepiSurv</code> <a href="https://github.com/jenast/bombLepiSurv" class="uri">https://github.com/jenast/bombLepiSurv</a> for the bumblebee and butterfly indicators, and <code>Norimon</code> <a href="https://github.com/jenast/Norimon" class="uri">https://github.com/jenast/Norimon</a> for the rest of the indicators.</p>
<p>For the indicators from the NMBB program, community reference values has been elicited from experts, and is explained further below. For the indicators from the NorIns program, this is significantly harder to do, because of the size of the communities, lack of historical timeseries and changes in sampling and identification techniques. We therefore currently lack reference values for many of these indicators.</p>
<p>Below is a list of the indicators calculated in this document, and their state of development. Some details is further expanded on below.</p>
<div class="inline-table"><table class="table table-sm">
<colgroup>
<col width="4%">
<col width="25%">
<col width="26%">
<col width="2%">
<col width="26%">
<col width="7%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">Dataset</th>
<th align="left">Indicator</th>
<th align="left">Norwegian type</th>
<th align="left">ECT</th>
<th align="left">Current geographical scope</th>
<th align="left">Indicator data</th>
<th align="left">Reference values</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">NMBB</td>
<td align="left">Bumblebees in semi-natural grasslands</td>
<td align="left">Funksjonelt viktige arter og biofysiske stukturer</td>
<td align="left">B2</td>
<td align="left">Sørlandet, Østfold-Vestfold, Vestlandet, Trøndelag</td>
<td align="left">Present</td>
<td align="left">Present</td>
</tr>
<tr class="even">
<td align="left">NMBB</td>
<td align="left">Butterflies in semi-natural grasslands</td>
<td align="left">Biologisk mangfold</td>
<td align="left">B1</td>
<td align="left">Sørlandet, Østfold-Vestfold, Vestlandet, Trøndelag</td>
<td align="left">Present</td>
<td align="left">Present</td>
</tr>
<tr class="odd">
<td align="left">NorIns</td>
<td align="left">Biomass of flying insects</td>
<td align="left">Funksjonelt viktige arter og biofysiske stukturer</td>
<td align="left">B2</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
<tr class="even">
<td align="left">NorIns</td>
<td align="left">Species richness of flying insects</td>
<td align="left">Biologisk mangfold</td>
<td align="left">B1</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
<tr class="odd">
<td align="left">NorIns</td>
<td align="left">Species richness of pollinating insects</td>
<td align="left">Biologisk mangfold</td>
<td align="left">B1</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
<tr class="even">
<td align="left">NorIns</td>
<td align="left">Species richness of dung associated insects</td>
<td align="left">Biologisk mangfold</td>
<td align="left">B1</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
<tr class="odd">
<td align="left">NorIns</td>
<td align="left">Quotent of Apocrita vs Symphyta species richness</td>
<td align="left">Funksjonelt viktige arter og biofysiske strukturer</td>
<td align="left">B2</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
<tr class="even">
<td align="left">NorIns</td>
<td align="left">Number of alien species</td>
<td align="left">Biologisk mangfold</td>
<td align="left">B1</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
<tr class="odd">
<td align="left">NorIns</td>
<td align="left">Intraspesific genetic variation</td>
<td align="left">Biologisk mangfold</td>
<td align="left">B1</td>
<td align="left">Sørlandet, Østlandet, Trøndelag, Nord-Norge</td>
<td align="left">Present</td>
<td align="left">Lacking</td>
</tr>
</tbody>
</table></div>
<p>Many of these indicators are self-explanatory and don’t require much justification. <strong>Bumblebees in semi-natural grasslands</strong> and <strong>Butterflies in semi-natural grasslands</strong> are functionally important pollinators, and are as a community dependent on a varied herbaceous flora. They therefore both represent key functionality, as well as indicators of habitat quality. <strong>Biomass of flying insects</strong> represent a key functional characteristic in ecosystem functioning as food for insectivores, and as a response to habitat quality and management. <strong>Species richness of flying insects</strong> represents the total breadth of the insect community, and <strong>Species richness of pollinating insects</strong> complements the butterfly and bumblebee indicators in taxonomic breadth and spatial resolution.
<strong>Species richness of dung associated insects</strong> represent a key functional group in decomposition, and function as an indicator of grazing intensity and diversity. <strong>Quotent of Apocrita vs Symphyta species richness</strong> tries to go beyond simple amounts, and capture compositional changes in the insect community. Apocrita covers predatory and parasitic wasps, while Symphyta is plant-feeding wasps. A relative decrease in predatory/parasitic wasps could indicate a degradation in insect prey communities, and a decrease in the potential for natural pest control. This indicator is optionally given as just the species richness of Apocrita. <strong>Number of alien species</strong> takes into account the number of alien species found in a region and optionally, the frequency of sampling events they are observed in. This is based on the alien species list from the Norwegian Biodiversity Information Center, which lists known alien species and an increasing list of door-knocker species. Note that due to large historical knowledge gaps of insect species distributions, there could potentially be many more alien species in Norway, which are not picked up on this alien species list. Lastly, <strong>Intraspecific genetic variation</strong> captures biodiversity that until now has been hidden, due to lack of practical technologies to measure it. The need to conserve genetic diversity is identified in Norwegian law, and decreases in intraspecific diversity could signal population decreases, and a decreased potential to adapt to environmental changes, before whole species are lost to an area.</p>
</div>
<div id="about-the-underlying-data-4" class="section level2" number="10.2">
<h2>
<span class="header-section-number">10.2</span> About the underlying data<a class="anchor" aria-label="anchor" href="#about-the-underlying-data-4"><i class="fas fa-link"></i></a>
</h2>
<p>The data used here comes from the ongoing monitoring programs “Norwegian insect monitoring” (<a href="https://www.nina.no/Naturmangfold/Insekter/Overv%C3%A5king-av-insekter">NorIns</a>) and “Norwegian monitoring of bumblebees and butterflies” (<a href="https://www.nina.no/V%C3%A5re-fagomr%C3%A5der/Milj%C3%B8overv%C3%A5king-p%C3%A5-land/Humler-og-dagsommerfugler">NMBB</a>). Both programs are initiated and financed by the Environmental agency, with the aim to produce continuous area representative time-series of insect community data. The <strong>NorIns</strong> program use malaise and window traps to sample a broad community of insect species, which are identified through metabarcoding of DNA. The <strong>NMBB</strong> program use sweep netting along transect walks where the species are identified manually in the field. More information is available in the links above.</p>
<div id="representativity-in-time-and-space-5" class="section level3" number="10.2.1">
<h3>
<span class="header-section-number">10.2.1</span> Representativity in time and space<a class="anchor" aria-label="anchor" href="#representativity-in-time-and-space-5"><i class="fas fa-link"></i></a>
</h3>
<p>The <strong>NorIns</strong> program started in 2020 with monitoring in semi-natural land and forests in the south-eastern region of Norway (Østlandet), with the long-term ambition to cover the entire country. In 2021, the program was expanded to semi-natural land in central Norway (Trøndelag), in 2022 to semi-natural land in south-western Norway (Sørlandet), and in 2023 to semi-natural land in northern Norway (Nord-Norge), with a possible expansion to western Norway (Vestlandet) in 2024. So far, forest habitats are only monitored in south-eastern Norway.</p>
<p>The monitoring program is designed to produce estimates with a spatial resolution of these 5 large regions. Finer resolution could be evaluated in the future, although it is unlikely to be feasible without more intense sampling. The program uses a staggered sampling scheme where we visit different locations each year, and return to each location every 5 years. This sampling scheme, coupled with the strong yearly variation in insect abundances makes it reasonable to use a temporal resolution of 5 years. The indicator values from <strong>NorIns</strong> are therefore calculated on 5 year long rolling windows.</p>
<p>The <strong>NMBB</strong> program started in 2009 in a subset of south-eastern Norway (counties of Vestfold and Østfold), and was expanded to central Norway (Trøndelag) in 2010, south-western Norway in 2013, and to western Norway (Vestland and Møre og Romsdal fylke) in 2022. The spatial resolution aligns with these (as of now) 4 regions, and it is not recommended to downscale the predictions due to the limited number of sampling locations in each region. Since the same locations are visited each year, yearly estimates of the condition could be calculated, but should be interpreted by caution due to natural yearly variability.</p>
<p>Both programs are designed to be area representative, within their respective habitat types and regions. Both use stratified randomized selection of localities. The <strong>NorIns</strong> program selects localities by random draws within habitat strata, and the <strong>NMBB</strong> by a predefined continuous network of survey squares, within which individual transects are then manually placed. In both programs, some subjective decisions are made in the definition of habitat criteria and in discarding unpractical survey locations.</p>
</div>
<div id="original-units-4" class="section level3" number="10.2.2">
<h3>
<span class="header-section-number">10.2.2</span> Original units<a class="anchor" aria-label="anchor" href="#original-units-4"><i class="fas fa-link"></i></a>
</h3>
<p>The original units from <strong>NorIns</strong> include number of species, total biomass as wet weight, and inter-specific genetic diversity (unit to be determined). For the <strong>NMBB</strong> program, original units are number of individuals of each species within a transect, although only presences in transects (as presence frequencies) are used for the indicator calculations.</p>
</div>
<div id="temporal-coverage-5" class="section level3" number="10.2.3">
<h3>
<span class="header-section-number">10.2.3</span> Temporal coverage<a class="anchor" aria-label="anchor" href="#temporal-coverage-5"><i class="fas fa-link"></i></a>
</h3>
<p>2009-present for the <strong>NMBB</strong> data, and 2020-present for the <strong>NorIns</strong> data.</p>
</div>
</div>
<div id="ecosystem-characteristic-5" class="section level2" number="10.3">
<h2>
<span class="header-section-number">10.3</span> Ecosystem characteristic<a class="anchor" aria-label="anchor" href="#ecosystem-characteristic-5"><i class="fas fa-link"></i></a>
</h2>
<div id="norwegian-standard-3" class="section level3" number="10.3.1">
<h3>
<span class="header-section-number">10.3.1</span> Norwegian standard<a class="anchor" aria-label="anchor" href="#norwegian-standard-3"><i class="fas fa-link"></i></a>
</h3>
<p>Three of the indicators are tied to the <em>Økologisk egenskap</em> (Ecological characteristic) of <strong>Funksjonelt viktige arter og biofysiske stukturer</strong> (Funktionally important species and structures), and the other six to <strong>Biologisk mangfold</strong> (Biodiversity).</p>
</div>
<div id="un-standard-2" class="section level3" number="10.3.2">
<h3>
<span class="header-section-number">10.3.2</span> UN standard<a class="anchor" aria-label="anchor" href="#un-standard-2"><i class="fas fa-link"></i></a>
</h3>
<p>Six of the indicators are tied to the UN standard (ECT-classes) <strong>B1 Compositional state characteristics</strong>, and the two remaining to <strong>B2 Structural state characteristics</strong>.</p>
</div>
</div>
<div id="collinearities-with-other-indicators-8" class="section level2" number="10.4">
<h2>
<span class="header-section-number">10.4</span> Collinearities with other indicators<a class="anchor" aria-label="anchor" href="#collinearities-with-other-indicators-8"><i class="fas fa-link"></i></a>
</h2>
<p>None measured so far, but it is reasonable to assume that some of these indicators will share some causal factors, and therefore correlate to some degree. They might also correlate to other indicators as well, not described here.</p>
</div>
<div id="reference-condition-and-values-4" class="section level2" number="10.5">
<h2>
<span class="header-section-number">10.5</span> Reference condition and values<a class="anchor" aria-label="anchor" href="#reference-condition-and-values-4"><i class="fas fa-link"></i></a>
</h2>
<p>The reference condition is meant to be one with minimal negative human impact, but this has a special interpretation for semi-natural land. These habitats are formed and maintained through human activities. Thus, “minimal negative human impact” is replaced by a state of “good maintenance”, here understood as resembling the traditional agricultural maintenance regime, existing for several hundred years up until the late 1800s. This is characterized by extensive grazing, meadows and lays, crop rotation with legumes to bind nitrogen in addition to fertilization from manure, relatively small field sizes and abundant field margins, and a lack of artificial fertilization and mechanized tilling.</p>
<div id="reference-condition-4" class="section level3" number="10.5.1">
<h3>
<span class="header-section-number">10.5.1</span> Reference condition<a class="anchor" aria-label="anchor" href="#reference-condition-4"><i class="fas fa-link"></i></a>
</h3>
<p>Assessing the current state of insect communities is a complicated task, due to their taxonomic and functional breadth and high temporal and spatial variability. Still, even more challenging is establishing reference values, when the reference states are extinct or prohibitively difficult to measure empirically. We currently lack reference values for all indicators from the <strong>NorIns</strong> program, except for the alien species indicators where the reference value is zero by definition. It is unlikely that this can be solved by “simply” surveying a state in a reference area, since intact reference areas of sufficient size and numbers likely no longer exist. The question of how to handle these reference values is currently unresolved, and we present these indicator value calculations with made up reference values as placeholders in the code.</p>
<p>For the <strong>NMBB</strong> indicators, the communities are small enough and well-known enough to identify reference communities by expert opinion. These communities specify the expected rate of observing each species. The current rates of observation are then compared to the expected rates. The calculations are further explained below.</p>
</div>
<div id="reference-values-thresholds-for-defining-good-ecological-condition-minimum-andor-maximum-values-4" class="section level3" number="10.5.2">
<h3>
<span class="header-section-number">10.5.2</span> Reference values, thresholds for defining <em>good ecological condition</em>, minimum and/or maximum values<a class="anchor" aria-label="anchor" href="#reference-values-thresholds-for-defining-good-ecological-condition-minimum-andor-maximum-values-4"><i class="fas fa-link"></i></a>
</h3>
<!-- Text here -->
<p>The specific reference communities for the <strong>NMBB</strong> indicators won’t be spelled out here, see instead the scripts below. We have not specified a value for <em>good ecological condition</em> as of yet, since we lack a straightforward empirical basis to do so. By default then, the values is set to 0.6 , following the general framework.</p>
</div>
</div>
<div id="uncertainties-6" class="section level2" number="10.6">
<h2>
<span class="header-section-number">10.6</span> Uncertainties<a class="anchor" aria-label="anchor" href="#uncertainties-6"><i class="fas fa-link"></i></a>
</h2>
<p>The uncertainties for the indicator values is calculated by bootstrapping the values for each locality within a year. This takes the variability between the localities into account. For the <strong>NorIns</strong> indicators, these uncertainties are given as standard deviation, as well as confidence intervals. The uncertainties for the <strong>NMBB</strong> indicators are constituted by a distribution of discrete values, and therefore we only provide confidence intervals and not a standard deviation.</p>
</div>
<div id="references-7" class="section level2" number="10.7">
<h2>
<span class="header-section-number">10.7</span> References<a class="anchor" aria-label="anchor" href="#references-7"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li><p>French, C. M., Bertola, L. D., Carnaval, A. C., Economo, E. P., Kass, J. M., Lohman, D. J., … Hickerson, M. J. 2023, juli 17. Global determinants of insect mitochondrial genetic diversity. bioRxiv. bioRxiv. <a href="https://doi.org/10.1101/2022.02.09.479762" class="uri">https://doi.org/10.1101/2022.02.09.479762</a></p></li>
<li><p>Miraldo, A., Li, S., Borregaard, M. K., Flórez-Rodríguez, A., Gopalakrishnan, S., Rizvanovic, M., … Nogués-Bravo, D. 2016. An Anthropocene map of genetic diversity. Science 353(6307): 1532–1535. <a href="https://doi.org/10.1126/science.aaf4381" class="uri">https://doi.org/10.1126/science.aaf4381</a></p></li>
<li><p>Rohde, K. 1992. Latitudinal Gradients in Species Diversity: The Search for the Primary Cause. Oikos 65(3): 514. <a href="https://doi.org/10.2307/3545569" class="uri">https://doi.org/10.2307/3545569</a></p></li>
<li><p>Theodoridis, S., Fordham, D. A., Brown, S. C., Li, S., Rahbek, C., &amp; Nogues-Bravo, D. 2020. Evolutionary history and past climate change shape the distribution of genetic diversity in terrestrial mammals. Nature Communications 11(1): 2557. <a href="https://doi.org/10.1038/s41467-020-16449-5" class="uri">https://doi.org/10.1038/s41467-020-16449-5</a></p></li>
<li><p>Åström, Jens; Birkemoe, Tone; Brandsegg, Hege; Dahle, Sondre; Davey, Marie Louise; Ekrem, Torbjørn; Fossøy, Frode; Hanssen, Oddvar; Laugsand, Arne Endre; Majaneva, Markus; Staverløkk, Arnstein; Sverdrup-Thygeson, Anne; Ødegaard, Frode.
Insektovervåking på Østlandet, Sørlandet og i Trøndelag. Rapport fra feltsesong 2022. Trondheim: Norsk institutt for naturforskning (NINA) 2023 (ISBN 978‐82‐426‐5037‐5) 99 s. NINA rapport(2241)</p></li>
<li><p>Åström, Sandra Charlotte Helene; Åström, Jens; Bøhn, Kristoffer; Gjershaug, Jan Ove; Staverløkk, Arnstein; Dahle, Sondre; Ødegaard, Frode. Nasjonal overvåking av dagsommerfugler og humler i Norge. Oppsummering av aktiviteten i 2022. Trondheim: Norsk institutt for naturforskning (NINA) 2023 (ISBN 978-82-426-5009-2) 54 s. NINA rapport(2214)</p></li>
<li><p>Öberg, S., Gjershaug, J. O., Diserud, O. &amp; Ødegaard, F. 2011. Videreutvikling av metodikk for
arealrepresentativ overvåking av dagsommerfugler og humler. Naturindeks for Norge. – NINA
Rapport 663.</p></li>
</ul>
</div>
<div id="analyses-2" class="section level2" number="10.8">
<h2>
<span class="header-section-number">10.8</span> Analyses<a class="anchor" aria-label="anchor" href="#analyses-2"><i class="fas fa-link"></i></a>
</h2>
<div id="calculation-principles-for-norins-indicators" class="section level3" number="10.8.1">
<h3>
<span class="header-section-number">10.8.1</span> Calculation principles for NorIns indicators<a class="anchor" aria-label="anchor" href="#calculation-principles-for-norins-indicators"><i class="fas fa-link"></i></a>
</h3>
<p>The indicators from the Norwegian insect monitoring are calculated with functions in the <code>Norimon</code> R-package <a href="https://github.com/NINAnor/Norimon">link</a>. This methodology is meant to facilitate the calculation of a broad variety of insect indicators using data from <strong>NorIns</strong>.</p>
<p><strong>Main steps of workflow:</strong></p>
<ol style="list-style-type: decimal">
<li>Fetch data of biomass or community diversity from a centralized database. Diversity data can be filtered on several taxonomic levels, but biomass is only available for whole samples.</li>
<li>Bootstrap observations and calculate mean and uncertainty of point estimates.</li>
<li>Compare observations to reference points.</li>
<li>Display and plot the results</li>
</ol>
<div class="inline-figure"><img src="images/NorIns_flow_chart.png" width="400pt"></div>
<p>Most of the ecology comes into step 1, in choosing what data to use to describe a quality. This can be a custom selection of species, or a set of higher taxa such as specific genuses, families or even order. The goal is to choose a set of taxa that represent specific qualities of the community that can indicate the ecological status of the ecosystem. In addition to the selection of taxa, we have to decide on the level of spatial and temporal aggregation, e.g. if we should aggregate the data spatially on the region scale, habitat scale or locality scale, and if we should aggregate it temporally on yearly or even the individual sample occasions within years. For the ecological indicators, we will aggregate the raw observation values to the year locality scale, where the data summarize all catches in a locality in a year.</p>
<p>The second step is to get an estimate of the uncertainty of the data, were we use the bootstrap method. This method is also flexible, as we could bootstrap the samples on different sampling levels (e.g. samples within localities, localities within a region, or regions within country). When working with raw data on the year locality level, the most granular bootstrap would be to summarize the variables to the yearly, habitat, and regional level, for example expressing the mean values of insect biomass in semi-natural land and forests in Østlandet in 2021, with uncertainty.</p>
<p>Ecological knowledge also comes into step 3, comparing the values to a reference state. Here we have several options. We could for example use a single defined value as the reference state. But we can also use a point in time as a reference point (e.g. the start of a time series).</p>
<p>Taken together, this framework is meant to facilitate the calculation of an arbitrary set of insect indicators, based on the combination of choices in data to fetch, aggregation level, and reference comparison. We showcase the framework by working through an example with pollinators.</p>
<p>In the example we start with connecting to the database, in order to fetch the data.</p>
<div class="sourceCode" id="cb307"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">connect_to_insect_db</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>For convenience, the pollinator families can be retrieved by the <code>get_pollinators()</code> function.</p>
<div class="sourceCode" id="cb308"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinators</span> <span class="op">&lt;-</span> <span class="fu">get_pollinators</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pollinators_fam</span> <span class="op">&lt;-</span> <span class="va">pollinators</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">family_latin</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th align="left">family_norwegian</th>
<th align="left">family_latin</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Gravebier</td>
<td align="left">Andrenidae</td>
</tr>
<tr class="even">
<td align="left">Langtungbebier</td>
<td align="left">Apidae</td>
</tr>
<tr class="odd">
<td align="left">Korttungebier</td>
<td align="left">Colletidae</td>
</tr>
<tr class="even">
<td align="left">Markbier</td>
<td align="left">Halictidae</td>
</tr>
<tr class="odd">
<td align="left">Buksamlerbier</td>
<td align="left">Megachilidae</td>
</tr>
<tr class="even">
<td align="left">Blomsterbier</td>
<td align="left">Melittidae</td>
</tr>
<tr class="odd">
<td align="left">Blomfluer</td>
<td align="left">Syrphidae</td>
</tr>
<tr class="even">
<td align="left">Smygere</td>
<td align="left">Hesperiidae</td>
</tr>
<tr class="odd">
<td align="left">Glansvinger</td>
<td align="left">Lycaenidae</td>
</tr>
<tr class="even">
<td align="left">Nymfevinger</td>
<td align="left">Nymphalidae</td>
</tr>
<tr class="odd">
<td align="left">Svalestjerter</td>
<td align="left">Papilionidae</td>
</tr>
<tr class="even">
<td align="left">Hvitvinger</td>
<td align="left">Pieridae</td>
</tr>
<tr class="odd">
<td align="left">Metallmerker (uoffisiell)</td>
<td align="left">Riodinidae</td>
</tr>
</tbody>
</table></div>
<p><br>
We then fetch the community data for these families through the <code>get_observations</code> function. We additionally subset the habitat type that we are interested in, here semi-natural land. The result is a tibble of the aggregated number of species, shannon diversity, and the mean number of genetic variants per species. The default aggregation level is “year_locality”, meaning the total observations for a locality within a year.</p>
<p>Note that the output currently contains some experimental values, awaiting a more robust methodology. “Shannon diversity” doesn’t really make sense without counts or amounts, etc. This will be replaced by measurements of genetic diversity that are based on published peer-reviewed methodology (French et al. 2023), in the (hopefully near) future.</p>
<div class="sourceCode" id="cb309"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">poll_loc_year</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>subset_families <span class="op">=</span> <span class="va">pollinators_fam</span>,</span>
<span>                                  subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb310"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poll_loc_year</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">slice</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span><span class="fu">kableExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html">kbl</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-table"><table class="table table-sm">
<thead><tr>
<th style="text-align:right;">
year
</th>
<th style="text-align:left;">
locality
</th>
<th style="text-align:left;">
habitat_type
</th>
<th style="text-align:left;">
region_name
</th>
<th style="text-align:right;">
no_species
</th>
<th style="text-align:right;">
shannon_div
</th>
<th style="text-align:right;">
mean_no_asv_per_species
</th>
<th style="text-align:right;">
GDE_by_asv
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_01
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
49.47
</td>
<td style="text-align:right;">
2.68
</td>
<td style="text-align:right;">
0.50
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_02
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
28.64
</td>
<td style="text-align:right;">
1.44
</td>
<td style="text-align:right;">
0.84
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_03
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
40
</td>
<td style="text-align:right;">
34.07
</td>
<td style="text-align:right;">
1.57
</td>
<td style="text-align:right;">
0.85
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_04
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:right;">
42.90
</td>
<td style="text-align:right;">
1.89
</td>
<td style="text-align:right;">
0.77
</td>
</tr>
<tr>
<td style="text-align:right;">
2020
</td>
<td style="text-align:left;">
Semi-nat_05
</td>
<td style="text-align:left;">
Semi-nat
</td>
<td style="text-align:left;">
Østlandet
</td>
<td style="text-align:right;">
78
</td>
<td style="text-align:right;">
52.19
</td>
<td style="text-align:right;">
1.90
</td>
<td style="text-align:right;">
0.67
</td>
</tr>
</tbody>
</table></div>
<div id="bootstrap-observations" class="section level4" number="10.8.1.1">
<h4>
<span class="header-section-number">10.8.1.1</span> Bootstrap observations<a class="anchor" aria-label="anchor" href="#bootstrap-observations"><i class="fas fa-link"></i></a>
</h4>
<p>The localities in the NorIns program are (semi-)randomly selected and can be viewed as samples of a larger population. Individually, they represent a single measurement of an insect community in a habitat type in a region. To get the sampling uncertainty for this representation, we can bootstrap the values by this simple process: we choose a random set of localities within a year and region (with replacement) and calculate the average value. We then repeat this process a large number of times to get a bootstrap sample of values, which can be used to express the uncertainty in the dataset.</p>
<p>We typically have 10 localities for a given habitat type and region each year, which is not very much to base our uncertainty estimates on. But bootstrapping works fairly well on small samples and has the advantage that is doesn’t make assumptions of the statistical distribution of the errors. We will use this level to showcase the functionality and to visualize some variation between years. However, since the sampling scheme of the monitoring program semi-randomly selects 10 new localities for each of the 5 years the rotating scheme, each years estimate will include some randomness that is caused by variability between localities. This means that the values between individual years might appear more random than the underlying overall trend. To get robust estimates that averages over this variability, we will summarize the actual indicator values to 5 year periods, further described below.</p>
<p>When <strong>NorIns</strong> is fully scaled up, it will have 50 localities for each covered habitat type in every country region, with 10 localities every year, in a rotating survey scheme over 5 years. Since the program is rolled out sequentially starting in 2020, for semi-natural land it currently covers 4/5 of the country, with varying amounts of localities within each region. The map below displays the localities surveyed so far.</p>
<div class="sourceCode" id="cb311"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">semi_loc</span> <span class="op">&lt;-</span> <span class="fu">get_localities</span><span class="op">(</span>dataset <span class="op">=</span> <span class="st">"NasIns"</span>,</span>
<span>                           habitat_type <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb312"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">norway_regions</span> <span class="op">&lt;-</span> <span class="fu">get_map</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb313"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tm_shape</span><span class="op">(</span><span class="va">norway_regions</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">tm_polygons</span><span class="op">(</span><span class="st">"region"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">tm_shape</span><span class="op">(</span><span class="va">semi_loc</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">tm_symbols</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>             shape <span class="op">=</span> <span class="fl">22</span>,</span>
<span>             size <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/location_map-1.png" width="672"></div>
<p>The bootstrap routine is implemented in the function <code>bootstrap_value()</code>, which takes a community (from <code>get_observation</code>) or weight dataset (from <code>get_biomass</code>) as its first input. It also needs to know what measurement in the dataset to bootstrap, and what, if any, grouping structure to aggregate the results on. In this example, we bootstrap the number of pollinator species, and aggregate the results on the year and regional scale.</p>
<div class="sourceCode" id="cb314"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">poll_richness_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span><span class="va">poll_loc_year</span>,</span>
<span>                                      value <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                                      groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                 <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                      rolling_year_window <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>                                      <span class="op">)</span></span></code></pre></div>
<p>This creates an object of type <code>boot_stat</code>. Calling it prints a simple summary of the bootstrap values.</p>
<div class="sourceCode" id="cb315"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poll_richness_boot</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Østlandet         59.9    6.00          48.3</span></span>
<span><span class="co">#&gt; 2  2021 Trøndelag         41.8    3.11          34.8</span></span>
<span><span class="co">#&gt; 3  2021 Østlandet         53.3    6.09          41  </span></span>
<span><span class="co">#&gt; 4  2022 Sørlandet         29.6    5.90          18.4</span></span>
<span><span class="co">#&gt; 5  2022 Trøndelag         34.8    2.57          29.8</span></span>
<span><span class="co">#&gt; 6  2022 Østlandet         35.9    2.85          30.2</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<p>But the <code>boot_stat</code> object also stores the individual bootstrap values for later computation. By default, we use 999 bootstrap samples, which here results in 999 samples * 6 groups = 5994 rows of bootstrap values.</p>
<div class="sourceCode" id="cb316"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poll_richness_boot</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="co">#&gt; $bootstrap_values</span></span>
<span><span class="co">#&gt; # A tibble: 5,994 × 3</span></span>
<span><span class="co">#&gt; # Groups:   year, region_name [6]</span></span>
<span><span class="co">#&gt;     year region_name boot_values</span></span>
<span><span class="co">#&gt;    &lt;int&gt; &lt;chr&gt;             &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1  2020 Østlandet          63.5</span></span>
<span><span class="co">#&gt;  2  2020 Østlandet          50.5</span></span>
<span><span class="co">#&gt;  3  2020 Østlandet          49.9</span></span>
<span><span class="co">#&gt;  4  2020 Østlandet          65.8</span></span>
<span><span class="co">#&gt;  5  2020 Østlandet          62.2</span></span>
<span><span class="co">#&gt;  6  2020 Østlandet          53.6</span></span>
<span><span class="co">#&gt;  7  2020 Østlandet          61.6</span></span>
<span><span class="co">#&gt;  8  2020 Østlandet          62.7</span></span>
<span><span class="co">#&gt;  9  2020 Østlandet          68.8</span></span>
<span><span class="co">#&gt; 10  2020 Østlandet          64.3</span></span>
<span><span class="co">#&gt; # ℹ 5,984 more rows</span></span></code></pre></div>
</div>
<div id="comparing-bootstrap-values-to-a-reference-point" class="section level4" number="10.8.1.2">
<h4>
<span class="header-section-number">10.8.1.2</span> Comparing bootstrap values to a reference point<a class="anchor" aria-label="anchor" href="#comparing-bootstrap-values-to-a-reference-point"><i class="fas fa-link"></i></a>
</h4>
<p>Setting aside the practical difficulties in establishing robust reference values, the next step in the methodology is to compare the observed values (with bootstrapped uncertainty) to a chosen reference value. This can be made in several ways. Most simply, if we have a single numeric value as a reference value, we can simply subtract that from the observed values. For example, if we observe 13 species of pollinators in 2022 at a location, and the reference point is 10, the 2022 value has increased by 13 - 10 = 3 species. Such subtractions should be made on the set of bootstrapped values, followed by new summary statistics being calculated, preserving the uncertainty from the bootstrap. The <code>boot_stat</code> object class has its own subtraction method <code>-</code> do to just that. In this example, we set the reference value arbitrarily to 30 species.</p>
<div class="sourceCode" id="cb317"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poll_richness_boot</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Østlandet         59.9    6.00          48.3</span></span>
<span><span class="co">#&gt; 2  2021 Trøndelag         41.8    3.11          34.8</span></span>
<span><span class="co">#&gt; 3  2021 Østlandet         53.3    6.09          41  </span></span>
<span><span class="co">#&gt; 4  2022 Sørlandet         29.6    5.90          18.4</span></span>
<span><span class="co">#&gt; 5  2022 Trøndelag         34.8    2.57          29.8</span></span>
<span><span class="co">#&gt; 6  2022 Østlandet         35.9    2.85          30.2</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb318"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_poll_richness_boot</span> <span class="op">&lt;-</span> <span class="va">poll_richness_boot</span> <span class="op">-</span> <span class="fl">30</span></span></code></pre></div>
<div class="sourceCode" id="cb319"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_poll_richness_boot</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Østlandet       29.9      6.00        18.3  </span></span>
<span><span class="co">#&gt; 2  2021 Trøndelag       11.8      3.11         4.8  </span></span>
<span><span class="co">#&gt; 3  2021 Østlandet       23.3      6.09        11    </span></span>
<span><span class="co">#&gt; 4  2022 Sørlandet       -0.412    5.90       -11.6  </span></span>
<span><span class="co">#&gt; 5  2022 Trøndelag        4.76     2.57        -0.200</span></span>
<span><span class="co">#&gt; 6  2022 Østlandet        5.90     2.85         0.200</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<p>Alternatively, we could use a reference point in the time series itself. Say for example that we want to use the values for species richness of pollinators in semi-natural land i Østlandet 2020 as a reference point. We can then calculate the difference (the contrast) between this level and all the other levels. We do this by the function <code>boot_contrast()</code></p>
<p><strong>NB! This functionality is in development. It currently works for single rows as reference points, but needs updating to allow for referencing several values simultaneously, e.g using the start values for all regions and habitat types as their own reference points.</strong></p>
<div class="sourceCode" id="cb320"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">diff_poll_richness_boot2</span> <span class="op">&lt;-</span> <span class="va">poll_richness_boot</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">boot_contrast</span><span class="op">(</span><span class="va">year</span> <span class="op">==</span> <span class="fl">2020</span> <span class="op">&amp;</span> <span class="va">region_name</span> <span class="op">==</span> <span class="st">'Østlandet'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb321"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_poll_richness_boot2</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Østlandet         0       0              0  </span></span>
<span><span class="co">#&gt; 2  2021 Trøndelag       -18.1     6.76         -31.6</span></span>
<span><span class="co">#&gt; 3  2021 Østlandet        -6.59    8.42         -23.3</span></span>
<span><span class="co">#&gt; 4  2022 Sørlandet       -30.3     8.42         -46.2</span></span>
<span><span class="co">#&gt; 5  2022 Trøndelag       -25.1     6.42         -37.4</span></span>
<span><span class="co">#&gt; 6  2022 Østlandet       -24.0     6.67         -37.2</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
</div>
<div id="normalizing-the-values" class="section level4" number="10.8.1.3">
<h4>
<span class="header-section-number">10.8.1.3</span> Normalizing the values<a class="anchor" aria-label="anchor" href="#normalizing-the-values"><i class="fas fa-link"></i></a>
</h4>
<p>After the comparison to a reference value, we need to normalize the indicator values so that they lie between 0 and 1. This can be done in several ways (se e.g. <a href="https://github.com/NINAnor/eaTools">eaTools</a>). The simplest case is to use a linear scaling, with a natural zero, which e.g. can be done by dividing the indicator values by the highest value state. A <code>boot_stat</code> class has a <code>/</code> function that divides each bootstrap value by a given value, truncates the highest values to 1, and recalculates the summary values. We can similarly divide by a predefined reference state. Here, we exemplify the method by dividing the value by 30.</p>
<div class="sourceCode" id="cb322"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_poll_richness_boot3</span> <span class="op">&lt;-</span> <span class="va">poll_richness_boot</span> <span class="op">/</span> <span class="fl">30</span></span></code></pre></div>
<div class="sourceCode" id="cb323"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_poll_richness_boot3</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Østlandet        1     0               1    </span></span>
<span><span class="co">#&gt; 2  2021 Trøndelag        1.00  0.00190         1    </span></span>
<span><span class="co">#&gt; 3  2021 Østlandet        1     0               1    </span></span>
<span><span class="co">#&gt; 4  2022 Sørlandet        0.913 0.116           0.613</span></span>
<span><span class="co">#&gt; 5  2022 Trøndelag        0.999 0.00651         0.993</span></span>
<span><span class="co">#&gt; 6  2022 Østlandet        0.999 0.00485         1    </span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
</div>
<div id="display-and-plot-bootstrap-values" class="section level4" number="10.8.1.4">
<h4>
<span class="header-section-number">10.8.1.4</span> Display and plot bootstrap values<a class="anchor" aria-label="anchor" href="#display-and-plot-bootstrap-values"><i class="fas fa-link"></i></a>
</h4>
<p>The boot_stat class also has its own plot function. It tries to plot a comparison of the bootstrap distributions over years, for each group. For example, if we plot the object <code>diff_poll_richness_boot</code>, we can look at the yearly differences in beetles species richness in the two geographic regions: (Note that we have only 1 year of data from Sørlandet so far)</p>
<div class="sourceCode" id="cb324"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">diff_poll_richness_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/poll_richness_boot-1.png" width="672"></div>
<p>In the cases where we have used a single row as a reference point, this shows up as a sharp spike at 0.</p>
<div class="sourceCode" id="cb325"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">diff_poll_richness_boot2</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/diff_beetle_richness_boot2-1.png" width="672"></div>
</div>
<div id="time-series-plots" class="section level4" number="10.8.1.5">
<h4>
<span class="header-section-number">10.8.1.5</span> Time-series plots<a class="anchor" aria-label="anchor" href="#time-series-plots"><i class="fas fa-link"></i></a>
</h4>
<p>The default plots are good for showing the uncertainty distributions, but can be a bit difficult to track the time series trends this way. We can instad use the <code>ts_plot</code> function for a simple time-series plot.</p>
<div class="sourceCode" id="cb326"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ts_plot</span><span class="op">(</span><span class="va">poll_richness_boot</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The `&lt;scale&gt;` argument of `guides()` cannot be `FALSE`. Use</span></span>
<span><span class="co">#&gt; "none" instead as of ggplot2 3.3.4.</span></span>
<span><span class="co">#&gt; ℹ The deprecated feature was likely used in the Norimon</span></span>
<span><span class="co">#&gt;   package.</span></span>
<span><span class="co">#&gt;   Please report the issue to the authors.</span></span>
<span><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where</span></span>
<span><span class="co">#&gt; this warning was generated.</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-22-1.png" width="672"></div>
</div>
<div id="map-plots" class="section level4" number="10.8.1.6">
<h4>
<span class="header-section-number">10.8.1.6</span> Map plots<a class="anchor" aria-label="anchor" href="#map-plots"><i class="fas fa-link"></i></a>
</h4>
<p>In addition to these plots, we can also display the values geographically. The <code>map_plot()</code> function takes a <code>boot_stat</code> object and plots the values according to its region names.</p>
<div class="sourceCode" id="cb327"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">map_plot</span><span class="op">(</span><span class="va">poll_richness_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/map_plot_cutout-1.png" width="672"></div>
<p>By default, it only shows the regions with data, but this can be overridden manually:</p>
<div class="sourceCode" id="cb328"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">map_plot</span><span class="op">(</span><span class="va">poll_richness_boot</span>,</span>
<span>         whole_country <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure">
<img src="insectIndicators_files/figure-html/map_plot_whole-1.png" width="672">
We can also choose a different palette, for example from the NinaR package, and visualize the uncertainty by setting the transparency of the colors from the bootstrap standard deviations. Most of these functions can be piped as well:</div>
<div class="sourceCode" id="cb329"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff_poll_richness_boot</span> <span class="op">%&gt;%</span></span>
<span><span class="fu">map_plot</span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"orange-green"</span>,</span>
<span>         whole_country <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>         alpha_from_sd <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/diff_beetle_richness_boot_map-1.png" width="672"></div>
<p>More plotting options are available, see the <code>Norimon</code> indicator workflow vignette.</p>
</div>
<div id="year-rolling-window-summaries" class="section level4" number="10.8.1.7">
<h4>
<span class="header-section-number">10.8.1.7</span> 5 year rolling window summaries<a class="anchor" aria-label="anchor" href="#year-rolling-window-summaries"><i class="fas fa-link"></i></a>
</h4>
<p>As described above, to take the staggered 5 year survey scheme into account and to average over random yearly variation, it makes sense to summarize the state over a whole 5 year period. This does not need to adhere to predefined periods, since any 5 year long period will result in a complete data set. We can therefore calculate 5 year long summaries in rolling windows, where each focal year is surrounded by +- 2 years. The bootstrap_value does this by default, if we don’t specify <code>rolling_year_average = FALSE</code></p>
<div class="sourceCode" id="cb330"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poll_richness_boot_5_year</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span><span class="va">poll_loc_year</span>,</span>
<span>                                      value <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                                      groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                 <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                      rolling_year_window <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>                                      <span class="op">)</span></span></code></pre></div>
<p>Since we are only on the third year in the survey scheme however, the rolling windows are too large to show any differences between years. This will change, starting with the 2023 year data.</p>
<div class="sourceCode" id="cb331"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">poll_richness_boot_5_year</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-24-1.png" width="672"></div>
<p>The averaging over 5-years also shows up in the time-series plots. Note the very wide confidence intervals of the Sørlandet region, with only 1-year worth of data.</p>
<div class="sourceCode" id="cb332"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ts_plot</span><span class="op">(</span><span class="va">poll_richness_boot_5_year</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-25-1.png" width="672"></div>
<p>This concludes the tour of the <code>Norimon</code> functionality.</p>
</div>
</div>
<div id="calculation-principles-for-nbbm-indicators" class="section level3" number="10.8.2">
<h3>
<span class="header-section-number">10.8.2</span> Calculation principles for NBBM indicators<a class="anchor" aria-label="anchor" href="#calculation-principles-for-nbbm-indicators"><i class="fas fa-link"></i></a>
</h3>
<p>These indicators were developed for the Nature Index of Norway, back in 2010 by Ola Diserud and Sandra Öberg (Öberg et. al 2011), and are calculated routinely on a yearly basis (see Åström et. al 2022 for the latest report).</p>
<p>These indicators summarize the state of a bumblebee or butterfly community by estimating the difference from a reference community. Formally, the community indicator (CI) is expressed as the relative change of the community from a state of reference (SR), were the change is calculated by a state of change (SC).</p>
<p><span class="math display">\[CI = \frac{SR-SC}{SR}\]</span>
The state of reference (SR) represents a community of species which can be expected to be observed in a given habitat type and region. It is calculated by assigning each species to class of expected commonality: common (C), sporadic (S) and rare (R) species. This classification is done by expert opinion, informed by known present and past species distributions. Note that this state of reference only contains a subset of highlighted species, making up a historical reference community. Potential observations of “new” species therefore don’t inform the indicator value at all. Common species are expected to be observed in at least 5 % of the surveyed transects in a habitat type and region. Sporadic species similarly are defined as having a presence below 5 % but above 1 %. Rare species are seen in no more than 1 % of the transects, and lastly, species not seen in any transect are assumed as lost (L) for the purpose of this calculation.</p>
<p>Each commonality class gets its own weight, so that common species inform the reference state more than sporadic species, followed by rare species. The state of reference (SR) is thus a value where each species adds to the state according to its commonality. Formally, it is defined as:</p>
<p><span class="math display">\[SR = n_C * w_{C, SR} + n_S * w_{S, SR} + n_R * w_{R, SR} = \sum_{i = (C, S, R)} n_i * w_{i, SR}, \]</span>
where <span class="math inline">\(n_i\)</span> is the number of species in a commonality class (C, S, R), and the weights <span class="math inline">\([w_{C, SR}, w_{S, SR}, w_{R, SR}]\)</span> specify their respective contribution to the reference state (SR). The weights used are <span class="math inline">\([w_{C, SR}, w_{S, SR}, w_{R, SR}] = [1.0, 0.75, 0.50]\)</span>, i.e. a sporadic species has 75 % the weight of a common species, and a rare species has 50 % the weight of a common species.</p>
<p>The state of change (SC) is calculated as
<span class="math display">\[SC = n_{CS} * w_{CS} + n_{CR} * w_{CR} + n_{CL} * w_{CL} + n_{SR} * w_{SR} + n_{SL} * w_{SL} + n_{RL} * w_{RL}, \]</span>
where <span class="math inline">\(n_{CS}\)</span> is the number of expected common species (C) that is observed sporadically (S), and <span class="math inline">\(w_{CS}\)</span> is the weight of this change. Similarly, <span class="math inline">\(n_{CR}\)</span> is the number of expected common species that is observed rarely, <span class="math inline">\(n_{RL}\)</span> is the number of rare species that is lost, and so on. Potential increases in observation rates do not inform the indicator. Changes for common species are weighted more heavily than changes for less common species, and larger decreases are weighted more heavily than smaller decreases, such that <span class="math inline">\([w_{CS}, w_{CR}, w_{CL}, w_{SR}, w_{SL}, w_{RL}]= [0.50, 0.75, 1.00, 0.50, 0.75, 0.50]\)</span>.</p>
<p>The indicator values are estimated with uncertainty, by bootstrapping the set of localities within a region, and calculating the indicator for each of 9999 such samples.</p>
<div id="calculation-example" class="section level4" number="10.8.2.1">
<h4>
<span class="header-section-number">10.8.2.1</span> Calculation example<a class="anchor" aria-label="anchor" href="#calculation-example"><i class="fas fa-link"></i></a>
</h4>
<p>These calculations with bootstrap sampling are implemented in the R-package <code>bombLepiSurv</code> (<a href="https://github.com/jenast/bombLepiSurv">Bombus and Lepidoptera Survey</a>). We briefly show this functionality with bumblebees in semi-natural lands as an example.</p>
<p>We connect to the internal database with a convenience function.</p>
<div class="sourceCode" id="cb333"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">bombLepiSurv</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/humlesommerfConnect.html">humlesommerfConnect</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>We can fetch all observations of bumblebees in semi-natural land with the <code>getAllData</code> function. Species names are here shown in Norwegian.</p>
<div class="sourceCode" id="cb334"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">allBombusGrassland2022</span> <span class="op">&lt;-</span> <span class="fu">getAllData</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"bumblebees"</span>,</span>
<span>                                     habitat <span class="op">=</span> <span class="st">"gressmark"</span>,</span>
<span>                                     year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                                     language <span class="op">=</span> <span class="st">"norsk"</span><span class="op">)</span></span></code></pre></div>
<p>A summary of the records is most easily plotted by <code>surveyBarPlot</code>:</p>
<div class="sourceCode" id="cb335"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">surveyBarplot</span><span class="op">(</span><span class="va">allBombusGrassland2022</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-29-1.png" width="672"></div>
<p>Note that for calculating the indicator values, we only consider the transects that were surveyed for the standard 3 times per year, to cover the phenology of the entire season without bias. Data of these complete survey rounds can be fetched from the database by the function <code>getComplData</code>, which fetches data from one habitat type and region at a time.</p>
<div class="sourceCode" id="cb336"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bombus_trond_2022</span> <span class="op">&lt;-</span> <span class="fu">getComplData</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                                  region_short <span class="op">=</span> <span class="st">"Trond"</span>,</span>
<span>                                  habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                                  year <span class="op">=</span> <span class="fl">2022</span><span class="op">)</span></span></code></pre></div>
<p>The observed relative to the expected occurrences can be visualized through the <code>plotArt</code> function. This requires a reference community, which can be fetched through the <code>getExpValues</code> function.</p>
<div class="sourceCode" id="cb337"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_bombus_trond</span> <span class="op">&lt;-</span> <span class="fu">getExpValues</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                                 region_short <span class="op">=</span> <span class="st">"Trond"</span>,</span>
<span>                                 habitat <span class="op">=</span> <span class="st">"Gressmark"</span><span class="op">)</span></span></code></pre></div>
<p>Species in green bars are species that are expected to be common, who need to reach the green areas in the plot not to decrease the indicator value (from 1). Similarly with the blue (sporadic species), and red (rare species). We see in this case that <em>Bombus soroeensis</em> is expected to be a sporadic species, but that it was only observed rarely.</p>
<div class="sourceCode" id="cb338"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plotArt</span><span class="op">(</span><span class="va">bombus_trond_2022</span>,</span>
<span>        <span class="va">exp_bombus_trond</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-32-1.png" width="672"></div>
<p>The actual calculation and bootstrapping of the indicator is done in the function <code>calcInd</code>. This function calls the <code>getComplData</code> and <code>getExpectedValues</code> functions internally. It also requires the weights for the commonality classes in the reference community (SR), and the weights for the changes in state (CS). This is fetched internally via the functions <code>getAmountWeights</code> and <code>getDiffWeights</code>, respectively. Here the classes are coded in Norwegian (v = common, m = sporadic, s = rare).</p>
<div class="sourceCode" id="cb339"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getDiffWeights</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;      v    m   s</span></span>
<span><span class="co">#&gt; v 0.00 0.00 0.0</span></span>
<span><span class="co">#&gt; m 0.50 0.00 0.0</span></span>
<span><span class="co">#&gt; s 0.75 0.50 0.0</span></span>
<span><span class="co">#&gt; t 1.00 0.75 0.5</span></span></code></pre></div>
<div class="sourceCode" id="cb340"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">getAmountWeights</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   v    m   s</span></span>
<span><span class="co">#&gt; 1 1 0.75 0.5</span></span></code></pre></div>
<p>Lastly, we specify the number of samples for the bootstrap. To speed up, only 999 samples are used here.</p>
<div class="sourceCode" id="cb341"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nIter</span> <span class="op">=</span> <span class="fl">999</span></span></code></pre></div>
<div class="sourceCode" id="cb342"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hInd2022TrondGress</span> <span class="op">&lt;-</span> <span class="fu">calcInd</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Trond"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p>The result is an object of class “comm_index” (community index). It comes with some (still rudimentary) print and plotting functions, showing the point estimate and the limits of a 95% confidence interval. Due to the limited number of species and the fixed “weights” of each species and state of change, the indicator calculation returns a distribution of discrete values. Therefore it can happen that the 95% and the 90% confidence intervals are the same, as in this case.</p>
<div class="sourceCode" id="cb343"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hInd2022TrondGress</span></span>
<span><span class="co">#&gt; $`Community indicator estimates, with percentiles`</span></span>
<span><span class="co">#&gt;           2.5%             5% Point estimate            95% </span></span>
<span><span class="co">#&gt;      0.5416667      0.5625000      0.7083333      0.7500000 </span></span>
<span><span class="co">#&gt;          97.5% </span></span>
<span><span class="co">#&gt;      0.7500000</span></span></code></pre></div>
<div class="sourceCode" id="cb344"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">hInd2022TrondGress</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-38-1.png" width="672"></div>
<p>Plotting functions for a series of indicator values is shown below in the indicator calculations. This concludes the tour of the <code>bombLepiSurv</code> package.</p>
</div>
</div>
<div id="data-sets-5" class="section level3" number="10.8.3">
<h3>
<span class="header-section-number">10.8.3</span> Data sets<a class="anchor" aria-label="anchor" href="#data-sets-5"><i class="fas fa-link"></i></a>
</h3>
<p>There are a few different ways to access the data required for these indicators. Both the NMBB and the NorIns project store their data in an internal database at NINA. They both also export most of their data to GBIF, but those exports need to be restructured before they can be processed in the following scripts.</p>
<div id="norins-data" class="section level4" number="10.8.3.1">
<h4>
<span class="header-section-number">10.8.3.1</span> NorIns data<a class="anchor" aria-label="anchor" href="#norins-data"><i class="fas fa-link"></i></a>
</h4>
<p>The <code>Norimon</code> package has convenience functions to fetch data from the database. This database is currently not available outside NINA, but we will implement a solution for this in the future. Either we will make the database externally available, or we will provide an alternative route to the data from the GBIF export. For now, we will fetch the data through the <code>Norimon</code> functions.</p>
</div>
<div id="nmbb-data" class="section level4" number="10.8.3.2">
<h4>
<span class="header-section-number">10.8.3.2</span> NMBB data<a class="anchor" aria-label="anchor" href="#nmbb-data"><i class="fas fa-link"></i></a>
</h4>
<p>The data for the bumblebee and butterfly indicators can similarly be accessed through the R-package <code>bombLepiSurv</code>.</p>
</div>
<div id="regions-1" class="section level4" number="10.8.3.3">
<h4>
<span class="header-section-number">10.8.3.3</span> Regions<a class="anchor" aria-label="anchor" href="#regions-1"><i class="fas fa-link"></i></a>
</h4>
<!-- In case you need to map the indicator value to regions, you can do that here. Remove this chapter if not relevant. -->
<p>We here show how to import a shape file with the regional delineation. The indicators associated with the NorIns project can be attributed to the 5 country regions of Norway. Currently however, the data program only covers 4 of the 5 regions.</p>
<div class="sourceCode" id="cb345"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">connect_to_insect_db</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">norway_regions</span> <span class="op">&lt;-</span> <span class="fu">Norimon</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Norimon/man/get_map.html">get_map</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb346"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">norway_reg_NorIns</span> <span class="op">&lt;-</span> <span class="va">norway_regions</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region</span> <span class="op">!=</span> <span class="st">"Vestlandet"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">region</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">region</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>geom <span class="op">=</span> <span class="fu">st_union</span><span class="op">(</span><span class="va">geom</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb347"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tm_shape</span><span class="op">(</span><span class="va">norway_reg_NorIns</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">tm_polygons</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span> </span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-42-1.png" width="672"></div>
<p>The indicators from the NMBB program are similarly connected to regions, with the exception that the south-east region only covers the old counties of Vestfold and Østfold, and that Nord-Norge isn’t covered yet.</p>
<div class="sourceCode" id="cb348"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">bombLepiSurv</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/humlesommerfConnect.html">humlesommerfConnect</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb349"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nbbm_norway_regions</span> <span class="op">&lt;-</span> <span class="fu">bombLepiSurv</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/get_map.html">get_map</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb350"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">norway_reg_NBBM</span> <span class="op">&lt;-</span> <span class="va">nbbm_norway_regions</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Øst"</span>, <span class="st">"Sørlandet"</span>, <span class="st">"Trøndelag"</span>, <span class="st">"Vestlandet"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">region</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">region</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>geom <span class="op">=</span> <span class="fu">st_union</span><span class="op">(</span><span class="va">geom</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb351"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tm_shape</span><span class="op">(</span><span class="va">norway_reg_NBBM</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">tm_polygons</span><span class="op">(</span>col <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span> </span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-47-1.png" width="672"></div>
</div>
</div>
<div id="calculation-of-norins-indicators" class="section level3" number="10.8.4">
<h3>
<span class="header-section-number">10.8.4</span> Calculation of NorIns indicators<a class="anchor" aria-label="anchor" href="#calculation-of-norins-indicators"><i class="fas fa-link"></i></a>
</h3>
<p>Here we calculate the indicators in abbreviated form, following the general framework outlined above.</p>
<div id="biomass-of-flying-insects-in-semi-natural-land" class="section level4" number="10.8.4.1">
<h4>
<span class="header-section-number">10.8.4.1</span> Biomass of flying insects in semi-natural land<a class="anchor" aria-label="anchor" href="#biomass-of-flying-insects-in-semi-natural-land"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb352"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">connect_to_insect_db</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb353"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biomass_sn</span> <span class="op">&lt;-</span> <span class="fu">get_biomass</span><span class="op">(</span>subset_year <span class="op">=</span> <span class="fl">2020</span><span class="op">:</span><span class="fl">2022</span>,</span>
<span>                          subset_region <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                          subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb354"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biomass_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span>df <span class="op">=</span> <span class="va">biomass_sn</span>,</span>
<span>                                   value <span class="op">=</span> <span class="va">avg_wet_weight</span>,</span>
<span>                                   groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                              <span class="st">"region_name"</span><span class="op">)</span></span>
<span>                                   <span class="op">)</span></span></code></pre></div>
<p>Compare values to a reference state. The reference state is here set uniformly for all regions. Alternatively, we could calculate the indicator values for each region separately, compare to individual reference states, and then put it all together again.</p>
<p>To exemplify, we here use a single “made-up” value as a reference state and normalize the values at the same time, using the <code>/</code> function.</p>
<div class="sourceCode" id="cb355"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biomass_ref</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span>  </span>
<span><span class="va">biomass_sn_diff</span> <span class="op">&lt;-</span> <span class="va">biomass_sn_boot</span> <span class="op">/</span> <span class="va">biomass_ref</span></span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb356"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biomass_sn_diff</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name avg_wet_weight boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;                &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet            0.469  0.0640         0.346</span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag            0.748  0.0734         0.591</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet            0.757  0.0528         0.652</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet            0.467  0.0630         0.350</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag            0.746  0.0695         0.610</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet            0.757  0.0521         0.658</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet            0.467  0.0644         0.346</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag            0.747  0.0717         0.618</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet            0.757  0.0521         0.653</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb357"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">biomass_sn_diff</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/biomass_diff-1.png" width="672"></div>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb358"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">biomass_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">biomass_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">biomass_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">biomass_sn_to_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">biomass_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">biomass_sn_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_biomass_semi_nat</span> <span class="op">&lt;-</span> <span class="va">biomass_sn_to_exp</span></span></code></pre></div>
</div>
<div id="species-richness-of-flying-insects-in-semi-natural-land" class="section level4" number="10.8.4.2">
<h4>
<span class="header-section-number">10.8.4.2</span> Species richness of flying insects in semi-natural land<a class="anchor" aria-label="anchor" href="#species-richness-of-flying-insects-in-semi-natural-land"><i class="fas fa-link"></i></a>
</h4>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb359"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_sn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>subset_year <span class="op">=</span> <span class="fl">2020</span><span class="op">:</span><span class="fl">2022</span>,</span>
<span>                                subset_region <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                                subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb360"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span>df <span class="op">=</span> <span class="va">richness_sn</span>,</span>
<span>                                   value <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                                   groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                              <span class="st">"region_name"</span><span class="op">)</span></span>
<span>                                   <span class="op">)</span></span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb361"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_ref</span> <span class="op">&lt;-</span> <span class="fl">4000</span></span>
<span>  </span>
<span><span class="va">richness_sn_diff</span> <span class="op">&lt;-</span> <span class="va">richness_sn_boot</span> <span class="op">/</span> <span class="va">richness_ref</span></span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb362"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_sn_diff</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet        0.545  0.0647         0.423</span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag        0.664  0.0275         0.606</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet        0.706  0.0357         0.637</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet        0.548  0.0630         0.428</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag        0.663  0.0274         0.607</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet        0.704  0.0334         0.637</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet        0.548  0.0638         0.425</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag        0.664  0.0275         0.607</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet        0.704  0.0366         0.633</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb363"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">richness_sn_diff</span>,</span>
<span>     limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-59-1.png" width="672"></div>
<div class="sourceCode" id="cb364"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">map_plot</span><span class="op">(</span><span class="va">richness_sn_diff</span>,</span>
<span>         whole_country <span class="op">=</span> <span class="cn">T</span>,</span>
<span>         limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/richness-1.png" width="672"></div>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb365"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">richness_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">richness_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">richness_sn_to_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">richness_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">richness_sn_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_richness_semi_nat</span> <span class="op">&lt;-</span> <span class="va">richness_sn_to_exp</span></span></code></pre></div>
</div>
<div id="species-richness-of-pollinators-in-semi-natural-land" class="section level4" number="10.8.4.3">
<h4>
<span class="header-section-number">10.8.4.3</span> Species richness of pollinators in semi-natural land<a class="anchor" aria-label="anchor" href="#species-richness-of-pollinators-in-semi-natural-land"><i class="fas fa-link"></i></a>
</h4>
<p>Fetch the data.</p>
<div class="sourceCode" id="cb366"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinator_fam</span> <span class="op">&lt;-</span> <span class="fu">get_pollinators</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">family_latin</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span>
<span>  </span></code></pre></div>
<div class="sourceCode" id="cb367"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinators_sn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>subset_year <span class="op">=</span> <span class="fl">2020</span><span class="op">:</span><span class="fl">2022</span>,</span>
<span>                                   subset_families <span class="op">=</span> <span class="va">pollinator_fam</span>,</span>
<span>                                   subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb368"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinators_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span>df <span class="op">=</span> <span class="va">pollinators_sn</span>,</span>
<span>                                       value <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                                       groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                  <span class="st">"region_name"</span><span class="op">)</span></span>
<span>                                       <span class="op">)</span></span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb369"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinators_ref</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span>  </span>
<span><span class="va">pollinators_sn_diff</span> <span class="op">&lt;-</span> <span class="va">pollinators_sn_boot</span> <span class="op">/</span> <span class="va">pollinators_ref</span></span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb370"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinators_sn_diff</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet        0.593  0.115          0.38 </span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag        0.765  0.0440         0.677</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet        0.971  0.0409         0.867</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet        0.593  0.115          0.374</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag        0.767  0.0436         0.681</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet        0.971  0.0410         0.864</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet        0.589  0.114          0.366</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag        0.765  0.0449         0.674</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet        0.972  0.0413         0.852</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb371"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">pollinators_sn_diff</span>,</span>
<span>     limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-66-1.png" width="672"></div>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb372"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pollinators_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">pollinators_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">pollinators_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">pollinators_sn_to_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">pollinators_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">pollinators_sn_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_pollinators_semi_nat</span> <span class="op">&lt;-</span> <span class="va">pollinators_sn_to_exp</span></span></code></pre></div>
</div>
<div id="species-richness-of-dung-associated-insects" class="section level4" number="10.8.4.4">
<h4>
<span class="header-section-number">10.8.4.4</span> Species richness of dung associated insects<a class="anchor" aria-label="anchor" href="#species-richness-of-dung-associated-insects"><i class="fas fa-link"></i></a>
</h4>
<p>We here note all observations of a set of species, known to be associated with dung. This species list can be expanded after a further taxonomic review. Here we take all observed species in the house fly/stable fly family Muscidae, together with a selection of 70 beetle species known to be associated with dung.</p>
<div class="sourceCode" id="cb373"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">muscidae_species</span> <span class="op">&lt;-</span> <span class="fu">tbl</span><span class="op">(</span><span class="va">con</span>,</span>
<span>                    <span class="fu">Id</span><span class="op">(</span>schema <span class="op">=</span> <span class="st">"views"</span>,</span>
<span>                       table <span class="op">=</span> <span class="st">"observed_muscidae_species"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">coleoptera_dung_species</span> <span class="op">&lt;-</span> <span class="fu">tbl</span><span class="op">(</span><span class="va">con</span>,</span>
<span>                    <span class="fu">Id</span><span class="op">(</span>schema <span class="op">=</span> <span class="st">"lookup"</span>,</span>
<span>                       table <span class="op">=</span> <span class="st">"coleoptera_dung_species"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">species_latin</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dung_species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">coleoptera_dung_species</span>, <span class="va">muscidae_species</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb374"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dung_sn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>subset_year <span class="op">=</span> <span class="fl">2020</span><span class="op">:</span><span class="fl">2022</span>,</span>
<span>                            subset_species <span class="op">=</span> <span class="va">dung_species</span>,</span>
<span>                            subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<p>Calculate the indicator values for each region and year.</p>
<div class="sourceCode" id="cb375"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dung_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span>df <span class="op">=</span> <span class="va">dung_sn</span>,</span>
<span>                                       value <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                                       groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                  <span class="st">"region_name"</span><span class="op">)</span></span>
<span>                                       <span class="op">)</span></span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb376"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dung_ref</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span>  </span>
<span><span class="va">dung_sn_diff</span> <span class="op">&lt;-</span> <span class="va">dung_sn_boot</span> <span class="op">/</span> <span class="va">dung_ref</span></span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb377"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dung_sn_diff</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name no_species boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;            &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet        0.608  0.0613         0.493</span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag        0.832  0.0317         0.764</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet        0.762  0.0324         0.697</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet        0.605  0.0633         0.492</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag        0.833  0.0329         0.758</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet        0.763  0.0331         0.697</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet        0.608  0.0618         0.494</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag        0.832  0.0327         0.759</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet        0.760  0.0332         0.694</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb378"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">dung_sn_diff</span>,</span>
<span>     limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-73-1.png" width="672"></div>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb379"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dung_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">dung_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">dung_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">dung_sn_to_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">dung_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">dung_sn_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_dung_assoc_semi_nat</span> <span class="op">&lt;-</span> <span class="va">dung_sn_to_exp</span></span></code></pre></div>
</div>
<div id="relationship-between-symphyta-and-parasitica-in-semi-natural-land" class="section level4" number="10.8.4.5">
<h4>
<span class="header-section-number">10.8.4.5</span> Relationship between Symphyta and Parasitica in semi-natural land<a class="anchor" aria-label="anchor" href="#relationship-between-symphyta-and-parasitica-in-semi-natural-land"><i class="fas fa-link"></i></a>
</h4>
<p>Parasitica and Symphyta wasps are here defined as a set of families.</p>
<div class="sourceCode" id="cb380"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">symphyta_fam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Argidae"</span>, <span class="st">"Cephidae"</span>, <span class="st">"Cimbicidae"</span>, </span>
<span>                  <span class="st">"Diprionidae"</span>, <span class="st">"Orussidae"</span>, <span class="st">"Pamphiliidae"</span>, </span>
<span>                  <span class="st">"Pergidae"</span>, <span class="st">"Siricidae"</span>, <span class="st">"Anaxyelidae"</span>,</span>
<span>                  <span class="st">"Tenthredinidae"</span>, <span class="st">"Xiphydriidae"</span>, <span class="st">"Xyelidae"</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">symphyta_sn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>subset_year <span class="op">=</span> <span class="fl">2020</span><span class="op">:</span><span class="fl">2022</span>,</span>
<span>                                subset_families <span class="op">=</span> <span class="va">symphyta_fam</span>,</span>
<span>                                subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">symphyta_sn</span></span>
<span><span class="co">#&gt; # A tibble: 60 × 8</span></span>
<span><span class="co">#&gt;     year locality    habitat_type region_name no_species</span></span>
<span><span class="co">#&gt;    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;        &lt;chr&gt;            &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1  2020 Semi-nat_01 Semi-nat     Østlandet           38</span></span>
<span><span class="co">#&gt;  2  2020 Semi-nat_02 Semi-nat     Østlandet           16</span></span>
<span><span class="co">#&gt;  3  2020 Semi-nat_03 Semi-nat     Østlandet           13</span></span>
<span><span class="co">#&gt;  4  2020 Semi-nat_04 Semi-nat     Østlandet           39</span></span>
<span><span class="co">#&gt;  5  2020 Semi-nat_05 Semi-nat     Østlandet           65</span></span>
<span><span class="co">#&gt;  6  2020 Semi-nat_06 Semi-nat     Østlandet           22</span></span>
<span><span class="co">#&gt;  7  2020 Semi-nat_07 Semi-nat     Østlandet           36</span></span>
<span><span class="co">#&gt;  8  2020 Semi-nat_08 Semi-nat     Østlandet           46</span></span>
<span><span class="co">#&gt;  9  2020 Semi-nat_09 Semi-nat     Østlandet           21</span></span>
<span><span class="co">#&gt; 10  2020 Semi-nat_10 Semi-nat     Østlandet           68</span></span>
<span><span class="co">#&gt; # ℹ 50 more rows</span></span>
<span><span class="co">#&gt; # ℹ 3 more variables: shannon_div &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   mean_no_asv_per_species &lt;dbl&gt;, GDE_by_asv &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb381"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parasitica_fam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Braconidae"</span>, <span class="st">"Ichneumonidae"</span>, <span class="st">"Chalcididae"</span>, </span>
<span>                    <span class="st">"Eulophidae"</span>, <span class="st">"Pteromalidae"</span>, <span class="st">"Aphelinidae"</span>, </span>
<span>                    <span class="st">"Scelionidae"</span>, <span class="st">"Eupelmidae"</span>, <span class="st">"Encyrtidae"</span>, </span>
<span>                    <span class="st">"Mymaridae"</span>, <span class="st">"Diapriidae"</span>, <span class="st">"Bethylidae"</span>, </span>
<span>                    <span class="st">"Evaniidae"</span>, <span class="st">"Ceraphronidae"</span>, <span class="st">"Torymidae"</span>, </span>
<span>                    <span class="st">"Dryinidae"</span>, <span class="st">"Eucharitidae"</span>, <span class="st">"Mymarommatidae"</span>, </span>
<span>                    <span class="st">"Orussidae"</span>, <span class="st">"Megaspilidae"</span>, <span class="st">"Stephanidae"</span>, </span>
<span>                    <span class="st">"Trigonalidae"</span>, <span class="st">"Platygastridae"</span>, <span class="st">"Aulacidae"</span>, </span>
<span>                    <span class="st">"Gasteruptiidae"</span>, <span class="st">"Rhopalosomatidae"</span>, <span class="st">"Larridae"</span>, </span>
<span>                    <span class="st">"Agaonidae"</span>, <span class="st">"Pompilidae"</span>, <span class="st">"Bradynobaenidae"</span></span>
<span>                    <span class="op">)</span></span>
<span></span>
<span><span class="va">parasitica_sn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>subset_year <span class="op">=</span> <span class="fl">2020</span><span class="op">:</span><span class="fl">2022</span>,</span>
<span>                                subset_families <span class="op">=</span> <span class="va">parasitica_fam</span>,</span>
<span>                                subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">parasitica_sn</span></span>
<span><span class="co">#&gt; # A tibble: 60 × 8</span></span>
<span><span class="co">#&gt;     year locality    habitat_type region_name no_species</span></span>
<span><span class="co">#&gt;    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;        &lt;chr&gt;            &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1  2020 Semi-nat_01 Semi-nat     Østlandet          480</span></span>
<span><span class="co">#&gt;  2  2020 Semi-nat_02 Semi-nat     Østlandet          112</span></span>
<span><span class="co">#&gt;  3  2020 Semi-nat_03 Semi-nat     Østlandet          119</span></span>
<span><span class="co">#&gt;  4  2020 Semi-nat_04 Semi-nat     Østlandet          369</span></span>
<span><span class="co">#&gt;  5  2020 Semi-nat_05 Semi-nat     Østlandet          642</span></span>
<span><span class="co">#&gt;  6  2020 Semi-nat_06 Semi-nat     Østlandet          190</span></span>
<span><span class="co">#&gt;  7  2020 Semi-nat_07 Semi-nat     Østlandet          253</span></span>
<span><span class="co">#&gt;  8  2020 Semi-nat_08 Semi-nat     Østlandet          323</span></span>
<span><span class="co">#&gt;  9  2020 Semi-nat_09 Semi-nat     Østlandet          216</span></span>
<span><span class="co">#&gt; 10  2020 Semi-nat_10 Semi-nat     Østlandet          531</span></span>
<span><span class="co">#&gt; # ℹ 50 more rows</span></span>
<span><span class="co">#&gt; # ℹ 3 more variables: shannon_div &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   mean_no_asv_per_species &lt;dbl&gt;, GDE_by_asv &lt;dbl&gt;</span></span></code></pre></div>
<p>Divide the richness of Parasitica by Symphyta. [Could make a function for this later on.]</p>
<div class="sourceCode" id="cb382"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sym</span> <span class="op">&lt;-</span> <span class="va">symphyta_sn</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">full_join</span><span class="op">(</span><span class="va">parasitica_sn</span>,</span>
<span>       by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span> <span class="op">=</span><span class="st">"year"</span>,</span>
<span>              <span class="st">"locality"</span> <span class="op">=</span> <span class="st">"locality"</span>,</span>
<span>              <span class="st">"habitat_type"</span> <span class="op">=</span> <span class="st">"habitat_type"</span>,</span>
<span>              <span class="st">"region_name"</span> <span class="op">=</span> <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>       suffix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"_sym"</span>, <span class="st">"_par"</span><span class="op">)</span></span>
<span>       <span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>par_per_sym_richn <span class="op">=</span> <span class="va">no_species_par</span> <span class="op">/</span> <span class="va">no_species_sym</span><span class="op">)</span></span></code></pre></div>
<p>Bootstrap the fraction of Parasitica to Symphyta.</p>
<div class="sourceCode" id="cb383"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sym_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span><span class="va">par_sym</span>,</span>
<span>                                value <span class="op">=</span> <span class="va">par_per_sym_richn</span>,</span>
<span>                                groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>, </span>
<span>                                           <span class="st">"region_name"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb384"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sym_ref</span> <span class="op">&lt;-</span> <span class="fl">40</span></span>
<span>  </span>
<span><span class="va">par_sym_sn_diff</span> <span class="op">&lt;-</span> <span class="va">par_sym_sn_boot</span> <span class="op">/</span> <span class="va">par_sym_ref</span></span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb385"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sym_sn_diff</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name par_per_sym_richn boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;                   &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet               0.817  0.170          0.528</span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag               0.732  0.0769         0.591</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet               0.446  0.0415         0.369</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet               0.818  0.166          0.523</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag               0.730  0.0802         0.591</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet               0.447  0.0400         0.371</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet               0.824  0.166          0.534</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag               0.737  0.0776         0.596</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet               0.451  0.0431         0.371</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb386"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">par_sym_sn_diff</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-81-1.png" width="672"></div>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb387"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sym_sn_exp</span> <span class="op">&lt;-</span> <span class="va">par_sym_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">par_sym_sn_exp</span> <span class="op">&lt;-</span> <span class="va">par_sym_sn_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">par_sym_sn_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">par_sym_sn_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_par_sym_semi_nat</span> <span class="op">&lt;-</span> <span class="va">par_sym_sn_exp</span></span></code></pre></div>
<p><strong>Also calculate and export only parasitica values</strong>
Could also do similar with symphyta, if needed.</p>
<div class="sourceCode" id="cb388"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span><span class="va">par_sym</span>,</span>
<span>                                value <span class="op">=</span> <span class="va">no_species_par</span>,</span>
<span>                                groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>, </span>
<span>                                           <span class="st">"region_name"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb389"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">par_sn_exp</span> <span class="op">&lt;-</span> <span class="va">par_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">par_sn_exp</span> <span class="op">&lt;-</span> <span class="va">par_sn_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">par_sn_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">par_sn_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_par_semi_nat</span> <span class="op">&lt;-</span> <span class="va">par_sn_exp</span></span></code></pre></div>
</div>
<div id="number-of-alien-species-in-semi-natural-land" class="section level4" number="10.8.4.6">
<h4>
<span class="header-section-number">10.8.4.6</span> Number of alien species in semi-natural land<a class="anchor" aria-label="anchor" href="#number-of-alien-species-in-semi-natural-land"><i class="fas fa-link"></i></a>
</h4>
<p>Here we measure the prevalence of alien species in the trap catches. We present two options for calculating the indicator, as it is not clear at this point which of them is superior.</p>
<p><strong>The simplest</strong> approach is to just sum up the number of alien species present in a region, i.e. the species richness of alien species. This can be related to the total species richness of the community, and expressed as the proportion of alien species in the community. The reference state of alien species is zero by definition, since we don’t want any alien species. The value for good ecological condition could be set at a low (but arbitrary) proportion of alien species in the community. This is done for the indicator of alien plant species, where the value for good ecological condition is set to 0.05 (&lt; 5 % of the plant community should be alien). For a hyper-diverse community such as insects, this is probably too high a value. In a community of say 20 000 insect species, 5 % would mean 1 000 alien species, which is hard to see as a good condition. It is not clear to us at present what level to set here, but we exemplify the calculations with a good condition set to 0.1 % (&lt; 20 species in a 20 000 species community). It is also likely that the species richness of the entire community will increase, perhaps substantially, with increased sampling. It is not clear if the number of detected native and alien species will increase at the same rate, so it may be that the proportion of alien species will change, perhaps substantially, with increased sampling effort. It may be difficult therefore to interpret this frequency of alien insect species.</p>
<p>In addition, this approach will not take into account the spread of the species, a measure that is especially relevant for rare but increasing species. A species found only once will count as much as a species found in every locality. <strong>To take the prevalence of the species into account</strong>, we can measure the proportion of localities each alien species is observed in. These proportions could then be summed together for at total score of alien species. Here, a species found in all localities will count much more than a species only found once.</p>
<p>For each alien species, the observation frequency is calculated as the number of localities the species is observed in, divided by the total amount of localities in a geographical region and time period. <span class="math display">\[p_i = \frac{1}{N} \sum_{i=1, j=1}^{N}{obs_{ij}},\]</span> where <span class="math inline">\(obs_ij\)</span> is an observation of species <span class="math inline">\(i\)</span> in locality <span class="math inline">\(j\)</span>, and <span class="math inline">\(N\)</span> is the total number of possible observations (localities in the region and time period).
These proportions are combined into a common score for all alien species simply by summing them together; <span class="math display">\[alien~score =\sum_{i=1}^{S}{~p_{i}},\]</span> where <span class="math inline">\(p_i\)</span> is the observation frequency of species <span class="math inline">\(i\)</span>, and <span class="math inline">\(S\)</span> is the set of all observed alien species. If all alien species are present in all localities, the indicator value simply becomes <span class="math inline">\(S\)</span>. If the species is not present in all localities, it’s influence on the indicator is weighted down linearly with it presence frequency. The uncertainty of both types of indicators is calculated by bootstrapping the set of localities within a temporal and spatial scope, similarly to the other indicators.</p>
<p><strong>Fetch the species list.</strong></p>
<p>The alien species observations are stored in a materialized view for convenience. This is simply a list of observations of the alien species of categories LO, PH, HI, or SE, together with the year localities and locality samplings they were observed in. For the purposes of this calculation, we will only consider observations aggregated to the year localities level (and not the prevalence of alien species within locality and season). So the first thing to do is to fetch this list, and aggregate it to the year locality level. This summary is based on the 2023 of the alien species evaluation, and we only consider species that are identified by metabarcoding with identification confidence “Moderate” or “High” (not Low or Poor). In addition, we filter out the forest sites.</p>
<div class="sourceCode" id="cb390"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_obs</span> <span class="op">&lt;-</span> <span class="fu">tbl</span><span class="op">(</span><span class="va">con</span>,</span>
<span>                 <span class="fu">Id</span><span class="op">(</span>schema <span class="op">=</span> <span class="st">"views"</span>,</span>
<span>                    table <span class="op">=</span> <span class="st">"alien_obs"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">identification_confidence</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MODERATE"</span>, <span class="st">"HIGH"</span><span class="op">)</span>,</span>
<span>         <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Semi-nat"</span>, <span class="va">locality</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb391"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">no_alien_obs_ls</span> <span class="op">&lt;-</span> <span class="va">alien_obs</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">ls_id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">no_alien_obs_yl</span> <span class="op">&lt;-</span> <span class="va">alien_obs</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">yl_id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">no_alien_spec</span> <span class="op">&lt;-</span> <span class="va">alien_obs</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">species_latin_fixed</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">no_alien_cat</span> <span class="op">&lt;-</span> <span class="va">alien_obs</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html">as.integer</a></span><span class="op">(</span><span class="fu">n_distinct</span><span class="op">(</span><span class="va">risikokategori_2023</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>We can have quick look at some statistics of the data. We see that the project thus far has identified 15 alien species, representing 3 risk-categories, and that the observations area spread across (all) 60 semi-natural localities.</p>
<p>To continue, we aggregate the presences of alien species to locality years, meaning recording in how many localities each alien species were detected at least once.</p>
<div class="sourceCode" id="cb392"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_obs_yl</span> <span class="op">&lt;-</span> <span class="va">alien_obs</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">group_by</span><span class="op">(</span><span class="va">yl_id</span>,</span>
<span>           <span class="va">locality</span>,</span>
<span>           <span class="va">year</span>,</span>
<span>           <span class="va">region_name</span>,</span>
<span>           <span class="va">risikokategori_2023</span>,</span>
<span>           <span class="va">species_latin_fixed</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">summarize</span><span class="op">(</span>present <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">present</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span>,</span>
<span>            na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">arrange</span><span class="op">(</span><span class="va">locality</span>, <span class="va">species_latin_fixed</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">collect</span><span class="op">(</span>na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Missing values are always removed in SQL aggregation</span></span>
<span><span class="co">#&gt; functions.</span></span>
<span><span class="co">#&gt; Use `na.rm = TRUE` to silence this warning</span></span>
<span><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span></code></pre></div>
<div class="sourceCode" id="cb393"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_obs_yl</span></span>
<span><span class="co">#&gt; # A tibble: 900 × 8</span></span>
<span><span class="co">#&gt;    yl_id      locality  year region_name risikokategori_2023</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;              </span></span>
<span><span class="co">#&gt;  1 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt;  2 a3f86975-… Semi-na…  2020 Østlandet   PH                 </span></span>
<span><span class="co">#&gt;  3 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt;  4 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt;  5 a3f86975-… Semi-na…  2020 Østlandet   PH                 </span></span>
<span><span class="co">#&gt;  6 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt;  7 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt;  8 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt;  9 a3f86975-… Semi-na…  2020 Østlandet   LO                 </span></span>
<span><span class="co">#&gt; 10 a3f86975-… Semi-na…  2020 Østlandet   SE                 </span></span>
<span><span class="co">#&gt; # ℹ 890 more rows</span></span>
<span><span class="co">#&gt; # ℹ 3 more variables: species_latin_fixed &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   present &lt;lgl&gt;, na.rm &lt;lgl&gt;</span></span></code></pre></div>
<p>Next step is to bootstrap the alien observations to get the uncertainty measurement. This has to be done in a slightly different way than for the aggregated diversity or biomass values that we have used for the other indicators. Here we have a set of alien species presences for a locality, and when we bootstrap the localities, all these presences should go together, as they represent a coherent community. So we first have to take bootstrap samples of sets of localities (with replacement). Then we calculate the presence (or presence frequency) of each alien species <span class="math inline">\(p_i\)</span>, before we can summarize all of these into the indicator value <span class="math inline">\(H_{alien}\)</span>. This is implemented in a separate function <code>bootstrap_alien_obs</code> in the <code>Norimon</code> package, were we can optionally either sum the species numbers, or sum the species presence frequencies. Note that for all alien species <span class="math inline">\(i\)</span> recorded in at least one locality, we have recorded the presences and absences in all localities <span class="math inline">\(j\)</span>. This is needed for his function to work. This function is more computer intensive than the bootstrap of already aggregated values, so we employ parallelization through the <code>parallel</code> package internally.</p>
<p><strong>We first calculate the indicator based on the the number of alien species</strong></p>
<div class="sourceCode" id="cb394"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_alien_obs</span><span class="op">(</span><span class="va">alien_obs_yl</span>,</span>
<span>                                     groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                     rolling_year_window <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     only_no_spec <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb395"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">alien_sn_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-90-1.png" width="672"></div>
<p>The jagged lines in the density plot comes from the discrete values of number of species. Østlandet has the most alien species, followed by Sørlandet, and lastly Trøndelag. Trøndelag has a fewer total number of species, but similar amount of alien species in each locality, leading to a very narrow confidence band from the bootstrap.</p>
<div class="sourceCode" id="cb396"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ts_plot</span><span class="op">(</span><span class="va">alien_sn_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-91-1.png" width="672"></div>
<p>As explained above, these values could be expressed as a proportion of the the total species richness. Since the total species richness is different for the different regions, we (currently) have to do this separately for each region.</p>
<div class="sourceCode" id="cb397"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_obs_yl_sor</span> <span class="op">&lt;-</span> <span class="va">alien_obs_yl</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region_name</span> <span class="op">==</span> <span class="st">"Sørlandet"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alien_obs_yl_ost</span> <span class="op">&lt;-</span> <span class="va">alien_obs_yl</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region_name</span> <span class="op">==</span> <span class="st">"Østlandet"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alien_obs_yl_trond</span> <span class="op">&lt;-</span> <span class="va">alien_obs_yl</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region_name</span> <span class="op">==</span> <span class="st">"Trøndelag"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb398"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_sn_sor_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_alien_obs</span><span class="op">(</span><span class="va">alien_obs_yl_sor</span>,</span>
<span>                                     groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                     rolling_year_window <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     only_no_spec <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">alien_sn_ost_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_alien_obs</span><span class="op">(</span><span class="va">alien_obs_yl_ost</span>,</span>
<span>                                     groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                     rolling_year_window <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     only_no_spec <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">alien_sn_trond_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_alien_obs</span><span class="op">(</span><span class="va">alien_obs_yl_trond</span>,</span>
<span>                                     groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                     rolling_year_window <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                     only_no_spec <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Next, we fetch a record of the species richness found so far, in each region.</p>
<div class="sourceCode" id="cb399"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">region_richn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>agg_level <span class="op">=</span> <span class="st">"region_habitat"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb400"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">region_richn</span></span>
<span><span class="co">#&gt; # A tibble: 4 × 6</span></span>
<span><span class="co">#&gt;   habitat_type region_name no_species shannon_div</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;        &lt;chr&gt;            &lt;int&gt;       &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 Forest       Østlandet        13726       4924.</span></span>
<span><span class="co">#&gt; 2 Semi-nat     Sørlandet         6679       3204.</span></span>
<span><span class="co">#&gt; 3 Semi-nat     Trøndelag        10063       3359.</span></span>
<span><span class="co">#&gt; 4 Semi-nat     Østlandet        15515       5716.</span></span>
<span><span class="co">#&gt; # ℹ 2 more variables: mean_no_asv_per_species &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   GDE_by_asv &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb401"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">region_richn_sor</span> <span class="op">&lt;-</span> <span class="va">region_richn</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region_name</span> <span class="op">==</span> <span class="st">"Sørlandet"</span>,</span>
<span>         <span class="va">habitat_type</span> <span class="op">==</span> <span class="st">"Semi-nat"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">no_species</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">region_richn_ost</span> <span class="op">&lt;-</span> <span class="va">region_richn</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region_name</span> <span class="op">==</span> <span class="st">"Østlandet"</span>,</span>
<span>         <span class="va">habitat_type</span> <span class="op">==</span> <span class="st">"Semi-nat"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">no_species</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">region_richn_trond</span> <span class="op">&lt;-</span> <span class="va">region_richn</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">region_name</span> <span class="op">==</span> <span class="st">"Trøndelag"</span>,</span>
<span>         <span class="va">habitat_type</span> <span class="op">==</span> <span class="st">"Semi-nat"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">no_species</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Next, we scale each indicator to the total number of observed species.</p>
<div class="sourceCode" id="cb402"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_sn_sor_prop</span> <span class="op">&lt;-</span> <span class="va">alien_sn_sor_boot</span> <span class="op">/</span> <span class="va">region_richn_sor</span></span>
<span><span class="va">alien_sn_ost_prop</span> <span class="op">&lt;-</span> <span class="va">alien_sn_ost_boot</span> <span class="op">/</span> <span class="va">region_richn_ost</span></span>
<span><span class="va">alien_sn_trond_prop</span> <span class="op">&lt;-</span> <span class="va">alien_sn_trond_boot</span> <span class="op">/</span> <span class="va">region_richn_trond</span></span></code></pre></div>
<p>We could scale these values with a reference value of 0, and a breaking point of 0.001. eaTools has a function for that, but for this to work, we currently can only operate on the aggregate values of the indicator (not the bootstrap samples).</p>
<div class="sourceCode" id="cb403"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_sn_prop</span> <span class="op">&lt;-</span> <span class="va">alien_sn_sor_prop</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">alien_sn_ost_prop</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">alien_sn_trond_prop</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         alien_score <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         <span class="va">boot_sd</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb404"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_sn_prop_sf</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">alien_sn_prop</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb405"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_sn_prop_sf</span><span class="op">$</span><span class="va">alien_score_scaled</span> <span class="op">&lt;-</span> <span class="fu">eaTools</span><span class="fu">::</span><span class="fu"><a href="https://ninanor.github.io/eaTools/reference/ea_normalise.html">ea_normalise</a></span><span class="op">(</span><span class="va">alien_sn_prop_sf</span>, </span>
<span>                                              vector <span class="op">=</span> <span class="st">"alien_score"</span>, </span>
<span>                                              upper_reference_level <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                                              lower_reference_level <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                                              break_point <span class="op">=</span> <span class="fl">0.001</span>, </span>
<span>                                              reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                              plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>This calculation is then an sf-object that can be exported.</p>
<div class="sourceCode" id="cb406"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">alien_sn_exp</span> <span class="op">&lt;-</span> <span class="va">alien_sn_prop_sf</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         <span class="va">region</span>,</span>
<span>         v <span class="op">=</span> <span class="va">alien_score</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         i <span class="op">=</span> <span class="va">alien_score_scaled</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="va">insect_alien_prop_semi_nat</span> <span class="op">&lt;-</span> <span class="va">alien_sn_exp</span></span></code></pre></div>
<p><strong>Secondly, we calculate the indicator based on the distribution of the alien species</strong></p>
<p>We call this alien_H, since the calculation resembles Hill numbers from calculations of Shannon information.</p>
<div class="sourceCode" id="cb407"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_H_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_alien_obs</span><span class="op">(</span><span class="va">alien_obs_yl</span>,</span>
<span>                                    groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                               <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                    rolling_year_window <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                    only_no_spec <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb408"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_H_boot</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name alien_score boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;             &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet         0.904   0.242         0.4  </span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag         0.691   0.161         0.4  </span></span>
<span><span class="co">#&gt; 3  2020 Østlandet         0.929   0.153         0.633</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet         0.911   0.241         0.5  </span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag         0.692   0.160         0.4  </span></span>
<span><span class="co">#&gt; 6  2021 Østlandet         0.934   0.152         0.633</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet         0.913   0.250         0.5  </span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag         0.708   0.159         0.4  </span></span>
<span><span class="co">#&gt; 9  2022 Østlandet         0.932   0.156         0.633</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb409"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">alien_H_boot</span><span class="op">)</span></span></code></pre></div>
<p><img src="insectIndicators_files/figure-html/unnamed-chunk-104-1.png" width="672"><br>
Interstingly, by this calculation method, the different regions have much more similar values. Specifically, the southern region has comparable values to the eastern region, while Trøndelag has lower values. This indicates that region Østlandet has markedly higher number of alien species than the Southern region, but with generally more limited distribution. While in comparison, the alien species in the south are more frequently distributed throughout the sampling localities.</p>
<div class="sourceCode" id="cb410"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ts_plot</span><span class="op">(</span><span class="va">alien_H_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-105-1.png" width="672"></div>
<p>This indicator also has a natural reference value, with zero presences of alien species, meaning an alien score of 0. We could scale this indicator similarly as with the indicator based on the pure number of alien species. Similar to that case, there is no upper limit to how many alien species there can be, but we set a high enough value, and use a breaking point for the scaling, this upper end point has limited influence.</p>
<p>We don’t have enough experience to determine an appropriate breakpoint for the scaling at this point, or an upper value, but exemplify the calculation with some subjective values</p>
<div class="sourceCode" id="cb411"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">alien_H_boot_sf</span> <span class="op">&lt;-</span> <span class="va">alien_H_boot</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         alien_score <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         <span class="va">boot_sd</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alien_H_boot_sf</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">alien_H_boot_sf</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">alien_H_boot_sf</span><span class="op">$</span><span class="va">alien_score_scaled</span> <span class="op">&lt;-</span> <span class="fu">eaTools</span><span class="fu">::</span><span class="fu"><a href="https://ninanor.github.io/eaTools/reference/ea_normalise.html">ea_normalise</a></span><span class="op">(</span><span class="va">alien_H_boot_sf</span>,</span>
<span>                                                            vector <span class="op">=</span> <span class="st">"alien_score"</span>, </span>
<span>                                                            upper_reference_level <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                                                            lower_reference_level <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                                                            break_point <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                                                            reverse <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                                            plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>This is then ready for export.</p>
<div class="sourceCode" id="cb412"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alien_H_sn_exp</span> <span class="op">&lt;-</span> <span class="va">alien_H_boot_sf</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="fl">10</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         <span class="va">region</span>,</span>
<span>         v <span class="op">=</span> <span class="va">alien_score</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         i <span class="op">=</span> <span class="va">alien_score_scaled</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span></span>
<span></span>
<span><span class="va">insect_alien_H_semi_nat</span> <span class="op">&lt;-</span> <span class="va">alien_H_sn_exp</span></span></code></pre></div>
</div>
<div id="genetic-diversity-of-flying-insects-in-semi-natural-land" class="section level4" number="10.8.4.7">
<h4>
<span class="header-section-number">10.8.4.7</span> Genetic diversity of flying insects in semi-natural land<a class="anchor" aria-label="anchor" href="#genetic-diversity-of-flying-insects-in-semi-natural-land"><i class="fas fa-link"></i></a>
</h4>
<p>The genetic diversity (GD) of a species is a measurement of the differences between DNA variants within the same species, here calculated on the COI region of the mitochondrial DNA. It is a different measurement in principle than the total number of species in a community, and the number of distinct DNA variants in a community. It is an empirical question whether these measurement correlate on the global or local scale. For instance, Miraldo et al. (2016) found global patterns of decreasing GD near the poles, coinciding with global patterns of species richness. These patterns may share causes, such as higher mutation rates and shorter generation times in warmer climates (Rohde K. 1992). However, it is not clear if and how these patterns are distributed within a smaller region, such as within Norway. Spatial and temporal trends in genetic diversity may indicate pressures on species with reduced effective population sizes, before the complete loss of a species.</p>
<p>Following e.g. Miranda et al. (2016), we define GD as the average number of nucleotide differences per nucleotide site across all pairwise sequence comparisons within a species. Mathematically this can be expressed as: <span class="math display">\[\hat{\prod}_s = \frac{1}{(\frac{n}{2})} \sum_{i=1}^{n-1}\sum_{j=i+1}^{n}\frac{k_{ij}}{m_{ij}},\]</span> where <span class="math inline">\(s\)</span> is a single species, <span class="math inline">\(k_{ij}\)</span> is the number of different nucleotides between sequence <span class="math inline">\(i\)</span> and sequence <span class="math inline">\(j\)</span>, <span class="math inline">\(m_{ij}\)</span> is the number of shared base pairs between sequence <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>, and <span class="math inline">\((\frac{n}{2})\)</span> represents the number of pairwise comparisons made (Theodoridis et al 2020). Following French et al. (2023), we here considered only species that where found at least 3 separate times, to make sure we had enough samples to observe any intraspecific diversity.</p>
<p>These species-specific measurements can be combined into single community measurements in several ways. Firstly, we can simply take the average GD of a community of species (GDM), or <span class="math display">\[GDM = \frac{1}{S}\sum_{s=1}^{S}\hat{\prod}_s, \]</span> where <span class="math inline">\(S\)</span> is the total number of species, and <span class="math inline">\(s\)</span> is each species (Theodoridis et al. 2020). The mean genetic diversity of a community is a natural starting point, but can - analogous to species richness -, hide important patterns in the relative contributions from individual species. In short, the distributions of GD might be skewed, so that a few species carries the most influence on the measurement. Therefore, French et al. (2023) has developed a genetic evenness measurement (GDE), that is analogous to Shannon’s entropy for communities on the species level. This is defined as <span class="math display">\[\frac{exp(\sum_{i=0}^{N} - \hat{\prod}_i ln(\hat{\prod}_i))}{N},\]</span> where the denominator <span class="math inline">\(N\)</span> is the number of species in the community, and <span class="math inline">\(\hat{\prod}_i\)</span> is the GD calculated above for a single species. The numerator is the Hill number of the Shannon entropy. This results in a measurement that can be compared across communities of different species richness.</p>
<p>We have calculated both the GDM and GDE, but here show only the results of the GDM. The calculated GDE values showed a strong relationship with the number of species in the samples. This could indicate a possible error in the calculation routines, or a more fundamental problem, which needs to be checked before going forward with this measurement. Alternatively, the calculation is correct, but then the values are superfluous, since they give essentially the same information as the total richness, already calculated as a separate indicator.</p>
<p>Therefore, we here present only the GDM as an indicator. Note that this is a relatively new measures, and this is the first time it is calculated for large insect communities in Norway. We need further experience with these measurements to tease out potential pro’s and con’s in the Norwegian setting. The raw-data on the DNA-sequences is as of yet not publicly available, and the calculations of GDM and GDE is not shown in detail.</p>
<div class="sourceCode" id="cb413"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"data/GDM_year_locality.rds"</span><span class="op">)</span></span></code></pre></div>
<p>The <code>gdm</code> contains the mean genetic diversity for the species found in every year locality combination. This is calculated after first filtering on species that is found at least 3 times.</p>
<p>The <code>gdm</code> object contains the year_locality_ids as a key, which we can join to the relevant metadata, e.g. through the alien observation object used above.</p>
<div class="sourceCode" id="cb414"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm_sn</span> <span class="op">&lt;-</span> <span class="va">alien_obs_yl</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">yl_id</span>, <span class="va">locality</span>, <span class="va">year</span>, <span class="va">region_name</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">distinct</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">gdm</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"yl_id"</span><span class="op">=</span> <span class="st">"year_locality_id"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gdm_sn</span></span>
<span><span class="co">#&gt; # A tibble: 60 × 5</span></span>
<span><span class="co">#&gt;    yl_id                  locality  year region_name     GDM</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                  &lt;chr&gt;    &lt;int&gt; &lt;chr&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 a3f86975-b0c2-4200-98… Semi-na…  2020 Østlandet   0.00723</span></span>
<span><span class="co">#&gt;  2 11b4435f-8cc9-4ba2-9d… Semi-na…  2020 Østlandet   0.00947</span></span>
<span><span class="co">#&gt;  3 0b2956ca-3449-47b8-b9… Semi-na…  2020 Østlandet   0.00883</span></span>
<span><span class="co">#&gt;  4 f0412a3b-7864-472d-a8… Semi-na…  2020 Østlandet   0.00618</span></span>
<span><span class="co">#&gt;  5 b462682b-d037-43fd-b1… Semi-na…  2020 Østlandet   0.00973</span></span>
<span><span class="co">#&gt;  6 c76f63db-56a2-47c4-b2… Semi-na…  2020 Østlandet   0.00950</span></span>
<span><span class="co">#&gt;  7 0256d2aa-07f1-45ce-a3… Semi-na…  2020 Østlandet   0.00613</span></span>
<span><span class="co">#&gt;  8 271cb773-e404-46a7-a7… Semi-na…  2020 Østlandet   0.00999</span></span>
<span><span class="co">#&gt;  9 9768a7ca-0bc8-43bf-94… Semi-na…  2020 Østlandet   0.0111 </span></span>
<span><span class="co">#&gt; 10 6a19c4e6-8226-432c-94… Semi-na…  2020 Østlandet   0.00840</span></span>
<span><span class="co">#&gt; # ℹ 50 more rows</span></span></code></pre></div>
<p>Since these calculations are so novel, it can be worth spending some time exploring how the mean genetic diversity relates to other richness measurement at the locality level. Remember that globally, the species richness and genetic mean diversity show a general positive correlation (Miraldo et al. 2016).</p>
<div class="sourceCode" id="cb415"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_sn</span> <span class="op">&lt;-</span> <span class="fu">get_observations</span><span class="op">(</span>agg_level <span class="op">=</span> <span class="st">"year_locality"</span>,</span>
<span>                                subset_habitat <span class="op">=</span> <span class="st">"Semi-nat"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb416"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">richness_gdm_join</span> <span class="op">&lt;-</span> <span class="va">richness_sn</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">gdm_sn</span>,</span>
<span>             by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"locality"</span> <span class="op">=</span> <span class="st">"locality"</span>,</span>
<span>                    <span class="st">"year"</span> <span class="op">=</span> <span class="st">"year"</span>,</span>
<span>                    <span class="st">"region_name"</span> <span class="op">=</span> <span class="st">"region_name"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb417"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">richness_gdm_join</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                 y <span class="op">=</span> <span class="va">GDM</span>,</span>
<span>                 color <span class="op">=</span> <span class="va">region_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                 y <span class="op">=</span> <span class="va">GDM</span>,</span>
<span>                 color <span class="op">=</span> <span class="va">region_name</span><span class="op">)</span>,</span>
<span>              method <span class="op">=</span> <span class="va">lm</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">NinaR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/NinaR/man/scale_color_nina.html">scale_color_nina</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Antall arter"</span><span class="op">)</span></span>
<span><span class="co">#&gt; `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre></div>
<p><img src="insectIndicators_files/figure-html/unnamed-chunk-112-1.png" width="672"><br>
Interestingly, and perhaps reassuringly, the number of species doesn’t seem to be clearly related to the average genetic diversity. We can look at the mean number of ASVs as well.</p>
<div class="sourceCode" id="cb418"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">richness_gdm_join</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">mean_no_asv_per_species</span>,</span>
<span>                 y <span class="op">=</span> <span class="va">GDM</span>,</span>
<span>                 color <span class="op">=</span> <span class="va">region_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">mean_no_asv_per_species</span>,</span>
<span>                 y <span class="op">=</span> <span class="va">GDM</span>,</span>
<span>                 color <span class="op">=</span> <span class="va">region_name</span><span class="op">)</span>,</span>
<span>              method <span class="op">=</span> <span class="va">lm</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">NinaR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/NinaR/man/scale_color_nina.html">scale_color_nina</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Antall genetiske varianter per art"</span><span class="op">)</span></span>
<span><span class="co">#&gt; `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre></div>
<p><img src="insectIndicators_files/figure-html/unnamed-chunk-113-1.png" width="672"><br>
Here we see a positive relationship in Trøndelag (although uncertain), but not the other regions.</p>
<div class="sourceCode" id="cb419"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">richness_gdm_join</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                 y <span class="op">=</span> <span class="va">mean_no_asv_per_species</span>,</span>
<span>                 color <span class="op">=</span> <span class="va">region_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">no_species</span>,</span>
<span>                 y <span class="op">=</span> <span class="va">mean_no_asv_per_species</span>,</span>
<span>                 color <span class="op">=</span> <span class="va">region_name</span><span class="op">)</span>,</span>
<span>              method <span class="op">=</span> <span class="va">lm</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">NinaR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/NinaR/man/scale_color_nina.html">scale_color_nina</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">xlab</span><span class="op">(</span><span class="st">"Antall arter"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ylab</span><span class="op">(</span><span class="st">"Antall genetiske varianter per art"</span><span class="op">)</span></span>
<span><span class="co">#&gt; `geom_smooth()` using formula = 'y ~ x'</span></span></code></pre></div>
<p><img src="insectIndicators_files/figure-html/unnamed-chunk-114-1.png" width="672"><br>
While, for comparison, the number of ASVs (genetic variants) and the number of species per locality show more clear relationships. This was just a quick look at the genetic aspect of biodiversity, and we would require further analysis to interpret them robustly. In the meantime, we continue with the calculation example of the indicators. We bootstrap the mean genetic diversity like with the other indicators.</p>
<div class="sourceCode" id="cb420"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm_sn_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span>df <span class="op">=</span> <span class="va">gdm_sn</span>,</span>
<span>                                   value <span class="op">=</span> <span class="va">GDM</span>,</span>
<span>                                   groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                              <span class="st">"region_name"</span><span class="op">)</span></span>
<span>                                   <span class="op">)</span></span>
<span><span class="va">gdm_sn_boot</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name     GDM  boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;         &lt;dbl&gt;    &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet   0.00890 0.000601       0.00789</span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag   0.00843 0.000373       0.00769</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet   0.00882 0.000256       0.00830</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet   0.00890 0.000554       0.00796</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag   0.00842 0.000367       0.00769</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet   0.00881 0.000267       0.00827</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet   0.00891 0.000577       0.00794</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag   0.00841 0.000359       0.00771</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet   0.00883 0.000264       0.00829</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb421"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">gdm_sn_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-116-1.png" width="672"></div>
<p>So far, Trøndelag seem to have lower genetic diversity than Sørlandet and Østlandet.</p>
<div class="sourceCode" id="cb422"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ts_plot</span><span class="op">(</span><span class="va">gdm_sn_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-117-1.png" width="672"></div>
<p>Although, as with the other indicators, we see variation between years. This can be seen if we do not bootstrap on a rolling 5 year window.</p>
<div class="sourceCode" id="cb423"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm_sn_yearly_boot</span> <span class="op">&lt;-</span> <span class="fu">bootstrap_value</span><span class="op">(</span>df <span class="op">=</span> <span class="va">gdm_sn</span>,</span>
<span>                                      value <span class="op">=</span> <span class="va">GDM</span>,</span>
<span>                                      groups <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"year"</span>,</span>
<span>                                                 <span class="st">"region_name"</span><span class="op">)</span>,</span>
<span>                                      rolling_year_window <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>                                   <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb424"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ts_plot</span><span class="op">(</span><span class="va">gdm_sn_yearly_boot</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-119-1.png" width="672"></div>
<p>Compare to arbitrary reference value and normalize.</p>
<div class="sourceCode" id="cb425"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm_sn_ref</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span>  </span>
<span><span class="va">gdm_sn_diff</span> <span class="op">&lt;-</span> <span class="va">gdm_sn_boot</span> <span class="op">/</span> <span class="va">gdm_sn_ref</span></span></code></pre></div>
<p>Plot the results and display uncertainty.</p>
<div class="sourceCode" id="cb426"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm_sn_diff</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;    year region_name   GDM boot_sd boot_lower2.5</span></span>
<span><span class="co">#&gt;   &lt;int&gt; &lt;chr&gt;       &lt;dbl&gt;   &lt;dbl&gt;         &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1  2020 Sørlandet   0.445  0.0300         0.395</span></span>
<span><span class="co">#&gt; 2  2020 Trøndelag   0.422  0.0186         0.385</span></span>
<span><span class="co">#&gt; 3  2020 Østlandet   0.441  0.0128         0.415</span></span>
<span><span class="co">#&gt; 4  2021 Sørlandet   0.445  0.0277         0.398</span></span>
<span><span class="co">#&gt; 5  2021 Trøndelag   0.421  0.0183         0.385</span></span>
<span><span class="co">#&gt; 6  2021 Østlandet   0.441  0.0134         0.414</span></span>
<span><span class="co">#&gt; 7  2022 Sørlandet   0.445  0.0289         0.397</span></span>
<span><span class="co">#&gt; 8  2022 Trøndelag   0.420  0.0179         0.385</span></span>
<span><span class="co">#&gt; 9  2022 Østlandet   0.441  0.0132         0.415</span></span>
<span><span class="co">#&gt; # ℹ 1 more variable: boot_upper97.5 &lt;dbl&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb427"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">gdm_sn_diff</span>,</span>
<span>     limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/unnamed-chunk-122-1.png" width="672"></div>
<p>Prepare export format.</p>
<div class="sourceCode" id="cb428"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gdm_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">gdm_sn_boot</span><span class="op">$</span><span class="va">bootstrap_summary</span></span>
<span></span>
<span><span class="va">gdm_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">gdm_sn_to_exp</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>reference_high <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         i <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">region_name</span>,</span>
<span>         v <span class="op">=</span> <span class="va">boot_value</span>,</span>
<span>         sd <span class="op">=</span> <span class="va">boot_sd</span>,</span>
<span>         <span class="va">reference_high</span>,</span>
<span>         <span class="va">reference_low</span>,</span>
<span>         <span class="va">thr</span>,</span>
<span>         <span class="va">i</span></span>
<span>         <span class="op">)</span></span>
<span></span>
<span><span class="va">gdm_sn_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NorIns</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">inner_join</span><span class="op">(</span><span class="va">gdm_sn_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_genetic_div_semi_nat</span> <span class="op">&lt;-</span> <span class="va">gdm_sn_to_exp</span></span></code></pre></div>
</div>
</div>
<div id="calculation-of-nbbm-indicators" class="section level3" number="10.8.5">
<h3>
<span class="header-section-number">10.8.5</span> Calculation of NBBM indicators<a class="anchor" aria-label="anchor" href="#calculation-of-nbbm-indicators"><i class="fas fa-link"></i></a>
</h3>
</div>
<div id="bumblebees-in-open-landscapes" class="section level3" number="10.8.6">
<h3>
<span class="header-section-number">10.8.6</span> Bumblebees in open landscapes<a class="anchor" aria-label="anchor" href="#bumblebees-in-open-landscapes"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb429"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">bombLepiSurv</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/humlesommerfConnect.html">humlesommerfConnect</a></span><span class="op">(</span>host <span class="op">=</span> <span class="st">"ninradardata01.nina.no"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb430"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">allBombusGressmark2022</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/getAllData.html">getAllData</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"bumblebees"</span>,</span>
<span>                          habitat <span class="op">=</span> <span class="st">"gressmark"</span>,</span>
<span>                          year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                          language <span class="op">=</span> <span class="st">"norsk"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/surveyBarplot.html">surveyBarplot</a></span><span class="op">(</span><span class="va">allBombusGressmark2022</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:bombusGressBarplot"></span>
<img src="insectIndicators_files/figure-html/bombusGressBarplot-1.png" alt="Bumblebees in grasslands 2022" width="672"><p class="caption">
Figure 10.1: Bumblebees in grasslands 2022
</p>
</div>
<div class="sourceCode" id="cb431"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nIter</span> <span class="op">=</span> <span class="fl">999</span></span>
<span></span>
<span><span class="va">hInd2022TrondGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Trond"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb432"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hInd2022SorGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Sor"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb433"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hInd2022OstGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Ost"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb434"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hInd2022VestGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Humler"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Vest"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="butterflies-in-open-landscapes" class="section level3" number="10.8.7">
<h3>
<span class="header-section-number">10.8.7</span> Butterflies in open landscapes<a class="anchor" aria-label="anchor" href="#butterflies-in-open-landscapes"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb435"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nIter</span> <span class="op">=</span> <span class="fl">999</span></span>
<span></span>
<span><span class="va">sInd2022TrondGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Sommerfugl"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Trond"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb436"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sInd2022SorGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Sommerfugl"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Sor"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb437"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sInd2022OstGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Sommerfugl"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Ost"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb438"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sInd2022VestGress</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/calcInd.html">calcInd</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"Sommerfugl"</span>,</span>
<span>                              region_short <span class="op">=</span> <span class="st">"Vest"</span>,</span>
<span>                              habitat <span class="op">=</span> <span class="st">"Gressmark"</span>,</span>
<span>                              year <span class="op">=</span> <span class="fl">2022</span>,</span>
<span>                              nIter <span class="op">=</span> <span class="va">nIter</span>,</span>
<span>                              save.draws <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p>The calculations for earlier years are already present as datasets in the R-package and can be fetched individually with <code>data("dataset")</code>. Once this is done, we can plot the timeseries of values using the function <code>indPlot</code>.</p>
<p>Once loaded, the yearly datasets can be organized through the convenience function <code>collectIndData</code>, and plotted with <code>indPlot</code>.</p>
<div class="sourceCode" id="cb439"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hGressIndVals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/collectIndData.html">collectIndData</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="st">"bumblebees"</span>, </span>
<span>                                habitat <span class="op">=</span> <span class="st">"gressmark"</span>,</span>
<span>                                years <span class="op">=</span> <span class="fl">2009</span><span class="op">:</span><span class="fl">2022</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/indPlot.html">indPlot</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">hGressIndVals</span>,</span>
<span>        confLevel <span class="op">=</span> <span class="st">"95%"</span>,</span>
<span>        lineWidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        dodge <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>        pointSize <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Indicator values for bumblebees in semi-natural landscapes"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">2009</span><span class="op">:</span><span class="fl">2022</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/bombusGressIndPlot-1.png" width="672"></div>
<div class="sourceCode" id="cb440"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sGressIndVals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/collectIndData.html">collectIndData</a></span><span class="op">(</span>taxa <span class="op">=</span> <span class="st">"butterflies"</span>, </span>
<span>                                habitat <span class="op">=</span> <span class="st">"gressmark"</span>,</span>
<span>                                years <span class="op">=</span> <span class="fl">2009</span><span class="op">:</span><span class="fl">2022</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/indPlot.html">indPlot</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">sGressIndVals</span>,</span>
<span>        confLevel <span class="op">=</span> <span class="st">"95%"</span>,</span>
<span>        lineWidth <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        dodge <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>        pointSize <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Indicator values for butterflies in semi-natural landscapes"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_x_continuous</span><span class="op">(</span>breaks <span class="op">=</span> <span class="fl">2009</span><span class="op">:</span><span class="fl">2022</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="insectIndicators_files/figure-html/lepiGressIndPlot-1.png" width="672"></div>
</div>
</div>
<div id="prepare-export" class="section level2" number="10.9">
<h2>
<span class="header-section-number">10.9</span> Prepare export<a class="anchor" aria-label="anchor" href="#prepare-export"><i class="fas fa-link"></i></a>
</h2>
<p>We here fetch the indicator regions as an sf and join the calculated indicator values.</p>
<div class="sourceCode" id="cb441"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">norway_reg_NBBM</span> <span class="op">&lt;-</span> <span class="fu">bombLepiSurv</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/bombLepiSurv/man/get_map.html">get_map</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">region</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Sør"</span>, <span class="st">"Vestlandet"</span>, <span class="st">"Øst"</span>, <span class="st">"Trøndelag"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">region</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">region</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span><span class="op">(</span>geom <span class="op">=</span> <span class="fu">st_union</span><span class="op">(</span><span class="va">geom</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb442"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">hGressIndVals_to_exp</span> <span class="op">&lt;-</span> <span class="va">hGressIndVals</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>reference_high <span class="op">=</span> <span class="fl">1</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         v <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>year <span class="op">=</span> <span class="va">Year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">Region</span>,</span>
<span>         i <span class="op">=</span> <span class="va">Indicator_value</span>,</span>
<span>         ind_sd <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>         ind_lower2.5 <span class="op">=</span> <span class="va">lower95</span>,</span>
<span>         ind_upper97.5 <span class="op">=</span> <span class="va">upper95</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hGressIndVals_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NBBM</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">hGressIndVals_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_bumblebees_semi_nat</span> <span class="op">&lt;-</span> <span class="va">hGressIndVals_to_exp</span></span></code></pre></div>
<div class="sourceCode" id="cb443"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">sGressIndVals_to_exp</span> <span class="op">&lt;-</span> <span class="va">sGressIndVals</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>reference_high <span class="op">=</span> <span class="fl">1</span>,</span>
<span>         reference_low <span class="op">=</span> <span class="fl">0</span>,</span>
<span>         thr <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>         v <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>year <span class="op">=</span> <span class="va">Year</span>,</span>
<span>         region <span class="op">=</span> <span class="va">Region</span>,</span>
<span>         i <span class="op">=</span> <span class="va">Indicator_value</span>,</span>
<span>         ind_sd <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>         ind_lower2.5 <span class="op">=</span> <span class="va">lower95</span>,</span>
<span>         ind_upper97.5 <span class="op">=</span> <span class="va">upper95</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sGressIndVals_to_exp</span> <span class="op">&lt;-</span> <span class="va">norway_reg_NBBM</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">sGressIndVals_to_exp</span>,</span>
<span>            by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"region"</span> <span class="op">=</span> <span class="st">"region"</span><span class="op">)</span>,</span>
<span>            multiple <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">insect_butterflies_semi_nat</span> <span class="op">&lt;-</span> <span class="va">sGressIndVals_to_exp</span></span></code></pre></div>
<div id="eksport-file-final-product-1" class="section level3" number="10.9.1">
<h3>
<span class="header-section-number">10.9.1</span> Eksport file (final product)<a class="anchor" aria-label="anchor" href="#eksport-file-final-product-1"><i class="fas fa-link"></i></a>
</h3>
<!-- Export final file. Ideally a georeferenced shape or raster with indicators values (raw and normalised), reference values and errors. -->
<div class="sourceCode" id="cb444"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">to_export</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"insect_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="va">as.list</span></span>
<span> </span></code></pre></div>
<div class="sourceCode" id="cb445"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">write_indicators</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>,</span>
<span>                             <span class="va">out_dir</span> <span class="op">=</span> <span class="st">"indicators/indicatorMaps"</span>,</span>
<span>                             <span class="va">suffix</span> <span class="op">=</span> <span class="st">".shp"</span><span class="op">)</span><span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.html">system</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"mkdir -p "</span>, <span class="va">out_dir</span>, <span class="st">"/"</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu">write_sf</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>,</span>
<span>           dsn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">out_dir</span>,<span class="st">"/"</span>, <span class="va">x</span>, <span class="st">"/"</span>, <span class="va">x</span>, <span class="va">suffix</span><span class="op">)</span><span class="op">)</span></span>
<span>  </span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">to_export</span>, <span class="va">write_indicators</span><span class="op">)</span></span></code></pre></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="trophic-level-biomass-ratios.html"><span class="header-section-number">9</span> Trophic level biomass ratios</a></div>
<div class="next"><a href="breeding-birds.html"><span class="header-section-number">11</span> Breeding bird indicators</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#insect-indicators"><span class="header-section-number">10</span> Insect indicators</a></li>
<li><a class="nav-link" href="#introduction-2"><span class="header-section-number">10.1</span> Introduction</a></li>
<li>
<a class="nav-link" href="#about-the-underlying-data-4"><span class="header-section-number">10.2</span> About the underlying data</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#representativity-in-time-and-space-5"><span class="header-section-number">10.2.1</span> Representativity in time and space</a></li>
<li><a class="nav-link" href="#original-units-4"><span class="header-section-number">10.2.2</span> Original units</a></li>
<li><a class="nav-link" href="#temporal-coverage-5"><span class="header-section-number">10.2.3</span> Temporal coverage</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#ecosystem-characteristic-5"><span class="header-section-number">10.3</span> Ecosystem characteristic</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#norwegian-standard-3"><span class="header-section-number">10.3.1</span> Norwegian standard</a></li>
<li><a class="nav-link" href="#un-standard-2"><span class="header-section-number">10.3.2</span> UN standard</a></li>
</ul>
</li>
<li><a class="nav-link" href="#collinearities-with-other-indicators-8"><span class="header-section-number">10.4</span> Collinearities with other indicators</a></li>
<li>
<a class="nav-link" href="#reference-condition-and-values-4"><span class="header-section-number">10.5</span> Reference condition and values</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#reference-condition-4"><span class="header-section-number">10.5.1</span> Reference condition</a></li>
<li><a class="nav-link" href="#reference-values-thresholds-for-defining-good-ecological-condition-minimum-andor-maximum-values-4"><span class="header-section-number">10.5.2</span> Reference values, thresholds for defining good ecological condition, minimum and/or maximum values</a></li>
</ul>
</li>
<li><a class="nav-link" href="#uncertainties-6"><span class="header-section-number">10.6</span> Uncertainties</a></li>
<li><a class="nav-link" href="#references-7"><span class="header-section-number">10.7</span> References</a></li>
<li>
<a class="nav-link" href="#analyses-2"><span class="header-section-number">10.8</span> Analyses</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#calculation-principles-for-norins-indicators"><span class="header-section-number">10.8.1</span> Calculation principles for NorIns indicators</a></li>
<li><a class="nav-link" href="#calculation-principles-for-nbbm-indicators"><span class="header-section-number">10.8.2</span> Calculation principles for NBBM indicators</a></li>
<li><a class="nav-link" href="#data-sets-5"><span class="header-section-number">10.8.3</span> Data sets</a></li>
<li><a class="nav-link" href="#calculation-of-norins-indicators"><span class="header-section-number">10.8.4</span> Calculation of NorIns indicators</a></li>
<li><a class="nav-link" href="#calculation-of-nbbm-indicators"><span class="header-section-number">10.8.5</span> Calculation of NBBM indicators</a></li>
<li><a class="nav-link" href="#bumblebees-in-open-landscapes"><span class="header-section-number">10.8.6</span> Bumblebees in open landscapes</a></li>
<li><a class="nav-link" href="#butterflies-in-open-landscapes"><span class="header-section-number">10.8.7</span> Butterflies in open landscapes</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#prepare-export"><span class="header-section-number">10.9</span> Prepare export</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#eksport-file-final-product-1"><span class="header-section-number">10.9.1</span> Eksport file (final product)</a></li></ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/NINAnor/ecosystemCondition/blob/master/insectIndicators.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/NINAnor/ecosystemCondition/edit/master/insectIndicators.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Indicators for Ecosystem Condition in Norway</strong>" was written by Anders L. Kolstad. It was last built on 2023-12-05.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
